# Methods {#methods}

```{r ch2-setup, include = F}

library(plyr)
library(dplyr)
library(ggplot2)
library(tmap)
library(ggmap)

wd <- "/Users/joelewis/Documents/PhD/Thesis/bookdown/chapter_2/"
output = "latex"

# generate figures
source("chapter_2/generate_study_flow_diagram.R")

```
## Chapter Overview

This chapter gives an overview of the clinical study which underpins this thesis, and the laboratory and computational procedures used in analysis. Further details are given in the individual chapters, where necessary.

## Study site

### Malawi

Malawi is a country of 17.5 million people in South-Eastern Africa[@NSOM]. It is one of the poorest countries in the world: it is a low income country under the World Bank classifcation, with a 2017 Gross National Income (GNI) per capita of \$320 in US dollars[@WorldBank], and was ranked 171st of 189 countries in 2017 by the human development index (HDI), a composite statistic of life expectancy, education and income per capita indicators[ (UNDP human development reports)]@UnitedNationsDevelopmentProgramme2018. In 2010, 71% of the population was estimated to survive on less than \$1.90 per day. Life expectancy at birth in 2017 was 63 years, and though significant progress is being made, neonatal and under-5 mortality remains high at 23 and 55 per 1000 live births, respectively. The population is largely rural (83% in 2017), with a young population (44% under the age of 15, 2017) and high fertility rate[@WorldBank]. Malaria is endemic, and there is an ongoing generalised HIV epidemic: adult HIV prevalance (age 15-49) was estimated to be 9.6% in 2017 (UNAIDS), though falling from a peak of 16.6% in 1999[@UNAIDS]. HIV antiretroviral therapy (ART) national scale up began in 2004 and in 2017 71% of elibigible adults and children were estimated to be recieveing ART[@UNAIDS]. It is classed by the WHO as a high-TB/high-HIV burden country, with an estimated TB incidence rate of 131 (95% CI 70-210) cases per 100000 population per year[WorldHealthOrganisation2018]. 

It has a subtropcial climate, with three main seasons: a warm wet season from November to April, a cooler dry winter period from May to August and a hot dry period from September to October. Blantyre city, the location of the study in this thesis, is the second city of Malawi with a population of 585000. It is located in Blantyre district, population 995000 in 2018[@NSOM], in the Shire highlands at an altitude of 1000m (Figure \@ref(fig:malawi-map)).

```{r malawi-map, warning = F,  echo = F, message = F, fig.scap="Map of study site", out.extra='', fig.cap = "Malawi, showing administrative boundaries (North, Central, and South regions), Lilongwe, the capital city and Blantyre, the study location. Source: openstreetmap.org, used under Creative Commons Attribution ShareAlike 2.0 licence CC-BY-SA", fig.align= 'center'}


afbox <- bbox <- c(left = -21.22, right = 55.02, top= 17.17, bottom = -37.75)

bbox <- c(left = 25.840, right = 42.737, top= -8.277, bottom = -18.167)
bbox2 <- c(left = 25.955, right = 38.717, top= -8.277, bottom = -18.167)
#get_stamenmap(bbox2, zoom = 8) -> m
get_stamenmap(bbox2, zoom = 7, maptype = "terrain-background") -> m2

get_stamenmap(afbox, zoom = 4, maptype= "terrain-background") -> a

#ggmap(m2) 

cities <- data.frame(city = c("Blantyre", "Lilongwe"), lat = c(-15.7667,-13.9626), lon = c(35.0168, 33.7741))


ggmap(a) + geom_polygon(data = mal2, aes(x= long, y = lat, group = group), fill = "black") +
  theme_inset() +
  geom_rect(xmin = bbox2["left"], xmax = bbox2["right"], ymin = bbox2["bottom"], ymax = bbox2["top"], fill = NA, colour = "black" )-> p1

ggmap(m2) +
  geom_path(data = mal2, aes(x= long, y = lat, group = group), alpha = 0.8) + 
  geom_point(data = cities, aes(x = lon, y = lat)) + 
  geom_text(data = cities, aes(x = lon, y = lat,label = city,  vjust = -0.5, hjust = 0.5, size = 7, fontface = "bold")) +
  theme_void() + theme(legend.position = "none") +  inset(grob = ggplotGrob(p1), xmin = 26.5, xmax = 31.5, ymin = -13, ymax = -9)


```

### Queen Elizabeth Central Hospital

Queen Elizabeth Central Hospital (QECH), located in Blantyre city, is the tertiary referral hospital for the Southern Region of Malawi. It has 1300 beds but often operates above capacity, and is the only site providing free inpatient healthcare to the adult population of Blantyre district. Since 2011 it has had a dedicated emergency department for adults, the Adult Emergency and Trauma Centre, staffed 24 hours a day. Since 2015 (and for the whole of the study period), attendees to the AETC must be referred from a primary health clinic. Adults attending the AETC are triaged by a nurse and then reviewed by a doctor or clinical officer; if admission under a specialty team (including medicine) is deemed appropriate then a patient will be reviewed by an intern or registrar from the admitting specialty and usually by a consultant within 24 hours. There is a 6-bed AETC resusitation area in which oxygen concentrators, cardiac monitors and a defibrillator are available; none of these items are available in the rest of the AETC. 

There are two dedicated single-sex medical wards, each of approximately 60 beds, and one mixed-sex TB ward. Male and female high-dependency units (HDUs), each with a capacity of six beds, have oxygen concentrators (or, if available, oxygen cylinders) to deliver supplemental oxygen. The medical wards are staffed by two or three trained nurses and a variable number of nursing students. Basic nursing care is usually provided by a patients relative or friend, called a 'guardian.' Patients on the medical wards are reviewed twice-weekly by a consultant physician and then variably at other times by junior doctors, clinical officers or medical students depending on the availability of staff. Malawi national treatment guidelines suggest ceftriaxone for the treatment of sepsis requiring hospitalisation. 

### Participating Laboratories

#### Malawi-Liverpool-Wellcome Clinical Research Programme

The Malawi-Liverpool Wellcome Trust Clinical Research Programme was established in 1995 and since them has been active in researching priority health issues in Malawi. It is an affiliate of the Malawi College of Medicine, and is based in the grounds of QECH in Blantyre. It runs an on-site microbiology laboratory which has provided an aerobic blood culture service to QECH since 1998, and also provides CSF microscopy and culture. Bacterial culture is carried out as per British Society of Antimicrobial Chemotherapy (BSAC) guidelines[@BritishSocietyforAntimicrobialChemotherapy2013] and the laboratory adheres to UK NEQAS external quality control. It is core funded by the Wellcome Trust.

#### Malawi College of Medicine Tuberculosis Laboratory

#### Wellcome Trust Sanger Institute

The Wellcome Sanger Institute is a research institute based in Hinxton, UK, which was established in 1993, and undertakes research in all aspects of genomics including bacterial genomics as part of the parasite and microbes programme. It has one of the largest DNA sequencing facilities in the world as well as high performace computing clusters. It is funded by the Wellcome Trust.

## Clinical Study

The DASSIM (Developing an Antimicrobial Strategy for Sepsis in Malawi) study was an observational cohort study, recruiting from the AETC at QECH with two broad aims: firstly, to describe the presentation, aetiology and determinents of outcome in sepsis in Malawi; and secondly to determinents of carriage of ESBL-E in sepsis survivors.

### Entry Criteria

The study was open for recruitment between 19 February 2017 and 2 October 2018; there were three arms. Firstly, in order to define the aetiology of sepsis, adults with sepsis attending AETC were recruited as early as possible in their attendance to QECH following triage. Secondly, in order to clearly define the relative effects of antimicrobial exposure versus hospital admission in ESBL-E acquisition and carriage, antibiotic unexposed adults attending AETC with no plan for antimicrobial administration were recruited. Finally, to define baseline flux of ESBL-E antibiotic-unexposed community members were recruited. Detailed inclusion and exclusion criteria for each arm are given shown in Table \@ref(tab:inc-crit). For logistic reasons, recruitment occured 7am - 5pm Monday to Friday. 

Exclusion criteria were the same for all arms of the study: Particiapnts were not eligible for enrollment if they were unable to give informed consent and no guardian available to provide proxy consent; they spoke neither English nor Chichewa; or they lived > 30km from Blantyre city. The antibiotic-unexposed inpatients and community members were matched on age (+/- 5yr) and sex to sepsis survivors (defined as patients surviving to 28 days). In addition community members were matched on location to sepsis survivors; putative households for recruitment were identified by random walk from the houses of surviving sepsis participants, with initial direction established by spinning a bottle on the floor.  

```{r inc-crit, echo = F, warning= F, tidy = F}

t1 <- read.csv(paste0(wd, "t1_inclusion_criteria.csv"), check.names = F, stringsAsFactors = F)


knitr::kable(t1, format = "latex", booktabs = T, longtable = T, caption = "Study inclusion criteria") %>%
  kable_styling(full_width = F)  %>% 
  column_spec(1, width = "10em") %>%
  column_spec(2, width = "25em") %>%
 footnote(general = "AETC = Adult Emergency and Trauma Centre")


```

### Study Visits and Patient Sampling

#### Enrollment assessment and first six hours

An overview of the study visit schedule is shown in Figure \@ref(fig:study-sampl-flow-diag). At enrollment, following consent a baseline questionnaire was completed (by the guardian if the participant was obtunded) to capture background demographic and clinical data, a Chichewa-language EQ-5D heath related quality of life questionairre completed, and sample collection undertaken (see below). For hospitalised patients, enrollment aimed to be as soon after triage as possible, and arm 1 (sepsis) patients underwent a more intensive assessment over the first six hours of enrollment. Data on therapies administered over this time period were captured by hourly review by a member of the study team, including the time of administration of  antimicrobial therapy and volumes of intravenous fluid administered, the latter visually comfirmed by the study team member. Hourly vital signs and standardised bedside assessment of intravascular status were performed: central capillary refill, limb temperature and passive leg raise. For this latter test, blood pressure and pulse rate were measured with the participant sitting at 45 degrees, and one minute after lying the participant supine and lifting the legs to 45 degrees. All blood pressure mesurements were made noninvasively with an automated cuff (Omron M2, Omron, Japan), oxygen saturations measured with a dedicated study pulse oximeter (Contec CM50, Contec Medical Systems, China), and temperature measured in the axilla with a digital thermometer (Omron FWH000, Omron, Japan).

All treatment decisions were at the discretion of the parrticipant's attending healthcare worker.

#### Subsequent visits

Hospitalised participants were briefly reviewed daily Monday to Friday following admission whilst on the wards to capture details of therapies administered. For the 72 hours following admission, vital signs were repeated and an assessment made of intravenous fluid administered; if records of fluid balance were incomplete (as wa anticipated to be the case in the majority of cases) then fluid administered was estimated from participant- or guardian-reported number of bags of fluid administered.

The visit schedule for the study is shown in Figure \@ref(fig:study-sampl-flow-diag). Arm 1 (sepsis) and 2 (hospital inpatients) were reviewed by a study team member 7 and 28 days following enrollment, and at 3 and 6 months; comunity members were reviewed at 28 days and 6 months. At each visit, details of any antimicrobials or other therapies recieved and any health care facility contact were collected, and the Chichewa language EQ-5D was repeated. Samples were collected as below. 

These study visits occured preferentially at QECH, but at the patient's home if it was difficult for them to attend the hospital. If patients missed a scheduled visit, then attempts were made to contact them by telephone. If these attempts were unsucessful then attempts to visit them at home were undertaken by a member of the study team.

#### Blood, urine, and stool, sputum and CSF collection 

Blood was collected from arm 1 (sepsis) participants only on enrollment and at the 4-week visit. At baseline, blood was collected aseptically directly into collection tubes with a vacutainer device with the following order of draw and volumes: one BacT/Alert (BioMerieux, France) aerobic blood culture bottle (7-10ml), one BD BACTEC Myco/F Lytic mycobacterial culture bottle (3-5ml) (Beckton Dickinson, United States), one serum (10ml) and two EDTA (4ml, both Grenier Bio-One, Austria) samples. Finger prick for capillary blood was used for point of care diagnostics as described below. At the 4-week visit, 10mls of convalescent serum was collected. At baseline, urine was collected from all arm 1 (sepsis) participants into a sterile polypropylene universal container (Alpha Laboratories, UK) either by the participant themselves (if this was possible) or with the aid of a disposable bedpan.

At the enrollment visit and all other visits for all arms of the study stool was collected into a sterile polypropylene universal container (Alpha Laboratories, UK) with aseptic technique. If it was not possible for a patient to provide stool then a rectal swab was taken using a sterile rayon-tipped swab (Technical Service Consultants Ltd, UK) inserted into 2-3cm into the rectum, rotated for approximately 10 seconds and placed into Amies gel media for transport to the laboratory.

The decision to collect sputum for Xpert testing for tuberculosis or to perform lumbar puncture (LP) rested with the participant's attending healthcare worker. Lumbar puncture, where done, was carried out by QECH AETC or medicine department doctors or clinical officers and 7-10ml of cerebrospinal fluid (CSF) was aseptically collected. Sputum samples were collected in sterile polypropylene universal container and trasported to the Malawi College of Medicine (CoM) tuberculosis laboratory. Myco/F lytic bottles for mycobacterial blood culture were also transported to the CoM tuberculosis laboratory; all other samples were transported to the MLW laboratories in the first instance.

```{r study-sampl-flow-diag, echo = F, warning = F, fig.scap="Overview of patient sampling schedule", out.extra='', fig.cap= "Overview of patient sampling schedule.FBC = Full blood count, U\\&E = Urea and elctrolytes, mRDT = Malaria rapid diagnostic test, uLAM = Urinary LAM, FASH USS = Focussed assessment for HIV-associated tubercusosis ultrasound scan", fig.align= "center", out.width= "400px"}

knitr::include_graphics(paste0(wd, "/figures/study_flow_diagram.pdf"))

```

#### Imaging: chest x-ray and ultrasound scanning

Ultrasound examination following the FASH (Focussed ASsessment for HIV-associated tuberculosis) protocol[@Heller2012] was undertaken for all sepsis participants by me, following training with a UK-registered consultant radiologist. The protocol aims to ascertain presence or absence of six features likely to be associated with extrapulmonary TB: pericardial effusion, abdominal lymph nodes, pleural effusion, ascites, hypoechoic focal liver lesions or hypoechoic focal splenic lesions. Scanning was undertaken with a xxxxxx device with a curvilinear probe; images were saved and 10% of scans were read by a UK-registered consultant radiologist for quality control.

Decision to undertake chest x-ray was at the discretion of the participant's attending healthcare worker. If chest x-ray was done then digital pictures were taken and stored in the study database.

### Outcomes and sample size calculations

The two co-primary outcomes for the study are:

1) 28 day mortality.

2) Aquisition of ESBL-E detected in stool by aerobic culture by 28 days.

Two power calculations were undertaken: to inform the size of the sepsis cohort, and one for the non-sepsis participants. The sepsis component was powered to detect clinically relevant risk factors for death; we made the a priori assumption that a risk ratio of 2 or more is clinically relevant in this setting and patient population. The initial calculation suggested 250 patients with severe sepsis with 50% mortality (as seen in previous studies in Malawi) would have 80% power to detect risk factors for death with prevalence of 20-50% that confer a risk ratio (RR) of 1.5-3; therefore, a RR of 2 is likely to be detected. Logistic considerations resulted in a reduction of a feasible sample size to 225; post-hoc power calculation suggested 80% power to detect risk factors coferring a RR of 2 with 25-50% prevalence, assuming 50% mortality.

For the second co-primary outcome, assuming 30% of participants with sepsis (and hence antimicrobial exposure) aquire ESBL-E by 28 days, and that 50% of them die by day 28, 125 antibiotic unexposed hospitalised participants would give 80% power to detect a 50% relative difference in aquisition. Logistic considerations resulted in the target sample size being reduced to 100 participants; revised post-hoc power calculation suggested that the 80% power to detet a 50% relative difference was maintained.

In order to define the baseline flux of ESBL-E organisms and ESBL genes, one community member for each antibiotic unexposed hospitalised participant, giving a target sample size of 100.

## Diagnostic Laboratory Procedures

Results of all laboratory diagnostic procedures were fed back to participant's attending healthcare worker in real time. If any investigation result became available after the participant was discharged (e.g. Mycobacteraemia was identified) then they were contacted and called back to the hospital to be reviewed by me, and referred for care to the relevant QECH department.

### Point of care diagnostics

Point of care tests were carried out on capillary blood for capillary lactate (Lactate Pro 2, Arkray, Japan), for _P. falciparum_ HRP-2 antigen (Paracheck Pf, Orchid Biomedical, India) and, if HIV status was unknown, to test for HIV antibodies using Determine HIV 1/2 kit (Abbott Diagnostic Division) and Unigold HIV 1/2 kit (Trinity Biotech Inc.) following Malwian national guidelines[@MinistryofHealth.GovernmentofMalawi2015a].

### Laboratory diagnostics

#### Haematology and biochemistry

Automated full blood counts (Beckman Coulter HmX Haematology Analyser, Beckman Coulter, USA) were undertaken in the MLW laboratories on EDTA whole-blood samples, as were CD4 cell counts (Becton Dickinson FACSCount, Becton Dickinson, USA) for HIV-infected participants. Serum samples were allowed to settle upright for 30-60 minutes and then centrifuged (at 1300 g for 10 min). Biochemistry testing for urea and electrolytes (Beckman Coulter AU480 Chemistry Analyser, Beckman Coulter, USA) was undretaken immediately and a maximum of three 1.8ml aliquots of serum stored at -80C for later analysis.

#### Aerobic blood and CSF culture 

Blood in aerobic culture bottles was incubated in an automated system (BacT/Alert BioMerieux) and identified to species level using the API system (Biomerieux, France) and standard techniques (ref). Anaerobic culture was not available. Coagulase-negative _Staphylococci_, _bacillus spp._, _diptheroids_ and alpha-haemolytic _Streptococci_ other than _S. pneumoniae_  were considered as contaminants. Antimicrobial sensitivity testing was undertaken using the disc diffusion method following BSAC guidelines to amoxicillin, chloramphenicol, co-trimoxazole, gentamicin, ciprofloxacin and ceftriaxone.

CSF, where available, was cultured onto blood, chocolate and Sabouraud agar and then identification of any growth undertaken as above. 

#### Mycobacterial blood culture

Blood in Myco/F Lytic bottles was cultured at 37C in and inspected daily for the first 14 days with a handheld UK Wood's lamp, and once every two days thereafter. Contents of the bottles were centrifuged (3000xg for 20 mninutes) within 48hr of detection of flourescence, examined by ZN and gram stain to exclude contamination, then inoculated into mycobacterial growth indicator tubes (MGIT, Becton Dickinson Diagnostic Systems, United States) for up to 6 weeks. Isolates were classified as _M. tuberculosis_ or nontuberculous Mycobacteria by microscopic cording and MPT-64 lateral flow assays (TAUNS Laboratories, Japan). 

After 8 weeks of culture of Myco/F Lytic bottles if no flourescence was seen, then centrifugation and microscopy was carried out. If no organisms wereseen, the culture was reported as "no growth."

#### Sputum Xpert

Sputum Xpert testing (Cepheid, United States) was carried out as per the manufacturers instructions: sputum specimens were mixed with sample reagent and incubated at room temperature for 15 minutes. The liquefied specimen was then loaded into the Xpert MTB/RIF test cartridge for processing.

#### Urinary LAM

Urinary LAM testing was carried out on all available urine samples from HIV-infected participants, using the Alere Deterine TB LAM Ag lateral flow assay (Alere, United States) following the manufacturer's instructions. Frozen urine samples were used: they were allowed to come to room temperature and then briefly centrifuged to remove debris. 60 microliters of urine was applied to the sample pad and the result read after 25 minutes by comparing to the provided reference scale card. The results were read independently by two readers, with a tie-break read by a third reader inthe even of disagreement.

#### Selective stool culture for ESBL-E

Stool and rectal swabs recieved in the laboratory were stored at 4C pending processing, before being plated onto commercially available ESBL selective media (CHROMagar ESBL media, CHROMagar, France) and cultured aerobically at 37C overnight. Rectal swabs were streaked directly onto the plate, or a cotton tipped applicator used for solid stool. CHROMagar is also chromagenic: _E. Coli_ colonies are pink, _Klebsiella spp._, _Enterobacter spp._ and _Citrobacter spp._ blue and other species white. Blue or white colonies were speciated using the API 20E system. Morphpologically distinct colonies were confirmed to be ESBL producers using combination disc methods: the putative ESBL producer was cultured overnight on ISO-sensitest agar with discs of cefotaxime and ceftazidime (30 micrograms) with and without clavulanic acid (10 micrograms), and ESBL production confirmed if ther was a difference of at least 5mm between discs with and without clavulanic acid . For organisms likely to carry a chromasomal AmpC and hence be capable of hydrolysing cefotaxime and ceftazadime, ( _Enterobacter spp._, _Citrobacter freundii_, _Morganella morganii_, _Providencia stuartii_, _Serratia spp._, _Hafnia alvei_) cefipime (30 micrograms), an AmpC-stable cephalosporin was used with and without clavulanic acid (10 micrograms). Antimicrobial sensitivity testing to meropenam, amikacin, gentamicin, co-trimoxazole, chloramphenicol and ciprofloxacin following BSAC guidelines[@BritishSocietyforAntimicrobialChemotherapy2013] was undertaken for a subset of confirmed ESBL-producers.

#### Acute and convalescent serologies

_PHE stuff_

## Molecular methods

One of each morphologically distinct colony from ESBL-E selective culture was taken forward for DNA extraction and whole genmone sequencing. DNA was extracted from overnight nutrient broth culture using the Qiagen DNA mini kit (Qiagen, Germany) following the manufacturers instructions. Extracted DNA was shipped to the Wellcome Sanger Institute where it underwent library preperation according to the Illumina protocol and paired-end sequencing on Illumina HiSeqX ( _check this_ ) by the WSI DNA pipelines team.

## Bioinformatics

All analyses were undertaken on the WSI computing cluster running Ubuntu v12.04.2, Precise Pangolin. Quality control of reads for each sample was carried out using metrics provided the WSI QC pipleine including total number of reads, and statistics obtained by mapping a random 100 Mbases fron each sample to a reference and calculating depth of coverage, number of heterogeneous SNPs, GC content and insert size. Kraken[@Wood2014] v0.10.6, a kmer-based taxonomic assignemnt tool, was used to assign reads to a species and to identify suspected sample contamination or species misclassification. Reads were _de novo_ assembled with SPAdes[@Bankevich2012a] v3.11.0, with default settings. Assembly metrics of quality (e.g. number of contigs, largest contig, N50 [the contig length on which at least half of the bases in an assembly have been assembled]) were calculated using QUAST[@Gurevich2013] v4.6.0 and completeness and contamination of the assemblies assessed by checkM[@Parks2015] v1.0.7, a tool that uses co-located sets of lineage associated genes to assess genome completeness and contamination. Contaminated samples (with > 25% contamination as defined by CheckM) or poor assemblies (with N50 < xxxxx) were discarded. 

Prokka[@Seemann2014a] v1.5 with a genus specific database from RefSeq was used to annotate the retained assemblies. The Roary v1.007 pan-genome pipeline[@Page2015] was used to construct a core genome with a blastp percentage identity of 95%; genes were considered core if they were present in 99% or more of isolates. A core gene multiple sequence alignment was generated using maaft[@Katoh2013] v7.205, SNP-sites identified using SNP-sites[@Page2016] v2.4.1 and the resultant SNP alignment used to build a maximum likelihood phylogenetic tree using IQ-TREE[@Nguyen2015] v1.6.3. Trees were visualised in the ggtree v1.14.4 package[@Yu2017] in R.

Ariba[@Hunt2017a] v2.12.1 was used to identify AMR-associated genes using the SRST2[@Inouye2014] database, to identify plasmid replicons using the PlasmidFinder database[@Carattoli2014] and to perform _in silico_ muli-locus sequence typing (MLST) using the database from http://mlst.warwick.ac.uk/mlst/dbs/Ecoli accessed via www.pubmlst.org. _E. coli_ phylogrouping was performed with a quadruplex _in silico_ PCR using the Clermont scheme[@Clermont2013]. To track carriage of similar bacteria within and between individuals, clustering of the core gene alignment was performed using hierarchical BAPS (bayesian analysis of population structure) algorithm[@Cheng2013], implemented in the rhierbaps package v1.1.0 in R, with 2 levels. To track putative mobile genetic elements (MGE), a difficult task with short-read sequencing, isolates ESBL-gene containing contigs were identified using BLASTn[@Altschul1990] v2.7.0 with and then contigs clustered using cd-hit[@Li2006] v4.6 with sequence identity threshold 0.9, word size 5 and accurate mode (command line flag -g) to produce mutually exclusive ESBL-gene contig groupings.

## Statistical Analysis

Details of anaysis methods are given in the relevant chapters. All analyses were undertaken using R v3.5.1 and any Bayesian modelling using Markov Chain Monte Carlo (MCMC) methods using Stan v2.18 via the Rstan interface with R. This thesis was written using the bookdown package in R and the code to generate it as well as all analysis scripts are avaiable at https://github.com/joelewis101/thesis. 

## Study Team

I am the principal investigator of the study, and led a study team which consisted of: Emma Smith, a medical doctor who managed the running of the study from March 2017 - May 2017; study nurses Lucy Keyala, Grace Mwaminawa and Tusekile Phiri who recruited patients, collected samples and completed follow up visits; field workers Witness Mtambo and Gladys Namacha who recruited patients, collected samples and completed follow up visits; and laboratory technicians Madlitso Mphasa and Rachel Banda who processed laboratory specimens, did stool culture and DNA extractions, and the uLAM testing. The MLW core laboratory staff undertook haematology and biochemistry testing and blood and CSF culture, and the Malawi CoM tuberculosis laboratory staff carried out tuberculosis culture and Xpert testing.

## Data Collection and Storage

Data collection used both the Open Data Kit software (ODK) and structured TeleForm paper forms. Electronic ODK forms were loaded onto Asos ZenTouch tablets running Android using the ODK Collect Android app. Completed forms were pushed daily to the dedicated secure study SQL (structired query language) database built and administered by the MLW data team. Teleform forms were checked by me, batched and scanned by the MLW data team, and variable values automatically extracted by the TeleForm system; validation was undertaken by the MLW data teame and data queries generated for missing or invalid values. Queries were resolved either by manually reconciling with the forms or by discussion with the clinical team. Once the data passed validation it was pushed to the SQL database. Completed paper TeleForm records were stored securely in the MLW data department. All data on the study database is stored securely with access restricted to the study PI and the database adminstrators in the MLW data department. Other paper records including consent forms were stored in a locked cabinet with access restricted to members of the study team. Results of laboratory investigations in the MLW laboratory were stored in the MLW PreLink laboratory information management system (LIMS), anonymised and linked only to the participant unique study ID number. For analysis, anonymised data were extracted from the study SQL and LIMS databases as comma delimited files.

## Ethical Approval, Consent and Participant Remuneration

The study protocol was granted ethical approval by the Malawi College of Medicine Research Ethics Committee (COMREC), protocol number P.11/16/2063 and the Liverpool School of Tropical Medicine (LSTM) Research Ethics Committee, protocol number 16-062. LSTM acted as the study sponsor. All study team member were trained in NIHR Good Clinical Practive (GCP). 

All participants in the study provided infomred, written consent, as follows: if a patient lacked capacity to provide informed consent themselves then informed consent was sought from the patient’s representative, usually their guardian. Patients with no representative and lacking capacity to provide informed comsent were not recruited. Initially, eligible patients (or their representative) were approached by a member of the study team and the study was explained to them including study procedures, risks and benefits, financial and confidentiality considerations and how to obtain more information. Written patient information leaflets were provided in English and Chichewa. If the patient/representative was willing to enter the study then they were asked to sign and date two copies of the consent form, and provided with a copy of the form to keep. If either the patient or representative was unable to read then the consent form wasread to them by the study team, and witnessed by an additional staff member who is not a member of the study team. If the patient (or their representative) agreed to enter the study, then the witness signed and dated the form. Any patient who was enrolled to the study after consent from a representative and who then regained the capacity for informed consent was then approached independently for informed consent.
A patient could withdraw at any time without giving a reason.

Hospitalised patients were not financially compensated for their time and/or transport costs, but all other participants were, follwing standard MLW guidelines: 500MWK (approximately USD\$0.7 at February 2019 exchange rates) was provided to participants if they were visited at home and 2000MWK (approximately USD\$2.8 at February 2019 exchange rates) or their reported transport costs (whichever was higher) for participants who attended QECH for study purposes.

