# ESBL-E carriage in Malawian adults in health and disease

\chaptermark{ESBL-E carriage}

```{r ch5-setup, include = F}

library(plyr)
library(tidyverse)
library(knitr)
library(kableExtra)
library(grid)
library(ggpubr)
library(pheatmap)
library(RColorBrewer)
library(scales)
library(lubridate)
library(survival)
library(ggsci)
library(survminer)
library(glue)
library(broom)


wd <- "chapter_5/"
output = "latex"
T <- TRUE

specify_decimal <- function(x, k) trimws(format(round(x, k), nsmall=k))


# generate figures
#source("chapter_5/make_consort_diagram.R")
#source("final_cleaning_scripts/generate_visits_df.R")
source("final_cleaning_scripts/load_and_clean_followup_and_enroll_labelled.R")
source("other_scripts/summary_table_functions.R")
source("final_cleaning_scripts/make_composite_hivstatus_variable.R")
source("final_cleaning_scripts/load_and_clean_hourly.R")
source("final_cleaning_scripts/load_and_clean_bloods.R")
#source("final_cleaning_scripts/load_and_clean_aetiol.R")
#source("final_cleaning_scripts/load_and_clean_hourly.R")
source("final_cleaning_scripts/load_and_clean_upto72.R")
source("final_cleaning_scripts/load_and_clean_post72.R")
source("final_cleaning_scripts/load_and_clean_hosp_oc.R")
#source("final_cleaning_scripts/load_and_clean_time_to_ab.R")
#source("final_cleaning_scripts/load_and_clean_fluid_hr1_to_6.R")


# lims




#panel data helper functions
source("other_scripts/panel_data_helpers/expand_covariates.R")
source("other_scripts/panel_data_helpers/sort_out_tb_rx_on_discharge.R")
source("other_scripts/panel_data_helpers/shuffle_a_in_b2.R")
source("other_scripts/panel_data_helpers/strip_post_dropout_rows.R")
source("other_scripts/panel_data_helpers/extract_covariate_exposure.R")
source("other_scripts/panel_data_helpers/collapse_covariates.R")
source("other_scripts/panel_data_helpers/ditch_everything_after_first_1.R")
source("other_scripts/panel_data_helpers/mstate_helper_functions.R")
source("other_scripts/panel_data_helpers/splice_ESBL2.R")

names(enroll)[names(enroll) == "data_date"] <- "enroll_date"
followup <- merge(followup, select(enroll, pid, arm, enroll_date), all.x = T)
followup$t <- followup$data_date - followup$enroll_date

# and load lims
source("final_cleaning_scripts/load_and_clean_lims.R")

```

## Chapter Overview

This chapter presents the longitudinal ESBL-E colonisation status of sepsis survivors and two comparator cohorts: antimicrobial-unexposed inpatients and community members. In total, 425 participants were recruited: 225 participants with sepsis, and 100 each of antimicrobial-unexposed inpatients and community members. Stool was sampled at 5 time points over six months. 1416 stool samples were collected and 51% (723/1417) of samples grew 1032 bacteria, most commonly  _Escherichia coli_ (n = 686) and _Klebsiella pneumoniae_ (n = 245). Baseline ESBL-E carriage prevalence was 49% (95% CI 42-56%) in participants with sepsis, 41% (95% CI 32-52%) and in antimicrobial-unexposed inpatients (both on the day of admission) and 28% (95% CI 20-38%) in "healthy" community members. In multivariable modelling, receipt of cotrimoxazole preventative therapy (CPT), hospitalisation with the previous 4 weeks, use of unprotected water sources, household crowding, and sample collection during the rainy season were all associated with ESBL-E colonisation at enrolment. This suggests that in the community, within-household person to person as well as environmental transmission may be important.

ESBL-E carriage prevalence rose rapidly after admission in antimicrobial-exposed participants, to 78% (95% CI 71-84%) by the day 7 visit, a rise which was not seen in the antimicrobial-unexposed arm of the study (51% at day 7 visit, 95% CI 38-64%), suggesting that antimicrobial exposure and not hospitalisation per se is driving carriage. However, this conclusion is open to confounding because antimicrobial-exposed participants differ from antimicrobial-unexposed in a number of important ways, including a lower HIV prevalence and shorter median length of hospital stay. Attempts to control for this confounding with logistic regression failed: collinearity and a small dataset resulted in very uncertain parameter estimates. In addition, aggregate prevalences obscure a complex pattern of shifting between colonised and non-colonised states at the individual level, and it is not clear whether this represents intermittent ESBL-E shedding, imperfectly sensitive tests, or true acquisition and loss events on a short time scale. I describe these difficulties, and outline the methods that will be used to address them over the next three chapters.

## Introduction and chapter aims

In the preceding two chapters I have described significant exposure to broad-spectrum antimicrobials in a sepsis cohort. I now turn my attention to the consequences of this antimicrobial exposure in terms of acquisition and carriage of ESBL-E and present a description of longitudinal carriage of ESBL-E in sepsis survivors, plus two comparator cohorts of antimicrobial unexposed inpatients and community members. I also attempt to describe risk factors for carriage.

Data from sSA suggest that ESBL-E gut mucosal carriage is common (Chapter 1), but routes of transmission are unknown. Antimicrobial exposure and hospitalisation have been associated with ESBL-E carriage, but such longitudinal cohort data that do exist are associated with short follow up time, often not following participants beyond hospital discharge. The role of hospital acquisition of ESBL-E and carriage into the community in driving drug-resistant infection versus community transmission is therefore unclear. In addition, the mechanism of ESBL-E acquisition in hospitalised adults in sSA including the relative effects of antimicrobials versus hospitalisation - exactly the understanding that would be needed to design effective interventions - is unknown. 

The aims of this chapter are therefore threefold: 

1. To present the details of recruitment, follow up and ESBL-E colonisation status of the participants recruited to the clinical study underpinning this thesis;

2. To explore associations of baseline ESBL-E colonisation to understand potential community ESBL-E transmission routes;

3. To explore associations of ESBL-E acquisition by 28 days, particularly the relative effects of hospitalisation and antimicrobial exposure.

## Methods

The methods for recruitment and follow up of the clinical cohort along with the laboratory methods of sample processing, stool culture, bacterial identification, ESBL confirmation and antimicrobial sensitivity testing are described in Chapter 2, Methods. Further methods of the statistical analysis carried out in this chapter are detailed here. 

Summary statistics across the three arms of the study are presented as proportions for categorical data and medians and interquartile ranges for continuous data, with p-values from Fisher's exact test and the Kruskall-Wallace test, respectively, used to test for differences between the arms. The magnitude of the time-varying exposures of interest - hospitalisation and antimicrobial exposure - were expressed in three ways, to ensure that the study procedures had generated three arms with good separation of exposures:

1. The proportion of participants in each arm who were exposed to a given exposure on any given day was plotted;

2. Total person-days of exposure for each arm was calculated; the person-days at risk were not equal across the three arms of the study because of varying numbers of participants and drop-out rates, so the total person-days at risk were also calculated;

3. The number of participants who were exposed to a given exposure, along with the median length of exposure were calculated. 

ESBL-E colonisation was expressed as a simple proportion at each time point, and visualised by plotting the proportion with binomial confidence intervals. For these plots, Arm 2 and 3 participants were censored at fist antimicrobial exposure (Arm 2 and 3) and hospitalisation (Arm 3). Whilst time of measurement of ESBL-E status was ostensibly at day 0,7,28,90 or 180, in reality it was distributed around these points. To visualise these data accounting for this, ESBL-E carriage status was plotted against time  with ESBL-E colonisation coded as 1 and not colonised a 0, and a non-parametric LOESS regression line (with first order polynomial and smoothing parameter 0.75) fitted to them. This fits a local smoothed linear regression using least squares and a proportion of the of the data points, to produce a smoothed local best fit curve through the data points; in this case broadly equivalent to binning observations in a large number of time category bins and calculating a prevalence for each one, and can be interpreted as a ESBL-E rolling prevalence which accounts for the varying measurement time.

This does not, however, account for the fact that the measurements are clustered within individuals. In order to do this, a two-state model (with ESBL-E colonised and uncolonised states) was fitted using the Aalen-Johansen estimate of state occupancy in the _survival_ package in R. This is a generalisation on the Kaplan-Meier curve and allows plotting of state occupancy probability - an estimate of carriage prevalence at any time point - as a function of time, rather than just the survival function as in a Kaplan-Meier curve. However, the Aalen-Johansen estimate assumes that the time of state transition is known. In fact, the data considered here are interval censored - that is, that transition is only known to have happened within a particular time period between two study visits, and so to generate this estimate of state occupancy it is necessary to assume a transition time: I assumed it happened halfway between measurements. A Markov model can account for all these difficulties; the development and fitting of such a model is the subject of Chapter 8. ESBL-E carriage was also visualised as a heat map, with each cell representing a sample, columns representing individuals and rows the study visits.

To explore the associations of ESBL-E carriage, two logistic regression models were fit. The first aimed to explore associations of baseline ESBL-E carriage. Variables that I hypothesised _a priori_ would be related to ESBL-E carriage were included in the model: age, sex, HIV status, study arm, receipt of antiretroviral therapy (ART) or cotrimoxazole preventative therapy (CPT), hospitalisation or receipt of antimicrobial therapy, household crowding (number of adults and number of children in the household separately as linear continuous variables), presence of animals at home, presence of a flushing toilet at home, use of unprotected water sources (defined as surface water or unprotected springs or wells), whether water was treated with chlorine, and sample collection during rainy season (defined as date of sample collection between 1st November and 30th April). These variables were fit individually (univariable model) and then in a full multivariable model. Odds ratios with 95% confidence intervals and p-values are presented.

The second logistic regression model aimed to explore associations of ESBL-E acquisition by the 28 day visit. This analysis included only participants who were a) ESBL-E uncolonised at baseline and b) had an available sample at 28 days +/- 2 weeks. To explore associations of ESBL-E acquisition, the exposures of interest (again specified _a priori:_ ceftriaxone, amoxicillin, ciprofloxacin and cotrimoxazole exposure and hospitalisation) were first quantified as days of exposure between the baseline and follow up visit, and binned into three groups: no exposure, five or fewer days of exposure, or more than five days of exposure. Proportion of participants who had a detectable ESBL-E at follow up in each group were plotted, stratified by these groups, to assess both for any association and for a dose-response relationship. Multivariable analysis was carried out by dichotomising each exposure as a binary variable (exposed/non-exposed) and including them all in a logistic regression model. They were not used as a continuous linear predictor because a linear relationship between antimicrobial exposure and ESBL-E acquisition seems unlikely, and there were not enough data points to categorise the variables with more granularity.

## Results

### Study population

In total, 425 participants were recruited to the study between 19th February 2017 and 2nd October 2018; 225 participants with sepsis (Arm 1), 100 inpatients without antimicrobial exposure at baseline (Arm 2) and 100 community members (Arm 3). Flow of participants through the study is shown in Figure \@ref(fig:ESBL-consort-diag). It was often challenging to collect stool samples from participants but 87% (1416/1631) eligible patient-visits resulted in the collection of a stool sample. Drop out from the study and failure to collect stool samples were similar in arm 1 and 2 and with no apparent systematic bias, but both drop out and missing samples were less frequent in arm 3 (Figure \@ref(fig:ESBL-prop-samples)A). There was significant variation in the timing of stool sample collection, with a broad distribution around the ostensible collation day (Figure \@ref(fig:ESBL-prop-samples)B).

The baseline characteristics of the enrolled participants are shown in Table \@ref(tab:dassim-2-demog). There were some important differences between the arms of the study: despite matching on age and sex, antimicrobial-unexposed participants were older. They were also less likely to be HIV-infected than participants with sepsis (13% [12/89] of those with known HIV status were HIV-infected versus 67% [143/213] with sepsis), and less likely to have been treated for TB. Sepsis participants were more likely to have received antimicrobials or been hospitalised in the previous 4 weeks. In the community arm of the study, there were a high proportion of participants (60% [60/100]) with an unknown HIV status, and there were some differences in toilet facilities, water sources, cooking fuel and presence of animals at home across the three groups. 


```{r ESBL-consort-diag, echo = F, fig.scap="ESBL carriage study recruitment and follow up.", out.extra='', fig.cap = "Study recruitment and follow up. At each time point \\textit{eligible participants} refers to participants who are known to be alive and have not withdrawn from the study by that time point, and \\textit{samples collected} refers to patients from whom a stool sample was sucessfully collected for that visit.",  out.height = '100%', fig.align = 'center'}

knitr::include_graphics(paste0(wd, "/figures/ESBLstudy_flow_diagram.pdf"))

```

```{r ESBL-prop-samples, echo = F, warning = F, message = F,fig.scap="Missing samples and variation in sample collection time", out.extra='', fig.cap = "A: Missing stool samples stratified by arm and visit. Bar height at a given visit represents the number of eligible participants, coloured by successful sample collection (blue) or failure to collect a sample (red). B: Distribution of actual day of sample collection for ostensible day 7, 28, 90 and 180 samples showing considerable variation.",  fig.height = 8, fig.width=8, ,fig.align = 'center'}

sampls <- read.csv("chapter_5/ESBL_carriage_recruitment.csv", stringsAsFactors = F)

sampls$a_missed <- sampls$eligible - sampls$collected

sampls %>% select(-eligible) %>% pivot_longer(-c(arm, visit)) %>% ggplot(aes(as.factor(visit), value, fill = name)) + geom_col() + facet_wrap(~arm, scales = 'free') + xlab("Visit day") + ylab("n") + theme_bw() + theme(legend.title = element_blank(), legend.position = "top") + scale_fill_manual(labels = c("No sample", "Sample collected"), values = hue_pal()(2)) -> p1


visit_names <- c(
  `1` = "Day 7",
  `2` = "Day 28",
  `3` = "Day 90",
  `4` = "Day 180"
)

followup %>% filter(arm != 4, foutcome == 1) %>% ggplot(aes(as.numeric(t), group = as.factor(d2visit))) + geom_histogram(bins = 60) + facet_wrap(~ d2visit, scale = 'free_y', ncol = 1, strip.position = "right", labeller = as_labeller(visit_names)) + xlim(c(0,300)) + theme_bw() + xlab("Time post enrolment (days)") + ylab("n") -> p2

ggarrange(p1,p2, ncol = 1, nrow = 2)

```


\renewcommand{\arraystretch}{0.8}

```{r dassim-2-demog, echo = F, warning=F, message = F}

enroll <- filter(enroll, arm != 4)
enroll$art.time <- as.numeric((enroll$enroll_date - enroll$hivartstart) /(365.25/12))
enroll$hivart[enroll$hivart != "5A" & !is.na(enroll$hivart)] <- "ART regimen: other"
enroll$animalskept[enroll$animalskept == "Other"] <- "Elsewhere"
enroll$keep.other[enroll$keep.cattle == "Yes" | enroll$keep.sheep == "Yes" | enroll$keep.mules == "Yes"] <- "Yes"
enroll.sum <- dplyr::select(enroll, arm,calc_age, ptsex, hivstatus,recieved_prehosp_ab,pmhxrechospital,tbstatus, tbongoing, hivonart, hivart, art.time, hivcpt, 
                  tobaccoyn, alcoholyn,  highestedu,
                 job, housholdadultsno, householdchildno, toilet , watersource, watertreated,
                 electricityyn, fuel, 
                 keepanim, keep.poultry,  keep.goats,
                 keep.dogs, keep.cattle,  keep.other)

enroll.sum$arm <- as.character(enroll.sum$arm)

enroll.sum %>% select_if(is.character) %>% 
  pivot_longer(-arm) %>% 
  group_by(name, arm, value) %>% 
  tally() %>% 
  pivot_wider(names_from = arm, values_from = n, , values_fill = list(n = 0) ) %>% filter(!is.na(value)) %>% group_by(name) %>% 
  mutate(sum1 = sum(`1`), 
         sum2 = sum(`2`), 
         sum3 = sum(`3`),
         str1 = glue('{`1`}/{sum1} ({specify_decimal(`1`*100/sum1, 0)}%)'),
         str2 = glue('{`2`}/{sum2} ({specify_decimal(`2`*100/sum2, 0)}%)'),
         str3 = glue('{`3`}/{sum3} ({specify_decimal(`3`*100/sum3, 0)}%)')
  ) -> t1

apply(t1[3:8], 1, function(x) fisher.test(matrix(x, ncol = 2))$p.value) -> t1$p
t1$tot <- t1$`1` + t1$`2` + t1$`3`
t1$n <- t1$sum1 + t1$sum2 + t1$sum3
t1$Total <- t1 %>% glue_data('{tot}/{n} ({specify_decimal(tot*100/n, 0)}%)')
t1$totalprop <- t1$tot /t1$n
t1 <- filter(t1, value != "No")

enroll.sum$arm <- as.numeric(enroll.sum$arm)

enroll.sum %>% select_if(is.numeric) %>% 
  pivot_longer(-arm) %>% 
  group_by(name) %>% mutate(p = kruskal.test(value ~ arm)$p.value,
                            median.t = median(value, na.rm = T),
                                    lqr.t = quantile(value, 0.25, na.rm = T),
                                    uqr.t = quantile(value, 0.75, na.rm = T),
                                    Total = glue('{specify_decimal(median.t,1)} ({specify_decimal(lqr.t,1)}-{specify_decimal(uqr.t,1)})')) %>%
  group_by(name, arm) %>% summarise(median = median(value, na.rm = T),
                                    lqr = quantile(value, 0.25, na.rm = T),
                                    uqr = quantile(value, 0.75, na.rm = T),
                                    strz = glue('{specify_decimal(median,1)} ({specify_decimal(lqr,1)}-{specify_decimal(uqr,1)})'),
                                    p = unique(p),
                                    Total = unique(Total)) %>% select(name, arm, strz, p, Total) -> t1.1 

dcast(t1.1, name+ p + Total ~ arm, value.var = "strz") -> t1.1
names(t1.1)[4:6] <- c("str1", "str2", "str3")
  bind_rows(dplyr::select(t1, name, value,str1, str2, str3, p, Total, totalprop), t1.1) -> t1
    rm(t1.1)  
    
    t1$value[grepl("keep.", t1$name) & t1$name != "keepanim"] <- t1$name[grepl("keep.", t1$name)& t1$name != "keepanim"]
     t1$name[grepl("keep.", t1$name)& t1$name != "keepanim"] <- "which.anim"
    ord <- c("calc_age", "ptsex", 
             "hivstatus", "tbstatus", "tbongoing",
             "hivonart", "art.time", "hivart", "hivcpt",
             "recieved_prehosp_ab", "pmhxrechospital",
             "tobaccoyn","alcoholyn", 
             "highestedu", "job", 
             "toilet",
             "watersource",
             "watertreated",
             "householdchildno",
             "housholdadultsno",
             "electricityyn",
             "fuel",
             "keepanim",
             "which.anim"
             )
        
       t1$name <- factor(t1$name, levels = ord)  
       t1[order(t1$name, -t1$totalprop),] -> t1
       
# tweaky tweaky
       
       
    t1$name <- as.character(t1$name)


t1$value[t1$name == "householdchildno"] <- "Children"
t1$value[t1$name == "housholdadultsno"] <- "Adults"
t1$name[t1$name == "householdchildno"] <- "No. household members"
t1$name[t1$name == "housholdadultsno"] <- "No. household members"
t1$value[t1$name == "calc_age"] <- "Age (yr)"
t1$value[t1$name == "art.time"] <- "Months on ART"

t1 <- subset(t1, value != "Female")
t1$value[t1$name == "ptsex"] <- "Male sex"
t1$name[t1$name %in% c("calc_age", "ptsex")] <- "Demographics"

# art variables
t1$value[t1$value %in% c("Reactive", "Non reactive", "Unknown")] <- 
  paste0("HIV ", str_to_title(t1$value[t1$value %in% c("Reactive", "Non reactive", "Unknown")]))

t1$value[t1$name == "hivonart"] <- "Current ART*"

t1$value[t1$name == "hivcpt"] <- "Current CPT\\textsuperscript{\\dag}"
t1$value[t1$value == "5A"] <- "ART regimen: EFV/3TC/TDF"
t1 <- subset(t1, value != "ART regimen: other")

#t1$levels[t1$variable == "art.time"] <- "Months on ART"
t1$name[t1$name == "hivstatus"] <- "HIV/TB status"
t1$name[grepl("art", t1$name)] <- "ART status*"

t1$name[grepl("cpt", t1$name)] <- "ART status"
# tb variables
t1$value[t1$name == "tbstatus"] <- "Ever treated for TB"
t1$value[t1$name == "tbongoing"] <- "Of those, current TB treatment"
t1$name[grepl("tb", t1$name)] <- "HIV/TB status"

t1$value[t1$name == "recieved_prehosp_ab"] <- "Antibiotics"
t1$value[t1$name == "pmhxrechospital"] <- "Hospitalised"

t1$name[t1$name == "recieved_prehosp_ab"] <- "Healthcare exposure last 4wk"
t1$name[t1$name == "pmhxrechospital"] <- "Healthcare exposure last 4wk"




# tobacco use
t1$value[grepl("tobacco", t1$name)] <- paste0(t1$value[grepl("tobacco", t1$name)], " tobacco" )
t1$name[grepl("tobacco", t1$name)] <- "Tobacco/alcohol use"

# alcohol use
t1$value[t1$name == "alcoholyn"] <- "Current alcohol"
t1$name[grepl("alcohol", t1$name)] <- "Tobacco/alcohol use"

# education
t1$name[grepl("highestedu", t1$name)] <- "Education"

# employment 
t1$name[grepl("job", t1$name)] <- "Employment"

# toilet

t1$name[grepl("toilet", t1$name)] <- "Toilet facilities"

# toilet
t1$value[t1$name == "watertreated"] <- "Treat water with chlorine"

t1$name[grepl("water", t1$name)] <- "Main water source"

# electricity
t1$value[t1$name == "electricityyn"] <- "Electricity available in house"
t1$name[grepl("electric", t1$name)] <- "Electricty"

t1$name[grepl("fuel", t1$name)] <- "Main cooking fuel"

# animals

#t1$levels[grepl("keep", t1$variable)] <- t1$variable[grepl("keep", t1$variable)]

sub("keep.", "",t1$value) -> t1$value
t1$name[t1$name == "keepanim" | t1$name == "which.anim"] <- "Animals at home?"
t1$value[t1$name == "Animals at home?"] <- str_to_title(t1$value[t1$name == "Animals at home?"])
t1$value[t1$name == "Animals at home?" & t1$value == "Yes"] <- "Any animal"

str_to_title(t1$levels[grepl("keep", t1$variable)] ) -> t1$levels[grepl("keep", t1$variable)]
#t1$levels[t1$levels == "Nim"] <- "Any animal"
names(t1)[1] <- "variable"
kstring <- make_kable_rowgroup_string(t1)


sub("%", "\\\\%", t1$str1) -> t1$str1
sub("%", "\\\\%", t1$str2) -> t1$str2
sub("%", "\\\\%", t1$str3) -> t1$str3
sub("%", "\\\\%", t1$Total) -> t1$Total
boldrows <- t1$p < 0.05

t1$p <- specify_decimal(t1$p, 3)
t1$p[t1$p == "0.000"] <- "<0.001"

t1 <- ungroup(t1)
boldrows <- which(boldrows %in% TRUE)

kable(select(t1,value, str1, str2, str3, p,Total), "latex", booktabs =T, 
      row.names = F, longtable = T, escape = F, col.names = c("Variable","Sepsis", "Inpatient", "Community", "p", "Total"), 
      caption = "Participant Characteristics" ) %>%  kable_styling(font_size = 9, full_width = F,latex_options = c("scale_down")) %>%
  pack_rows(index = kstring) %>%
  row_spec(boldrows, bold = T) %>%
  footnote(general = "ART = Antiretroviral therapy, CPT = Cotrimoxazole preventative therapy, EFV: Efavirenz, 3TC: Lamivudine, TDF: Tenofovir. Numeric values are median (IQR)) unless otherwise stated. P-values are to assess for differences across the three groups: Fisher's exact test across the groups for categorical variable, and Kruskal-Wallace test for continuous variables.", symbol = c("ART status includes HIV reactive only as denominator", "Missing CPT data for two participants. ") ,threeparttable = T, fixed_small_size = T) %>% landscape()



                                               
```

### Exposures during the study period

Exposures to antimicrobials and hospitalisation of the cohort are shown in Figure \@ref(fig:ESBL-study-exp) and Table \@ref(tab:exposure-tab). Antimicrobial-unexposed inpatients (Arm 2 participants) had a shorter length of hospital stay than participants with sepsis (Arm 1 participants): median (IQR) 2 (2-7) versus 5 (2-10) days, p = 0.002 by Kruskal-Wallace test. Five of the 100 Arm 2 participants were taking CPT at baseline, 18 received further courses of antimicrobials during the study period, and two were started on TB therapy. Some participants received combinations of these therapies, so in total 23% (23/100) Arm 2 participants received an antibacterial during the study period, mostly within 30 days following enrolment (Figure \@ref(fig:ESBL-study-exp))), and most commonly ceftriaxone (Table \@ref(tab:exposure-tab)). All of these participants were cesored on first exposure.

Both antimicrobial exposure and hospitalisation were unusual in the community cohort; 7% (7/100) community (Arm 3) participants were taking CPT and one received a 5-day course of amoxicillin meaning that 8% (8/100) Arm 3 participants received an antibacterial during the study period. In addition one Arm 3 participant was hospitalised for 1 day in the study period. No Arm 3 participant received any TB therapy, and no Arm 2 or 3 participants received any antimalarial or antifungal therapy during the study period.

Because of the chronic nature of the therapy, the greatest antimicrobial exposure (in terms of participant-days) in all arms were to cotrimoxazole and TB therapy, by an order of magnitude (Table \@ref(tab:exposure-tab)). Apart from these, the most commonly received antibacterial by Arm 1 participants  was ceftriaxone by some distance with 998 participant-days of exposure in 189 participants during the study period, and a median 5 (IQR 3-7) day course. Ciprofloxacin and amoxicillin were also commonly received, with 61 participants receiving 398 participant-days of exposure to ciprofloxacin with a median 7 (IQR 5-7) day course, and 39 participants receiving 235 participant-days of exposure to amoxicillin with a median 5 (IQR 5-7) day course.  

```{r ESBL-study-exp, echo = F, warning = F, message = F,fig.scap="Hospital and antibacterial exposure of participants", out.extra='', fig.cap = "Hospital and antibacterial exposure of participants expressed as (A) proportion of Arm 1 and Arm 2 participant who are hospitalised and/or exposed to the most commonly received antibacterials on any given day and (B) cumulative proportion of participants who have been exposed to any antibacterial over the study period.",  fig.height = 6, fig.width=8, fig.align = 'center'}

read.csv("data/longit_covariate_data.csv") -> df.covs

df.covs <- merge(df.covs, select(enroll, pid, arm))


df.covs %>% group_by(pid) %>% do(uncompress_covariates(.)) -> outlong
df.covs.cpt <- merge(df.covs, select(enroll, pid, hivcpt), all.x = T)

df.covs.cpt$hivcpt <- as.numeric(df.covs.cpt$hivcpt == "Yes")
df.covs.cpt$cotri[df.covs.cpt$hivcpt == 1] <- 1
df.covs.cpt <- select(df.covs.cpt, -hivcpt)

df.covs.cpt %>% group_by(pid) %>% do(uncompress_covariates(.)) -> outlong.cpt

abs.lookup <- c(
  "Amoxicillin",
  "Gentamicin",
  "Azithromycin",
  "TB therapy",
  "Penicillin",
  "Fluconazole",
  "Ceftriaxone",
  "Amphotericin",
  "Quinine",
  "Chloramphenicol",
  "LA",
  "Ciprofloxacin",
  "Co-amoxiclav",
  "Artesunate",
  "Cotrimoxazole",
  "Clindamycin",
  "Doxycycline",
  "Erythromycin",
  "Aciclovir",
  "Flucloxacillin",
  "Metronidazole",
  "Streptomycin"
)

names(abs.lookup) <- outlong %>% ungroup() %>% select(-c(pid, assess_type, died, hosp, arm)) %>% names 

abs.lookup <- c(abs.lookup, "Hospitalised", "Other AB")
names(abs.lookup) <- c(names(abs.lookup)[1:(length(abs.lookup) - 2)], "hosp", "other")

#other.antibac <- c("genta", "azithro","benzy", "coamo","clinda", "doxy","erythro", "fluclox", "metro", "strepto")

#apply(outlong.cpt[other.antibac],1,sum) -> outlong.cpt$other


arm_names <- c(
  `1` = "Arm 1: Sepsis",
  `2` = "Arm 2: Inpatient",
  `3` = "Arm 3: Community"
)


outlong.cpt %>%
  ungroup() %>% 
  select(-c(pid, died)) %>% filter(arm != 3) %>%
  pivot_longer(-c(assess_type, arm)) %>%
  group_by(arm, assess_type, name)%>%
  summarise(n = n(), x = sum(value == 1), prop = x/n)  %>% 
  filter(name %in% c("cefo","amoxy","hosp", "cipro", "cotri", "tb")) %>%
  mutate(name = recode(name, !!!abs.lookup)) %>%
  ggplot(aes(assess_type,prop, group = name, color = name, linetype = name)) +
  geom_line(alpha = 0.8, size = 0.75) + 
  facet_wrap(~ arm, ncol = 1, , labeller = as_labeller(arm_names)) + 
  theme_bw() + 
  scale_x_continuous(breaks = c(0,5,10,15,20,25,30), limits = c(0,30)) +
  scale_linetype_manual(values = c("Amoxicillin" =  "solid",
                                   "Ceftriaxone" = "solid",
                                   "Ciprofloxacin" = "solid",
                                   "Cotrimoxazole" = "solid",
                                   "Hospitalised" = "dashed",
                                   "TB therapy" = "dotted"),
                        breaks = c("Hospitalised",
                                   "TB therapy" ,
                                   "Ceftriaxone",
                                   "Cotrimoxazole", 
                                   "Ciprofloxacin", 
                                   "Amoxicillin")) + 
  scale_colour_manual(values = c("Amoxicillin" =  pal_lancet()(4)[1],
                                 "Ceftriaxone" = pal_lancet()(4)[2],
                                 "Ciprofloxacin" = pal_lancet()(4)[3],
                                 "Cotrimoxazole" = pal_lancet()(4)[4],
                                 "Hospitalised" = "black",
                                 "TB therapy" = "grey20"),
                      breaks = c("Hospitalised",
                                 "TB therapy" ,
                                 "Ceftriaxone",
                                 "Cotrimoxazole", 
                                 "Ciprofloxacin", 
                                 "Amoxicillin")) +
  theme(legend.title = element_blank()) + 
  xlab("Day post enrolment") + 
  ylab("Proportion") -> p1

outlong %>% ungroup() %>% select(-c(pid, assess_type, died, hosp,tb, cotri)) %>% 
  group_by(arm) %>% 
  summarise_all(sum) %>%
  pivot_longer(-arm) %>% group_by(name) %>% mutate(tots = sum(value)) %>% ungroup() %>%
  mutate(name = recode(name, !!!abs.lookup)) %>%
  ggplot(aes((fct_reorder(name, tots)), value)) + 
  geom_col() + 
  facet_wrap(~arm, scales = "free_x", labeller = as_labeller(arm_names)) +
  coord_flip() +
  theme_bw()  + 
  theme(panel.spacing = unit(1,"lines")) +
  ylab("Participant-days of exposure") +
  xlab(element_blank()) -> p2 

# cumulative fraction with no ab and no hosp
# start with hosp

abs.lookup[!names(abs.lookup) %in% c("tb","fluco", "ampho", "quin", "coart", "arte", "acicl", "flu", "hosp", "other")] -> anti_bacterials


apply(df.covs.cpt[c(names(anti_bacterials), "tb")], 1, sum) -> df.covs.cpt$any_antibacterial

df.covs.cpt$any_antibacterial[df.covs.cpt$any_antibacterial > 1] <- 1

time.to.ab.cot <- df.covs.cpt %>% group_by(pid) %>% do(ditch_everything_after_first_1(.,28))

time.to.ab.cot$assess_type[time.to.ab.cot$any_antibacterial == 0] <- 1000
time.to.ab.cot$any_antibacterial[time.to.ab.cot$any_antibacterial == 0] <- 1
#time.to.ab.cot <- filter(time.to.ab.cot, any_antibacterial > 0, arm == 3)

time.to.ab.cot %>%
  filter(any_antibacterial == 1) %>% group_by(arm) %>% mutate(n.arm = n()) %>%
  group_by(arm, assess_type) %>%
  summarise(n.tot = unique(n.arm),
            n = length(assess_type)) %>% 
  group_by(arm) %>%
  mutate(cumsm = cumsum(n),
         cumsm.prop = cumsm/n.tot) %>% 
  ungroup() %>%
  mutate(arm = recode(arm, `1` = "Arm 1", `2` = "Arm 2", `3`= "Arm 3")) %>%
  ggplot(aes(assess_type, cumsm.prop, group = arm, color = arm)) +
  geom_step(size = 0.75)  +  theme_bw() +
  ylab("Proportion") +
  xlab("Days post enrolment") +
  theme(legend.title = element_blank()) + coord_cartesian( ylim = c(0,1),xlim  =c(0,180)) +
  scale_x_continuous(limits = NA, breaks = c(0,30,60,90,120, 150,180)) +
  scale_color_manual(values = pal_lancet()(3)[c(2,1,3)]) +
  scale_linetype_manual(values = c(1,2,3))-> p3

time.to.ab.cot <- df.covs.cpt %>% group_by(pid) %>% do(ditch_everything_after_first_1(.,28))


ggarrange(p1,
ggarrange(p3,NULL, widths = c(1,0.06)),
ncol = 1, nrow = 2, labels = c("A", "B"), heights  = c(2,1))


```

```{r exposure-tab, echo = F, warning = F, message = F}
outlong.cpt %>% ungroup() %>% 
  select(-c(assess_type, died)) %>% 
  pivot_longer(-c(pid, arm)) %>% 
  filter(value > 0) %>%
  group_by(pid, name,arm) %>% 
  summarise(pid.exp = sum(value)) %>% group_by(arm, name) %>%
  summarise(n.p = length(unique(pid)),
            exposure = sum(pid.exp),
            median.exp.len = median(pid.exp),
            lq.exp.len = quantile(pid.exp, 0.25),
            uq.exp.len = quantile(pid.exp, 0.75)) -> exposures
  
outlong.cpt %>% group_by(arm) %>%
  summarise(n.tot = length(unique(pid)),
            tot.exp = length(pid)) -> tots.exposures

exposures <- merge(tots.exposures, exposures)


exposures$exp.perc <- specify_decimal(exposures$exposure*100/exposures$tot.exp,0)
exposures$exp.perc[exposures$exp.perc == "0"] <- "<1"

exposures$exp.str <- as.character(exposures %>% glue_data('{exposure}')) #({exp.perc}%)'))
exposures$tr.len.str <- as.character(exposures %>% glue_data('{specify_decimal(median.exp.len, 0)} ({specify_decimal(lq.exp.len,0)}-{specify_decimal(uq.exp.len,0)})'))

exposures %>% select(name, arm,n.p, exp.str, tr.len.str) %>%
  pivot_wider(names_from = arm, 
              values_from = c(n.p, exp.str, tr.len.str),
              values_fill= list(n.p = 0, exp.str = "0", tr.len.str = "-")) -> exp.tab

#subset(exposures, arm == 1 & name != "hosp")
exp.tab$name <- factor(exp.tab$name, levels = c("hosp",
                                                subset(exposures, arm == 1 & name != "hosp")$name[order(subset(exposures, arm == 1 & name != "hosp")$exposure, decreasing = T)]))

exp.tab %>% mutate(name = recode(name, !!!abs.lookup)) -> exp.tab

exp.tab[order(exp.tab$name),] -> exp.tab
tots.exposures %>% pivot_wider(names_from = arm, values_from = c(n.tot, tot.exp)) -> tots.exposures
tots.exposures$name <- "Total At Risk"
names(tots.exposures)[1:6] <- names(exp.tab)[2:7]
tots.exposures[4:6] <- sapply(tots.exposures[4:6], as.character)
tots.exposures <- select(tots.exposures, names(exp.tab)[1:7] )
bind_rows(tots.exposures, exp.tab) -> exp.tab
exp.tab[is.na(exp.tab)] <- "-"

#exp.tab <- select(exp.tab, 
#                  n.p_1, exp.str_1, tr.len.str_1,
#                  n.p_2, exp.str_2, tr.len.str_2,
 #                 n.p_3, exp.str_3, tr.len.str_3)


kable(exp.tab, "latex", booktabs = TRUE,
      row.names = FALSE, escape = TRUE, col.names = c("Exposure", "Arm 1", "Arm 2", "Arm 3", "Arm 1", "Arm 2", "Arm 3", "Arm 1", "Arm 2", "Arm 3"),
      caption = "Antimicrobial and hospital exposure stratified by arm") %>% kable_styling(full_width = FALSE, font_size = 9) %>%
#  pack_rows("Total", 1,1) %>%
#  pack_rows("Hospitalisation", 2,2) %>%
 pack_rows("Exposures", 2,23) %>%
  add_header_above(c(" " = 1, "Number exposed" = 3, "Exposure (person-days)" = 3,
                     "Median (IQR) exposure length (days)" = 3)) %>% 
  footnote(general = "TB = tuberculosis, LA =lumefantrine artemether. Median exposure length includes only those exposed. Total at risk shows the total number of participants and participant-days of follow up included in the study." ,threeparttable = T, fixed_small_size = T) %>% landscape()

```

\clearpage

### ESBL-E colonisation

ESBL-E colonisation prevalence as a function of time across the three arms of the study is shown in Table \@ref(tab:ESBL-carriage-tab) and Figure \@ref(fig:ESBL-carriage-plots). Baseline colonisation prevalence was high in all groups, and higher in Arm 1 and 2 participants than community members: 49% (95% CI 42-56%) in Arm 1 participants, 41% (95% CI 32-52%) in Arm 2 and 28% (95% 20-38%) in Arm 3. Both hospitalised groups showed a rise in colonisation prevalence following admission, though this is much more marked in Arm 1 participants: by the day 7 visit 78% (95% CI 71-84%) of Arm 1 participants were colonised compared to 51% (38-64%) of Arm 2 participants. This difference persisted through to day 28, when the crude prevalence in Arm 1 was 72% (95% CI 64-79%) versus 52% (95% CI 40-64%) in Arm 2. By the end of the study period the prevalence had fallen back to baseline levels in both groups. Within an individual, there was often frequent flipping between the ESBL colonised to uncolonised state and back again at different study visits and often on short time-scales (Figure \@ref(fig:ESBL-heatmap)).

In total, 723/1417 (51%) of samples grew at least one ESBL-E; 1032 organisms were grown from the 723 samples, with a median 1 (IQR [1-2]) ESBL-E per sample. The most commonly isolated organism as identified by the API system was _E. coli_ (n = 686) followed by _Klebsiella pneumoniae_ (n = 245, Figure \@ref(fig:ESBL-spec-sens)). Antimicrobial sensitivity testing was carried out on the first 694/1032 (67%) organisms. Cotrimoxazole resistance was near universal (675/694 [97%] of isolates), with intermediate proportions of gentamicin (367/694 [53%]) and ciprofloxacin (457/694 [65%]) resistance and less chloramphenicol resistance (232/694 [33%] of isolates). Meropenem and amikacin resistance was very unusual (14/694 [2%] and 15/694 [2%] of isolates respectively) but the fact that resistance to these antibacterials of last resort was seen at all is troubling.

```{r ESBL-carriage-tab, echo =F, message = F, warning = T}

merge(
lims %>% group_by(visit, arm) %>% 
  summarise(n = n(),
            e = sum(ESBL == "Positive"),
            prop = e/n,
            ESBLpropstr = as.character(glue('{e} ({specify_decimal(prop*100, 0)}%)')),
            nstr = as.character(n())) %>%
  select(visit, nstr, arm, ESBLpropstr) %>%
  pivot_wider(names_from = arm,values_from = c(ESBLpropstr,nstr), values_fill =list(ESBLpropstr = "-", nstr = "-")),

lims_orgs %>% group_by(visit, arm) %>% 
  mutate(n.smpl = length(unique(lab_id)),
         organism = recode(organism, `Escherichia coli` = "esco", `Klebsiella pneumoniae` = "klpn", .default = "other")) %>%
  filter(!is.na(organism)) %>% group_by(visit, arm, organism) %>% 
  summarise(n.sampl = unique(n.smpl),
            n = length(organism),
            prop = n/n.sampl,
            ORGpropstr = as.character(glue('{n} ({specify_decimal(prop*100, 0)}%)'))) %>%
  select(visit, arm, organism, ORGpropstr) %>%
  pivot_wider(names_from = c(arm,organism), values_from = ORGpropstr, values_fill = list( ORGpropstr = "-"))
) -> ESBLtab

ESBLtab <- select(ESBLtab, visit, 
                  nstr_1, ESBLpropstr_1,#`1_esco`,`1_klpn`, `1_other`,
                  nstr_2, ESBLpropstr_2,#`2_esco`,`2_klpn`, `2_other`,
                  nstr_3, ESBLpropstr_3)#`3_esco`,`3_klpn`, `3_other`)

ESBLtab$visit <- c("Day 0", "Day 7", "Day 28","Day 90", "Day 180")

kable(ESBLtab, "latex", booktabs = TRUE,
      row.names = FALSE, escape = TRUE, col.names = c("Visit",rep(c("n", "Any ESBL"),3)),
      caption = "ESBL carriage stratified by arm and visit") %>% kable_styling(full_width = FALSE, font_size = 9) %>%
#  pack_rows("Total", 1,1) %>%
#  pack_rows("Hospitalisation", 2,2) %>%
# pack_rows("Exposures", 2,23) %>%
  add_header_above(c(" " = 1, "Arm 1 (Sepsis)" = 2, "Arm 2 (Inpatient)" = 2,
                     "Arm 3 (Community)" = 2))# %>% 
 # footnote(general = "TB = tuberculosis, LA =lumefantrine artemether. Median exposure length includes only those exposed. Total at risk shows the total number of participants and participant-days of follow up included in the study." ,threeparttable = T, fixed_small_size = T) %>% landscape()

```

```{r ESBL-carriage-plots, echo = F, warning = F, message = F,fig.scap="ESBL colonisation prevalence as a function of time", out.extra='', fig.cap = "ESBL carriage prevalence as a function of time visualised in three different ways. In each case participants from Arm 2 are censored on first antimicrobial exposure and Arm 3 are censored on first antimicrobial exposure or hospitalisation. Top (A) prevalence at each visit plotted at ostensible visit time; however, the visits are in fact distributed in time themselves so the middle plot (B) is an attempt to show this by fitting a nonparametric smoothed LOESS regression line with a local linear regression. However the confidence intervals in this method are too narrow because they assume independence of the measurements, which are in fact clustered within patients. The bottom panel (C) is an estimate of the proportion of ESBL-colonised participants from the Aalen-Johansen estimate, which is a generalisation of the Kaplan-Meier curve. This takes into account the nonindependence of the measurements, but does not take into account the interval-censored nature of the data, and transitions to and from the ESBL colonised state are hence assumed to happen halfway between measurements.",  fig.height = 8, fig.width=6, out.height = '70%', fig.align = 'center'}


p.lci <- function(x,n) {
  return(binom.test(x,n)$conf.int[[1]])
}

p.uci <- function(x,n) {
  return(binom.test(x,n)$conf.int[[2]])
}



#ggplot(lims_dates, aes(assess_type, as.numeric(ESBL == "Positive"), group = arm)) + geom_point() + geom_smoo#th() + xlim(c(0, 200))

# first - plot at visit date

censor_list <- subset(time.to.ab.cot, any_antibacterial == 1 & arm != 1)

lims_dates_censored <- lims_dates

for (i in 1:nrow(censor_list)) {
  lims_dates_censored <- subset(lims_dates_censored, !(pid == censor_list$pid[i] & assess_type > censor_list$assess_type[i]))
}

lims_dates_censored %>% 
  group_by(arm, visit) %>%
  summarise(e = sum(ESBL == "Positive"),
            n = length(ESBL),
            prop = e/n,
            lci = p.lci(e,n),
            uci = p.uci(e,n) 
            ) %>% ungroup() %>% mutate(arm = as.factor(arm)) %>% 
  mutate(visit = recode(visit, `0` = 0, `1` = 7, `2` = 28, `3` = 90, `4` = 180),
         arm = recode(arm, `1` = "Arm 1", `2` = "Arm 2", `3` = "Arm 3")) %>%
  ggplot(aes(visit, prop, ymin = lci, ymax = uci, group = arm, fill = arm, color = arm)) +
  geom_line() + geom_ribbon(color = NA, alpha = 0.3) + theme_bw() +
  scale_color_manual(values = pal_lancet()(3)[c(2,1,3)]) + 
  scale_fill_manual(values = pal_lancet()(3)[c(2,1,3)]) +
  xlab("Day post enrolment") +
  ylab("Proportion ESBL") + coord_cartesian(ylim = c(0.2,0.9)) +
  theme(legend.title = element_blank()) -> p1.c

lims_dates_censored %>%
  mutate(arm = recode(arm, `1` = "Arm 1", `2` = "Arm 2", `3` = "Arm 3")) %>%
ggplot(aes(assess_type, as.numeric(ESBL == "Positive"),
           group = arm, 
           fill = as.factor(arm), 
           color = as.factor(arm ))) + 
  geom_smooth(size = 0.5) +
  coord_cartesian(xlim = c(0,180), ylim = c(0.2,0.9)) +
  theme_bw() +
  scale_color_manual(values = pal_lancet()(3)[c(2,1,3)]) + 
  scale_fill_manual(values = pal_lancet()(3)[c(2,1,3)]) +
  xlab("Day post enrolment") +
  ylab("Proportion ESBL") +
  theme(legend.title = element_blank()) -> p2.c



# both underestimate errors because assume independence of measurements
# fit a multistate model



#lims_dates %>% group_by(pid, visit, .drop = FALSE) %>% tally() %>% filter(n == 0)

#lims_dates_censored %>% unique %>% group_by(pid, visit) %>% tally() %>% filter(n > 1)

lims_dates_censored %>% filter(!pid %in% c("DAS1441S", 
                                  "DAS1497X",
                                  "DAS1519G")) %>%
  group_by(pid) %>% do(pointESBLmeas_tostartstop(.)) -> l.c

fit1.c <- survfit(Surv(time = tstart, time2 = tstop, event = ESBL, type = "mstate") ~ arm, data = l.c, id = pid, istate = init.state)

pull_state_occ_from_surv_obj(fit1.c) -> pstate.c

ggplot(pstate.c, aes(x = t, y = `p(ESBL.pos)`, colour = arm)) + 
  geom_line() + 
  geom_ribbon(aes(ymin = `ll.ESBL.pos`, ymax = `ul.ESBL.pos`, fill = arm), colour = NA, alpha = 0.3) + 
  theme_bw() + 
  coord_cartesian(ylim = c(0.2,0.9), xlim = c(0,180))+
  labs(x = "Days post recruitment", y = "Proportion ESBL") -> p3.c

l.trans.halfway.c <- l.c %>% group_by(pid) %>% do(change_transition_times(., 0.5))
fit2.c <- survfit(Surv(time = tstart, time2 = tstop, event = ESBL, type = "mstate") ~ arm, data = l.trans.halfway.c, id = pid, istate = init.state)
pull_state_occ_from_surv_obj(fit2.c) -> pstate2.c


pstate2.c %>%
  mutate(arm = recode(arm, `1` = "Arm 1", `2` = "Arm 2", `3` = "Arm 3")) %>%
ggplot(aes(x = t, y = `p(ESBL.pos)`, colour = arm)) + 
  geom_line() + 
  geom_ribbon(aes(ymin = `ll.ESBL.pos`, ymax = `ul.ESBL.pos`, fill = arm), colour = NA, alpha = 0.3) + 
  theme_bw() + 
  coord_cartesian(ylim = c(0.2,0.9), xlim = c(0,180)) +
  scale_color_manual(values = pal_lancet()(3)[c(2,1,3)]) + 
  scale_fill_manual(values = pal_lancet()(3)[c(2,1,3)]) +
  labs(x = "Days post recruitment", y = "Proportion ESBL") -> p4.c

ggarrange(p1.c,p2.c,p4.c, ncol = 1, nrow = 3, common.legend = TRUE, labels = c("A", "B", "C"))       

```


```{r ESBL-heatmap, echo = F, warning = F, message = F,fig.scap="Heatmap showing distribution of ESBL positive samples", out.extra='', fig.cap = "Heatmap showing distribution of stool samples with detectable ESBL-E. Each column represents a patient, and each cell a stool sample with valid result, coloured by presence or absence of detectable ESBL-E, to demonstrate the complex patterns of apparent acquisition and loss in many individuals over the study period. ",  fig.height = 6, fig.width=10, fig.align = 'center'}
lims %>% 
    group_by(pid) %>% 
    mutate(n.pos = sum(ESBL == "Positive"), n.neg = sum(ESBL == "Negative"), n = n()) %>%
    arrange(desc(n.pos/n.neg),n.pos,desc(n.neg)) %>% ungroup() %>%
    mutate(pid = factor(pid, levels = unique(pid))) %>%
    mutate(visit = recode_factor(visit, `0` = "Day 0", `1` = "Day 7", `2` = "Day 28", `3` = "Day 90", `4` = "Day 180", .ordered = TRUE),
           arm = recode(arm, `1` = "Arm 1", `2` = "Arm 2", `3` = "Arm 3")) %>%
    ggplot(aes(pid, visit, fill = ESBL)) + 
    geom_tile()  + 
    facet_wrap(~ arm, scales = "free", ncol = 1) + 
    theme_bw() + 
    theme(axis.text.x  = element_blank(), 
          axis.ticks.x = element_blank(),
          axis.title = element_blank(),
          legend.position = "top",
          legend.title = element_blank()) +
    scale_fill_manual(values = c("#B3B3B3", "#FC8D62"), labels = c("Negative  ","Positive  "))

```

```{r ESBL-spec-sens, echo = F, warning = F, message = F,fig.scap="Species and antimicrobial sensitivities of cultured ESBL-E", out.extra='', fig.cap = "Species (A) and antimicrobial sensitivities (B) of cultured ESBL-E",  fig.height = 5, fig.width=4, fig.align = 'center' }

labz <-  c(expression(italic("Escherichia coli")),
           expression(italic("Klebsiella pneumoniae")),
           expression(italic("Enterobacter cloacae")),
           expression(italic("Klebsiella oxytoca")),
           expression(italic("Raoultella ornithinolytica")),
           expression(italic("Acinetobacter baumanii")),
           "Other"
)
      
                      

lims_orgs %>% 
  filter(!is.na(organism)) %>%
  mutate(organism = recode_factor(organism,
                           `Escherichia coli` = "Escherichia coli",
                           `Klebsiella pneumoniae` = "Klebsiella pneumoniae",
                           `Enterobacter cloacae` = "Enterobacter cloacae",
                           `Klebsiella oxytoca` = "Klebsiella oxytoca",
                           `Raoultella ornithinolytica` = "Raoultella ornithinolytica",
                           `Acinetobacter baumanii` = "Acinetobacter baumanii",
                           .default = "Other", .ordered = TRUE)) %>% 
  ggplot(aes(fct_rev(organism))) + 
  geom_bar() + coord_flip() + 
  scale_x_discrete(labels = rev(labz)) +
  xlab(element_blank()) + ylab("n") + theme_bw() + theme(axis.text = element_text(size = 8))-> p.org

lims_sens[3:8][lims_sens[3:8] == "Sensitive"] <- 0
lims_sens[3:8][lims_sens[3:8] == "Resistant" | lims_sens[3:8] == "Intermediate"] <- 1
lims_sens[3:8] <- sapply(lims_sens[3:8], as.numeric)
lims_sens$n_not_na <- apply(lims_sens[3:8], 1, sum)
lims_sens <- filter(lims_sens, n_not_na > 0 & !is.na(n_not_na))

lims_sens %>% 
  select(-c(sample_number, n_not_na)) %>%
  pivot_longer(-organism) %>% 
  filter(organism %in% c("Escherichia coli","Klebsiella pneumoniae")) %>% 
  mutate(value = as.character(value),
    value = recode_factor(value, 
                          `0` = "Sensitive", 
                          `1` = "Resistant/Intermediate", 
                          .ordered = TRUE)) %>%
  ggplot(aes(name, fill= value)) +
  geom_bar(position = "fill") + 
  coord_flip() + 
  facet_wrap(~ organism, ncol = 1) + 
  scale_fill_manual(values = c("#B3B3B3", "#FC8D62")) +
  theme_bw() + 
  theme(legend.title = element_blank(),legend.position = "bottom", axis.text = element_text(size = 8)) + 
  xlab(element_blank()) +
  ylab("Proportion") -> psens

ggarrange(p.org, psens, ncol = 1, nrow = 2, heights = c(1,2))
```

### Associations of ESBL colonisation

I then used logistic regression to explore associations of ESBL-E colonisation at baseline Of the 420 participants with an available enrolment stool culture result, 42% (178/420) cultured at least one ESBL-E. Univariable and multivariable associations of colonisation at enrolment are shown in \@ref(tab:esbl-baseline-model). In univariable analysis HIV infection, ART and CPT are associated with ESBL-E colonisation. This seems to be largely mediated by CPT as the HIV and ART associations largely disappear on multivariable modelling but the effect of CPT is still apparent (aOR 2.3 [95% CI 1.0 - 5.5]). Hospitalisation within the 4 weeks prior to admission was strongly associated with ESBL-E colonisation on multivariable modelling, though with wide confidence intervals (aOR 5.9 [95% CI 1.8-27.0]). Antimicrobial exposure was not, but with confidence intervals that contained a clinically relevant effect size (aOR 1.3 [95% 0.7 - 2.6]). ESBL-E colonisation was more likely with more adults in the household (aOR 1.2 [95% CI 1.0-1.4] per extra adult) , with use of an unprotected water source (aOR 3.0 [95% CI 1.1 - 8.8]) and in the rainy season (aOR 2.2 [95% CI 1.4-3.4]).

To explore associations of acquisition of ESBL-E by the day 28 visit, I analysed only those participants who had no detectable ESBL-E at baseline, and an available follow up samples at 28 days +/- 14 days. These numbered 150 participants: 64 Arm 1, 37 Arm 2 and 49 Arm 3 participants, and 49% (73/150) of them had a detectable ESBL-E at day 28. Bivariable associations of ESBL-E acquisition with antimicrobial and hospital exposures are shown in Figure \@ref(fig:esbl-aq-mod)A, stratified by the length of exposure; all antibacterials (including TB therapy) showed an association with ESBL-E acquisition, with a suggestion of a dose-response effect (in that longer exposures generally had a higher colonisation prevalence), but confidence intervals were large in many cases. Antimalarials did not show this effect though here uncertainty in the estimates precludes drawing any firm conclusions, as it does for antifungals. These relationships are very likely confounded, so should be regarded with caution; however, due to a small dataset size and collinearity, logistic regression modelling of ESBL-E acquisition (Figure \@ref(fig:esbl-aq-mod)B) produces such uncertain parameter estimates that no conclusions can be drawn. A better modelling strategy using continuous time Markov models is presented in Chapter 8.

\clearpage

```{r esbl-baseline-model, echo = FALSE, warning = FALSE, message = FALSE}

merge(enroll, subset(lims, visit == 0)) -> bl.esbl
#bl.esbl <- merge(bl.esbl, select(bloods, pid, CD4_Absolute), all.x = T )
bl.esbl %>% select(d2arm, calc_age, ptsex, hivstatus, hivonart, hivcpt, tbongoing,
                   recieved_prehosp_ab,pmhxrechospital,
                   watersource, watertreated, toilet,
                   housholdadultsno, householdchildno,
                   keepanim, 
                   ESBL, enroll_date, ) -> bl.esbl


bl.esbl$d2arm[is.na(bl.esbl$d2arm)] <- "Arm 3 (Community)"

bl.esbl$watersource <- recode(bl.esbl$watersource,
                              `Unprotected well/spring` = "unprotected",
                              `Surface water (including rainwater collection)` = "unprotected",
                              .default = "protected")

bl.esbl$toilet <- as.numeric(bl.esbl$toilet == "Flush Toliet (any type)")

bl.esbl$ESBL <- as.numeric(bl.esbl$ESBL == "Positive")  
  
bl.esbl$season <- "dry"
bl.esbl$season[month(bl.esbl$enroll_date) >= 11 | month(bl.esbl$enroll_date) <= 4] <- "rainy"
bl.esbl$hivcpt[is.na(bl.esbl$hivcpt)] <- 0
bl.esbl$hivonart[is.na(bl.esbl$hivonart)] <- 0
bl.esbl <- select(bl.esbl, -enroll_date)
bl.esbl[bl.esbl == "Yes"] <- 1

bl.esbl[bl.esbl == "No"] <- 0

bl.esbl$tbongoing[bl.esbl$d2arm == "Arm 3 (Community)"] <- 0
bl.esbl$tbongoing[is.na(bl.esbl$tbongoing)] <- 0
bl.esbl$recieved_prehosp_ab[bl.esbl$tbongoing == 1] <- 1
bl.esbl <- select(bl.esbl, -tbongoing)

univ_or <- function(df, oc_var) {
  out <- list()
  varz <- names(df)
  varz <- varz[!grepl(oc_var, varz)]
  for (i in 1:length(varz)) {
    form <- formula(paste0(oc_var, " ~ " ,varz[i]))
    modout <- glm(formula = form, data = df, family = "binomial")
    dfout <- bind_cols(tidy(modout), confint_tidy(modout))
    dfout[c(2:4,6,7)] <- sapply(dfout[c(2:4,6,7)], exp)
    dfout <- dplyr::select(dfout, term, estimate, conf.low, conf.high, p.value)
    dfout <- dplyr::filter(dfout, term != "(Intercept)")
    out[[i]] <- dfout
  }
  out <- do.call(rbind, out)
  return(out)
}

uv <- univ_or(bl.esbl, "ESBL")

glm(ESBL ~ ., data = bl.esbl, family = 'binomial') -> mv.mod
  dfout <- bind_cols(tidy(mv.mod), confint_tidy(mv.mod))
    dfout[c(2:4,6,7)] <- sapply(dfout[c(2:4,6,7)], exp)
    dfout <- dplyr::select(dfout, term, estimate, conf.low, conf.high, p.value)
    dfout <- dplyr::filter(dfout, term != "(Intercept)")

  uv$str <- uv %>% glue_data('{specify_decimal(estimate, 2)} ({specify_decimal(conf.low, 2)}-{specify_decimal(conf.high, 2)})')
  dfout$str <- dfout %>% glue_data('{specify_decimal(estimate, 2)} ({specify_decimal(conf.low, 2)}-{specify_decimal(conf.high, 2)})')
    names(uv)[2:6] <- paste0("uv_", names(uv)[2:6] )
    names(dfout)[2:6] <- paste0("mv_", names(dfout)[2:6] )
    bl.esbl.tab <- merge(uv, dfout) %>% select(term, uv_str, uv_p.value, mv_str, mv_p.value)
    bl.esbl.tab$boldrows.mv <- bl.esbl.tab$mv_p.value < 0.05
    bl.esbl.tab$boldrows.uv <- bl.esbl.tab$uv_p.value < 0.05
    bl.esbl.tab$uv_p.value <- specify_decimal(bl.esbl.tab$uv_p.value, 3)
     bl.esbl.tab$mv_p.value <- specify_decimal(bl.esbl.tab$mv_p.value, 3)
      bl.esbl.tab$uv_p.value[ bl.esbl.tab$uv_p.value  == "0.000"] <- "<0.001" 
      
      recode.str <- c(calc_age = "Age (per year)",
                      ptsexMale = "Male sex (vs female)",
                      `d2armArm 2 (Inpatient)` = "Arm 2 (vs 1)",
                      `d2armArm 3 (Community)` = "Arm 3 (vs 1)",
                      
                      hivstatusReactive = "HIV+ (vs HIV-)",
                      hivstatusUnknown = "HIV unknown (vs HIV-)",
                      hivcpt1 = "CPT (vs none)",
                      hivonart1 = "ART (vs none)",
                      tbongoing1 = "Current TB treatment",
              
                      pmhxrechospital1 = "Hospitalisation",
                      recieved_prehosp_ab1 = "Antibiotics*",
                      
                      householdchildno = "Children (per 1)",
                      housholdadultsno = "Adults (per 1)",
                      keepanim1 = "Keep animals (vs not)",
                      toilet = "Flushing toilet (vs not)",
                      watersourceunprotected = "Unprotected water source",
                      watertreated1 = "Treat water (vs not)",
                      
                      seasonrainy = "Rainy season (vs. dry)"
                      
                      )
  
      recode_factor(bl.esbl.tab$term , !!!recode.str, .ordered = TRUE)   -> bl.esbl.tab$term
      bl.esbl.tab[order(bl.esbl.tab$term),] -> bl.esbl.tab
      
      bl.esbl.tab[2:3]<- sapply(bl.esbl.tab[2:3], function(x) text_spec(x, "latex", bold = bl.esbl.tab$boldrows.uv))
           bl.esbl.tab[4:5]<- sapply(bl.esbl.tab[4:5], function(x) text_spec(x, "latex", bold = bl.esbl.tab$boldrows.mv))
      
      kable(select(bl.esbl.tab, -c(boldrows.mv, boldrows.uv)), "latex", booktabs = TRUE,
      row.names = FALSE, escape = FALSE, col.names = c("Variable","OR (95\\% CI)", "p-value", "aOR (95\\% CI)", "p-value"),
      caption = "Univariable and multivariable associations of ESBL colonisation at enrolment") %>% 
        kable_styling(full_width = FALSE, latex_options = "hold_position", font_size = 9) %>%
       pack_rows("Demographics", 1,2, bold = FALSE) %>%
       pack_rows("Study Arm", 3,4, bold = FALSE) %>%
       pack_rows("HIV status", 5,8, bold = FALSE) %>%
       pack_rows("Exposures last month", 9,10, bold = FALSE) %>%
       pack_rows("Household size", 11,13, bold = FALSE) %>%
       pack_rows("WaSH behaviour", 14,16, bold = FALSE) %>%
       pack_rows("Season", 17,17, bold = FALSE) %>%
  add_header_above(c(" " = 1, "Univariable" = 2, "Multivariable" = 2)) %>% 
  footnote(general = "CPT = Cotrimoxazole preventative therapy, ART = antiretroviral therapy, WaSH = Water, sanitation and hygiene. Entries in bold are those for which 95% confidence intervals do not cross 1.", symbol = "Antibiotics includes TB therapy but excludes CPT.", threeparttable = T, fixed_small_size = T) 
      
```


```{r, esbl-aq-mod, echo = FALSE, message = FALSE, warning = FALSE, fig.scap="Univariable and multivariable associations of ESBL-E acquisition by 28 days", out.extra='', fig.cap = "Univariable (A) and multivariable (B) associations of antimicrobial and hospital exposure with acquisition of ESBL-E by 28 days. A: These plots show the proportion of participants who have no detectable ESBL-E baseline but who do at 28 days, as a function of various exposures. All antibacterials and hospitalisation show an association between exposure and ESBL-E acquisition, with a suggestion of a dose-response relationship, though confidence intervals are wide in many cases. Antimalarials show no apparent relationship though, as with fluconazole, the wide confidence intervals make it difficult to draw any conclusions. The results from logistic regression to predict ESBL-E acquisition are shown in (B); colinearity and small dataset size means that confidence intervals are so large as to make the model useless. ",  fig.height = 7, fig.width=7, fig.align = 'center'}

lims_dates$delta28 <- abs(28 - lims_dates$assess_type)

lims_dates %>% 
  filter(assess_type >= 14 & assess_type <= 42) %>%
  mutate(delta28 = abs(28-assess_type)) %>%
  group_by(pid) %>% slice(which.min(delta28)) -> d14.42


merge(d14.42, lims_dates %>% filter(visit == 0) %>% select(pid, ESBL), all.x = T , by = "pid") -> d14.42

names(d14.42)[names(d14.42) == "ESBL.x"] <- "ESBL.d28"
names(d14.42)[names(d14.42) == "ESBL.y"] <- "ESBL.d0"

exposures.42 <- df.covs.cpt %>% group_by(pid) %>% do(extract_covariate_exposure(dfin = ., 4, 26, 0, 42))

d14.42 <- merge(d14.42, exposures.42, by = "pid", all.x = T)

d14.42[c(12:34)] <- apply(d14.42[c(12:34)], 2, function(x) ifelse(x > d14.42$assess_type, d14.42$assess_type, x))

#d14.42 <- select(d.14.42pid, assess_type, ESBL.d0, ESBL.d28, names(exposures.42))

select(d14.42, pid, assess_type, ESBL.d0, ESBL.d28, names(exposures.42)) -> d

d <- filter(d,ESBL.d0 == "Negative")
d$ESBL.d28 <- as.numeric(d$ESBL.d28 == "Positive")

cutz <- c(-1,0,5, 42)

cutnames <- levels(cut(d$assess_type, c(-1,0,5, 42)))

out <- list()
for (i in 1:(length(cutz)-1)) {
  out[[i]] <- paste0(cutz[i], "-",cutz[i+1])
}

unlist(out) -> cutnewnames

cutnewnames[2] <- "1-5"

paste0(cutnewnames, " days") -> cutnewnames
cutnewnames[1] <- "0 Days"
names(cutnewnames) <- cutnames

d %>% select(ESBL.d28,  amoxy, cotri, cefo, cipro, hosp, tb, coart, arte, fluco) %>%
    pivot_longer(-ESBL.d28) %>%
    mutate(value.cut = cut(value, c(-1,0,5, 42))) %>% 
    group_by(name, value.cut) %>% 
    summarise(n = n(), e = sum(ESBL.d28 == 1), prop = e/n, lci = binom.test(e,n)$conf.int[[1]], uci = binom.test(e,n)$conf.int[[2]]) %>% ungroup() %>%
    mutate(name = recode_factor(name, !!!abs.lookup[c("amoxy", "cipro", "cotri", "cefo", "arte", "coart", "tb", "fluco","hosp")], .ordered = TRUE),
           value.cut = recode_factor(value.cut, !!!cutnewnames, .ordered = TRUE )) %>%
    ggplot(aes(value.cut, prop, ymin = lci, ymax = uci, group = name)) + geom_point() + geom_line() + theme_bw() + facet_wrap(~ name, ncol = 4) + 
  geom_errorbar(width = 0) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + xlab(element_blank()) +
  ylab("Proportion acquire ESBL-E at 28 days") -> pa

glm(ESBL.d28 ~ (cefo > 0) + (amoxy > 0) +  (cotri > 0)+ (hosp > 0) + (tb > 0) + (cipro > 0), data = d, family = 'binomial') -> m                           
bind_cols(tidy(m),confint_tidy(m)) -> mod

mod$term <- map_chr(mod$term, ~  strsplit(., ">")[[1]][[1]]) %>% str_trim

mod %>% filter(term != "(Intercept)") %>% 
  mutate(term = recode_factor(term, !!!abs.lookup[c("amoxy", "cipro", "cotri", "cefo", "tb","hosp")], .ordered = TRUE)) %>%
           ggplot(aes(fct_rev(term), exp(estimate), ymin = exp(conf.low), ymax = exp(conf.high))) + 
           geom_point() +
           geom_errorbar(width = 0) +
           coord_flip() +
           theme_bw() +
           geom_hline(yintercept = 1, linetype = "dashed") +
  xlab(element_blank()) +
  ylab("aOR of ESBL acquisition by 28 days") -> pb

ggarrange(pa, 
          ggarrange(NULL,pb, NULL, ncol = 3, nrow = 1, widths = c(0.4,1.5,0.6)),
          ncol =1 , nrow =2, heights = c(3,1.5), labels = c("A", "B")
)

```

\clearpage

## Discussion

In this chapter, I have presented the data which will be used to address the second aim of this thesis - to describe, and identify determinants of, ESBL-E acquisition and carriage in Malawian adults. It is possible to draw several conclusions from these data. First, community ESBL-E carriage in Blantyre is common. The baseline community carriage prevalence of 28% is considerably higher than the 4-7% seen in Europe[@McNulty2018;@Wielders2017;@Ny2017;@Valverde2004] and comparable to the sSA pooled community prevalence estimate presented in Chapter 1 of 18% (95% CI [11-27%]). 

The associations of baseline ESBL-E carriage give insight into the routes of community ESBL-E transmission in the urban Malawian setting. The high community prevalence without apparent healthcare contact suggests that community transmission is common. Household crowding and use of unprotected water sources are associated with ESBL-E colonisation, suggesting both household person to person and environmental transmission routes are of relevance. The number of adults in the household, rather than the number of children, was associated with ESBL-E carriage in this study - which recruited only adults - suggesting that within the household adult to adult transmission is a more important route than child to adult transmission. This could be for a number of reasons - if the ESBL-E prevalence were low in children, for example. Though children were not sampled in this study and so the data here can not address that hypothesis, data from other studies suggest this is unlikely: community prevalence in children ranged from 10-59% in four studies in the Central African Republic[@Farra2016], Senegal[@Ruppe2009] and Tanzania[@Tellevik2016;@Moremi2017], and is hence comparable to the adult community prevalence seen in this study. Behavioural factors, or a lower bacillary burden in children could also account for the associations seen here. The seasonality of ESBL-E carriage prevalence is also consistent with an environmental transmission route of ESBL-E - environmental faecal and hence ESBL-E contamination would be likely higher in the rainy season - but behavioural factors (e.g. more indoor crowding during the rains) or other causal pathways (e.g. more febrile illness and hence hospitalisation and/or antimicrobial exposure) should also be considered as possibilities.

These data also offer insight into the role of HIV as a driver of ESBL-E colonisation.  Prior to this study, only two studies in sSA had assessed associations of HIV and community ESBL-E carriage: one, in pregnant women, found no association[@Nelson2014], and one, in healthy children under two, found a strong association[@@Tellevik2016]. HIV is known to have a profound effect on gut function[@Dillon2016] and it is conceivable that a direct HIV effect on e.g. the microbiota could result in reduced colonisation resistance to ESBL-E, or that increased antimicrobial or healthcare exposure in the HIV infected could result in a higher colonisation prevalence. However, in the Malawian adults considered here, there is a clear association between HIV-infection and ESBL-E carriage which seems to be largely mediated by CPT. Given that the adult HIV prevalence in Malawi is estimated to be 9.6%[@UNIADS2018] and Malawian HIV guidelines mandate lifelong CPT for people living with HIV[@MinistryofHealth.GovernmentofMalawi2016], CPT is potentially a significant driver of ESBL-E colonisation in this setting.

It is also clear that there are significant associations between antimicrobial exposure, hospitalisation, and ESBL-E carriage prevalence, though the analysis presented here is unable to provide an insight into the relevant importance of hospitalisation versus antimicrobial exposure. Prior hospitalisation was strongly associated with baseline ESBL-E carriage on multivariable logistic regression, but prior antimicrobial exposure (apart from CPT) was not, though with confidence intervals that include a clinically relevant effect size. A better estimate of the relative effects can come from the longitudinal sampling data. The study was designed to explore the relative effects of hospital exposure and antimicrobial exposure on ESBL-E colonisation, and so to produce groups with different levels of these exposures. Despite some crossover of exposure, this was largely a success, with much reduced antimicrobial exposure in Arm 2 compared to Arm 1, and virtually no antimicrobial of hospital exposure in Arm 3. In an unadjusted analysis, Arm 1 participants show a greater increase in ESBL-E carriage prevalence than Arm 2 participants, suggesting that antimicrobial exposure, rather than hospitalisation, is driving apparent acquisition. The prevalence of ESBL-E colonisation following antimicrobial exposure is striking, with 78% of Arm 1 participants colonised by the day 7 visit. A return to pre-admission prevalence is apparent over the 6-month study period.

However this conclusion - that antimicrobial exposure rather than hospitalisation is driving apparent ESBL-E acquisition - is very open to bias from confounding, because antimicrobial-unexposed participants are different from antimicrobial unexposed: younger, less likely to be HIV-infected, less CPT exposure, and a shorter length of hospital stay. A multivariable approach is necessary to provide unbiased estimates of the effects of antimicrobial use and hospitalisation, but collinearity of exposures and a small dataset size means that a simple modelling approach - logistic regression - produces such uncertain effect sizes as to be useless. A different modelling approach, using continuous time Markov models, is presented in Chapter 8.

The overall trend of ESBL-E carriage at any time point in the study obscures the complex within-individual picture, with a pattern of multiple transitions between the detectable and undetectable ESBL-E states over the study period for many individuals. There could be several reasons for this: if a participant is truly continually colonised, then intermittent shedding of ESBL-E or inadequate test sensitivity could explain apparent lack of ESBL-E in culture. There are clearly some participants who are continually colonised except for at one or two time points, which is suggestive of one or both of these scenarios. Alternatively, these patterns could represent true acquisition or loss events.

The ESBL-E cultured from the participants in this study are, as expected, largely _E. coli_ and, to a lesser extent _K. pneumoniae_, as seen in other studies in sSA (see Chapter 1), but also in high income settings such as Europe[@VanDuijkeren2018]. The prevalence of amikacin and meropenem resistance was very low, although was present. Meropenem was introduced to the Malawian national formulary in 2015[@MinistryofHealth.GovernmentofMalawi2015], and is only sporadically available in QECH. Previous studies of antimicrobial resistance in Blantyre have not found any genotypic or phenotypic carbapenem resistance, though the MLW laboratories do not routinely test for carbapenem sensitivity. To my knowledge the carbapenem resistant isolates found in this study are the first to be described in Malawi, and the mechanisms of carbapenem resistance are further explored in Chapter 6. Cotrimoxazole resistance was near universal, which may be related to widespread CPT use. Interestingly, many of the isolated ESBL-E were sensitive to chloramphenicol. This drug was, prior to the introduction of ceftriaxone, first line treatment (along with penicillin) for sepsis in Malawi, but was replaced due to a poor side effect profile and the ease of administration of once-daily ceftriaxone. Clearly these isolates are carriage rather than invasive isolates and so may have different sensitivity patterns to the bacteria that cause invasive infections, but this raises the prospect that chloramphenicol might be reintroduced as a reserve antibiotic in the treatment of ESBL-E in Malawi. 

### Limitations

There are limitations to this study and analysis. There was significant drop-out of Arm 1 and 2 participant through the study period. The proportion of participants truly lost to follow up or voluntarily withdrawn from the study was low, however, and most withdrawals were due to death or transfer out, which reflect the study populations of interest in urban Blantyre. Logistic difficulties - participants being unable to travel to deliver a sample, or being unavailable for a home visit - resulted in failure to collect 13% of eligible sample, and again reflects the challenges inherent in the setting. Though the study protocol mandated the timing of visits, the actual visit times have a broad temporal distribution. Nevertheless, there does not seem to be any systematic bias in the missingness of participants or samples, and if that is the case then the conclusions drawn should stand. HIV testing was not carried out on community members, and so there is a high proportion Arm 3 participants with an unknown HIV status, and the status that is recorded is self reported, which could result in misclassification of HIV status and hence bias. Indeed, all baseline exposures were self-reported and not verified, which could result in bias. 

## Conclusions and further work

ESBL-E colonisation in common in adults in Blantyre, and ongoing community transmission with both person to person and environmental transmission routes seems likely. Rapid apparent acquisition of ESBL-E occurs in hospitalised participants exposed to antimicrobials with a return to pre-admission ESBL-E colonisation prevalence after around six months. The aggregate data conceals a complex picture within individuals, however, with multiple apparent transitions between the colonised and uncolonised state for many individuals over the study period. The relative contribution of hospitalisation and antimicrobial exposure in driving apparent acquisition events is not clear and logistic regression models failed to provide any insight. 

Several questions arise from this analysis and form the basis of further work in this thesis. 

1. What is the mechanism underlying the frequent state transitions (from colonised to uncolonised and back again) for many participants in this study? 

2. Are these true acquisitions and losses, intermittent shedding, or a failure to detect ESBL-E which are present?  

3. What are the relative contributions of hospitalisation versus antimicrobial exposure in driving the sharp increase in ESBL-E carriage prevalence following hospital admission and antimicrobial exposure? Specifically, what is the biological mechanism of this apparent increase? Is it associated primarily with hospitalisation - which could represent a true ESBL-E acquisition from a contaminated hospital environment - or primarily with antimicrobial exposure - which could be an enrichment of already-carried ESBL-E which was undetected at baseline - or a synergistic combination of the two? 

I will spend the next three chapters attempting to address these questions, using whole genome sequencing as a high resolution bacterial typing tool to track bacteria within participants, and continuous-time Markov models to model state transitions between colonised and uncolonised states over time. 


