# The genomic landscape of ESBL producing _E. coli_ in Blantyre, Malawi
\chaptermark{Genomic landscape of ESBL E. coli}

``` {r ch7-setup, include = F}

wd <- "chapter_7/"

# and packages
library(plyr)
library(tidyverse)
library(reshape2)
library(knitr)
library(kableExtra)
library(grid)
library(ggpubr)
library(ggtree)
library(ape)
library(phytools)
library(RColorBrewer)
library(scales)
library(cowplot)
library(pheatmap)
library(viridis)
library(ggraph)
library(igraph)
library(gggenes)
library(IRanges)
library(glue)


 parse_checkm_lines <- function(s, n) {
  strsplit(s, " ") -> stemp
  stemp <- stemp[[1]][stemp[[1]] != ""] 
  stemp <- c(stemp[1], paste0(stemp[2],"_", stemp[3]), stemp[4:15])
  names(stemp) <- n
  return(stemp)
 }


output = "latex"
source("other_scripts/load_metadata.R")
source("other_scripts/function_parse_cdhitest.R")
source("other_scripts/parse_blast_op_IRranges.R")

specify_decimal <- function(x, k) trimws(format(round(x, k), nsmall=k))

```
## Chapter overview

In this chapter I present the results of whole-genome sequencing of 473 _E. coli_ isolates, cultured from 230 participants at a median of 2 (IQR 1-5) time points per participant. These represent one colony pick from all stool samples which cultured _E. coli_ up to the time of shipping in October 2018. The isolates were largely phylogroup A (43% [204/473]), and multilocus sequence type 131 was the most common ST (14% [64/473]). A global collection of 800 _E. coli_ genomes was used to put the isolates from this study in a global context and a maximum likelihood phylogenetic tree constructed. Isolates from this study were largely spread throughout the tree, but there were exceptions: Malawian ST410 and ST167 were monophyletic suggesting perhaps recent introduction to Malawi. These are recognised emerging high-risk clones. These findings suggest that there no obvious restrictions in mixing of _E. coli_ between Blantyre and the rest of the world. 

A diverse range on AMR genes were identified, with a median 16 (IQR 12-17) per isolate. ESBL genes were largely of the $bla_{CTXM}$ family, and dominated by $bla_{CTXM-15}$. The abundance of genes seems to reflect local antibiotic pressures: genes conferring cotrimoxazole and aminoglycoside resistance were almost ubiquitous, and quinolone, chloramphenicol and tetracycline resistance genes also frequently seen. Carbapenem antibiotics are rarely available, and carbapenem resistance genes rare. One carbapenemase, a $bla_{NDM-5}$, was identified, on an IncX plasmid very similar to one first identified in India around 2011. To my knowledge the first carbapenemase described in Malawi. Its identification despite little antibiotic pressure along with the apparent ease international of spread of _E. coli_ suggested by this analysis suggest widespread resistance could rapidly emerge following wider carbapenem roll out. Antibiotic stewardship interventions that can balance access to and restriction of last line antibiotics in low resource settings are urgently needed.

## Introduction and chapter aims

The previous chapter presented an overview of the longitudinal ESBL-E sampling that was undertaken in this study. It is the overall aim of this thesis to use whole genome sequencing (WGS) as a high-resolution typing method to longitudinally track bacteria within participants. In this chapter, however, I present a descriptive analysis of the genomes of 473 _E. coli_ isolates sequenced for this study, before moving on to this specific aim in the following chapter. The aims of the analysis presented in this chapter are, therefore:

1. Perform quality assurance and control of the sequenced genomes for downstream analysis.

2. Place the isolates from this study in a global context using phylogenetics and a global _E. coli_ collection comprised of available genomes.

3. Provide a description of the AMR determinants in the sequenced _E. coli_ isolates.

## Methods

### Bioinformatic pipeline

One _E. coli_ colony from each patient sample was picked for DNA extraction and 150bp paired-end short-read whole genome sequencing (WGS) using Illumina HiSeq X10 at the Wellcome Sanger Institute (WSI). Read quality control was undertaken with Kraken v0.10.6 and Braken v1.0 to assign reads to species[@Wood2014] and the WSI QC pipeline which maps a random 100 Mbases from each sample to a reference and calculates depth of coverage, number of heterogeneous SNPs, GC content and insert size. Samples that contained > 80% non _E coli_ reads were discarded the remainder were _de novo_ assembled with SPAdes v3.11.0[@Bankevich2012a]. Assembly statistics (e.g. assembly length, N50) were calculated with QUAST v4.6.0[@Gurevich2013] and completeness and contamination of the assemblies assessed by checkM v1.0.7[@Parks2015], which uses presence or absence of lineage-associated collocated ubiquitous single-copy genes to estimate these quantities. Contaminated assemblies (with checkM-defined contamination of > 25%) or poor assemblies (with less than 1Mb assembled length) were discarded. Annotation was carried out with prokka v1.5[@Seemann2014a] with a genus specific database from RefSeq and the Roary v1.007 pan-genome pipeline[@Page2015] was used to identify a core genome, considering genes contained in > 99% of isolates to be core. A core gene multiple sequence alignment was generated using maaft v7.205[@Katoh2013], SNP-sites identified using SNP-sites v2.4.1[@Page2016] and the resultant SNP alignment used to build a maximum likelihood phylogenetic tree using IQ-TREE v1.6.3[@Nguyen2015], using ascertainment bias correction to correct for the fact that the input pseudosequence contained only variable sites, and using the ModelFinder module used to find the best fitting nucleotide substitution model. This calculates the likelihood of a number of different models and chooses the model with the highest (best fitting) Bayesian Information Criterion, a statistic which penalises model parameters. Reliability of inferred branch partitions was assessed with 1000 bootstrap replicates. Trees were visualised in the ggtree v1.14.4 package[@Yu2017] in R. 

ARIBA v2.12.1[@Hunt2017a] was used to identify AMR-associated genes using the SRST2 database[@Inouye2014], to identify plasmid replicons using the PlasmidFinder database[@Carattoli2014] and to perform _in silico_ muli-locus sequence typing (MLST) using the database from http://mlst.warwick.ac.uk/mlst/dbs/Ecoli accessed via www.pubmlst.org. The $\beta$-lactamase genes $ampC1$, $ampC2$ and $ampH$ were excluded from the analysis of AMR determinants as they do not usually cause a resistant phenotype in _E. coli_. Because quinolone resistance often results from SNPs in the chromosome in the quinolone resistance determining regions (QRDRs) of the _gyrA_, _gyrB_, _parE_ and _parC_ genes - rather than acquisition of whole AMR-determining genes, as is the case with the other genes sought by ARIBA - these genes were downloaded from the comprehensive antimicrobial resistance database (CARD, https://card.mcmaster.ca/) and ARIBA used to call SNPs in them, with default settings. _E. coli_ phylogrouping was performed with a quadruplex _in silico_ PCR using the Clermont scheme[@Clermont2013] and _isPcr_ v33x2 (https://github.com/bowhan/kent/tree/master/src/isPcr) 


### Global _E. coli_ collection

In order to place the isolates from this study in a global context, published _E. coli_ assemblies were downloaded from the WSI servers. These included 149 ESBL-producing _E. coli_ from a single centre study in Chachoengsao province, eastern Thailand[@Runcharoen2017]. In this study, human clinical isolates from standard care in Bhuddhasothorn hospital were selected on the basis of the ESBL phenotype, and environmental samples were collected as part of a cross sectional study and selectively cultured for ESBL-E in 2014-2015. I also downloaded assemblies of 362 enterotoxogenic _E. coli_ (ETEC), selected for an ETEC genomic study from the Gothenburg University ETEC collection to represent a broad collection of ETEC isolated worldwide from 1980-2011[@VonMentzer2014]; 185 atypical enteropathogenic _E. coli_ (aEPEC) sequenced for a study of aEPEC and selected from samples from the Global Enteric Multicentre Study (GEMS) in seven centres in Africa and Asia between 2007-2011[@Ingle2016]; and 94 _E. coli_ from QECH in Blantyre, Malawi, a combination of invasive (bloodstream and CSF) and carriage isolates, selected for diversity in AMR phenotype from 1996-2014[@Musicha2017]. Details of the included samples are given in the appendix to this chapter.

Phylogroup and MLST were determined for these context genomes as described above. AMR genes were identified with Ariba and the SRST2 database, as above, and context genomes were classified as ESBL if they contained any Group 2be ESBL gene (see Bush-Jacoby scheme in Chapter 1).

### Statistical analysis

Association of AMR genes with phenotype was expressed as odds ratios and tests of association used Fisher's exact test. In order to explore clustering of AMR genes, the Jaccard index was calculated for a given AMR-gene pair using the _philentropy_ v0.3.0 package in R. The Jaccard index, a measure of the similarity of two sets of data, is defined as _intersection over union_; in this context, for a given pair of AMR genes $x$ and $y$, the Jaccard index $J(x,y)$ is the number of isolates that contain both gene $x$ and $y$ divided by the total number that contain either $x$ or $y$. By definition it lies between 0 ($x$ and $y$ never co-occur) and 1 ($x$ and $y$ always co-occur). Co-occurrence matrices using the Jaccard index were plotted using the _pheatmap_ v1.0.12 package in R. The statistical significance of co-occurrence of genes was assessed by generating 2x2 contingency tables for a given gene pair and p values generated using a Fisher's test with Bonferroni correction; a p value of less that 0.05 was considered statistically significant. Co-occurrence networks of genes occurring commonly together (defined as Jaccard index > 0.5) at a rate greater than expected by chance (p < 0.05 following Bonferroni correction) were plotted using _igraph_ v1.2.2[@Csardi2006] and _ggraph_ v1.0.2 in R.

## Results

### Samples and quality assurance and control

There is a detailed description of microbiological procedures in Chapter 2. In total, 519 _E. coli_ underwent DNA extraction and were shipped from Malawi to WSI; these represented all sequential isolates at the time of final DNA extraction, which occurred in two batches in February 2018 and October 2018. Kracken/Bracken read assignment of these samples is shown in Figure \@ref(fig:wgs-qc-braken). The majority of samples have > 90% or reads assigned to _E. coli_; a minority have < 90% of reads assigned to _E. coli_ but a very closely related species such as _Shigella_, and as such are likely to be pure _E. coli_ culture with read misclassification. However, 12 samples have > 80% reads assigned to a non- _E. coli_ species such as _Klebsiella pneumoniae_. These samples were assumed to represent upstream species misidentification, and were excluded. There also exists the possibility of within-participant transcription error. In the freezer archive, all samples from a single participant at a single time point have the same sample ID, making an error possible; by definition (as only one _E. coli_ was sequenced at any time point in any individual) this would result in a species substitution. Samples from different time points and different participants are clearly demarcated however, making between-time point or between-participant errors very unlikely. Any such error would therefore result in a sample being identified as non- _E. coli_ and exclusion from the analysis, reducing power but not introducing bias.



```{r wgs-qc-braken,  echo = F, warning = F, message = F, fig.scap="Species read assignment of sequenced isolates", out.extra='', fig.cap= "Species read assignment of all samples. Each horizontal bar is one sample. Most samples have > 90\\% reads assigned to \\textit{E. coli} or related species (e.g. \\textit{Shigella spp.}) but twelve samples are likely not \\textit{E. coli} and excluded from further analysis.", fig.align= "center",fig.height = 10, fig.width = 6 , out.height = '90%'}

df1 <- read_tsv(paste0(wd, "D1_Braken_readscreen_species_composition.tsv"))
df2 <- read_tsv(paste0(wd, "D2_Bracken_species_composition.tsv"))
df3 <- read_tsv(paste0(wd, "D2_extra_krakenout_species_composition_16.tsv"))

df.brac <- bind_rows(df1, df2, df3)
df.brac[is.na(df.brac)] <- 0

melt(df.brac, id.vars = "name") -> dftemp

dftemp$variable <- factor(dftemp$variable, levels = unique(dftemp$variable[order(as.character(dftemp$variable))]))

dftemp$name <- factor(dftemp$name, levels = df.brac$name[order(df.brac$`Escherichia coli`, decreasing = T)])

ggplot(dftemp, aes(name, value, fill = variable)) + geom_col() + coord_flip() + scale_fill_brewer(palette = "Paired") +  theme_bw() + theme(axis.text.y = element_blank(), legend.title = element_blank(), legend.text = element_text(size = 8)) + labs(x= "Sample", y = "% reads" ) 




```

```{r wgs-load-qc,  echo = F, warning = F, message = F}

# load sanger QC

df1 <- read.csv(paste0(wd, "5341.pathfind_stats.csv"), stringsAsFactors = F)
df2 <- read.csv(paste0(wd, "5510.pathfind_stats.csv"), stringsAsFactors = F)

#df2$`No. Het SNPs`<- as.numeric(df2$`No. Het SNPs`)

df.qc <- rbind(df1, df2)


# laod checkM

con <- file(paste0(wd, "checkm_quast/D1/checkm.report"))
lines <- readLines(con)
close(con)

con <- file(paste0(wd, "checkm_quast/D220190318/checkm.report"))
lines2 <- readLines(con)
close(con)

con <- file(paste0(wd, "checkm_quast/D220190503/checkm.report"))
lines3 <- readLines(con)
close(con)

# line 2 has headings - get em
strsplit(lines[[2]]," ") -> heads
heads[[1]][heads[[1]] != "" & heads[[1]] != "#"] -> heads

# sigh
heads <- c(paste0(heads[1], "_", heads[2]), paste0(heads[3], "_", heads[4]), heads[5:6],
           paste0(heads[7],"_",heads[8]), heads[9:16], paste0(heads[17], "_", heads[18]))

lines <- lines[-c(1,2,3, length(lines))]
lines2 <- lines2[-c(1,2,3, length(lines2))]
lines3 <- lines3[-c(1,2,3, length(lines3))]
 
 

lines <- lapply(lines, parse_checkm_lines, heads)
lines2 <- lapply(lines2, parse_checkm_lines, heads)
lines3 <- lapply(lines3, parse_checkm_lines, heads)

lines <- data.frame(do.call(rbind, lines))
lines2 <- data.frame(do.call(rbind, lines2))
lines3 <- data.frame(do.call(rbind, lines3))

lines <- rbind(lines, lines2, lines3)
rm(lines2)
rm(lines3)
lines[3:ncol(lines)] <- sapply(lines[3:ncol(lines)], function(x) as.numeric(as.character(x)))

names(lines)[names(lines) == "Bin_Id"] <- "Assembly"

#sub( "_1_", "_1#", q$Assembly) -> q$Assembly
sub(".contigs_spades", "", lines$Assembly) -> lines$Assembly

checkm <- lines
rm(lines)

## and quast
rbind(
  read.delim(paste0(wd, "checkm_quast/D1/transposed_report.tsv"), stringsAsFactors = F ),
  read.delim(paste0(wd, "checkm_quast/D220190318/transposed_report.tsv"), stringsAsFactors = F ),
  read.delim(paste0(wd, "checkm_quast/D220190503/transposed_report.tsv"), stringsAsFactors = F )
  ) -> quast

for (i in 1:10) {
sub("\\.", "", names(quast)) -> names(quast)
}

sub(".contigs_spades","", quast$Assembly) -> quast$Assembly
sub("_1_","_1#", quast$Assembly) -> quast$Assembly
sub("_2_","_2#", quast$Assembly) -> quast$Assembly

df.qc <- merge(df.qc, df.brac, by.x = "Lane.Name", by.y = "name", all.x = T)
df.qc <- merge(df.qc, checkm, by.x = "Lane.Name", by.y = "Assembly", all.x = T)
df.qc <- merge(df.qc, quast, by.x = "Lane.Name", by.y = "Assembly", all.x = T)

# remove non ESCO

df.qc <- subset(df.qc, `Escherichia coli` > 20)
```

Of the remaining 507 samples, there were a median (IQR) of $2.3\times10^{6}$ ($2.1-2.5\times10^{6}$) reads, with a median (IQR) depth of coverage (obtained by mapping a random 100Mbases to a refernce _E. coli_ genome, Escherichia coli strain K-12 substrain MG1655, NCBI reference NC_000913.3) of 58 (51-66). One sample had an order of magnitude lower number of reads ($2.9\times10^{5}$) with depth of coverage 0; this was assumed to represent sequencing failure and it was excluded from further analysis.

The output from quast and checkM are shown in Figure \@ref(fig:wgs-qc), where N50 (the minimum contig length upon which at least half assembled bases are contained) is plotted as a function of total assembled length. The expected _E. coli_ genome length is around 4.6Mb and most samples cluster close to this at a total assembled length of ~ 5Mb. However it is clear that some assemblies  have failed, with low N50 and low assembled length. It is also apparent that some samples seem to be contaminated, as indicated by low N50 and much longer than expected total assembled length. Defining assembly failure as < 1Mb assembled length (triangles in the plot, n = 9) and contamination as checkM-defined contamination of > 25% (blue points in the plot, n = 24) and excluding both groups results in 33 further samples being excluded from further analysis.

In total, therefore, 46/519 (9%) of samples which were submitted for sequencing were excluded from downstream analysis. The remaining 473 samples represent 69% (474/686) of the cultured _E. coli_ in this study, and were recovered from 230 participants. 354 are from patients with sepsis, 86 are from hospitalised inpatients and 33 are from community members, with a median of 2 (range 1-5) samples per participant. 

```{r wgs-qc, echo = F, warning = F, message = F, fig.scap="N50 and total assembly length for included assemblies", out.extra='', fig.cap= "N50 as a function of total assembled length. Failed assemblies with less than 1Mb assembled shown as triangles. Contaminated assemblies with checkM-defined contamination above 25\\% shown in blue. Both of these groups of assemblies were exclduded from further analysis.", fig.height = 3, fig.width = 5, fig.align="center"  }

df.qc <- subset(df.qc, Reads > 300000)

ggplot(df.qc, aes(Totallength, N50, colour = Contamination > 25, shape = Xcontigs < 10)) + geom_point() + theme_bw() + labs(shape = "< 1Mb assembled", colour = "Contamination > 25%", x = "Total assembled length (bp)") 

```

```{r wgs-mlst-pgroup, echo = F, warning = F, message = F }

mlst <- read_csv(paste0(wd, "phylogroup_and_mlst/mlst.csv"))
pgroups <- read_csv(paste0(wd, "phylogroup_and_mlst/phylogroups.csv"))
```

### Phylogroup, MLST and core genome phylogeny of study isolates


```{r wgs-pgroup-tab, echo = F, warning = F, message = F }

tree <- read.tree(paste0(wd, "core_genome_tree/core_alnD2ESCO_snp_sites.aln.treefile"))
midpoint.root(tree) -> tree

pgroups %>% dplyr::filter(Lane %in% tree$tip.label) %>%
    dplyr::group_by(Phylogroup) %>%
    tally() %>%
    ungroup() %>% 
    mutate(tot = sum(n),
           prop = n/tot,
           n = glue('{n}/{tot} ({specify_decimal(prop*100,0)}%)')) %>%
  arrange(desc(prop)) -> pgroup.sum

kable(select(pgroup.sum, Phylogroup, n), "latex", booktabs = TRUE,
      row.names = FALSE, escape = TRUE,
      caption = "Phylogroup distribution of sequenced \\textit{E. coi} isolates.", linesep = "") %>%
  kable_styling(full_width = FALSE, font_size = 9) 


```
The commonest _E. coli_ phylogroup was phylogroup A, followed by phylogroup B2, F, B1 and C and D (Table \@ref(tab:wgs-pgroup-tab)). Two samples were Clade I or II (so called cryptic clades) and six were unknown phylogroup using the Clermont PCR scheme. In the MLST analysis, 56 recognised sequence types (STs) were identified, and 12 samples were novel STs; however over half (249/473 [53%]) of samples were represented by the top seven most frequent STs (Figure \@ref(fig:wgs-mlst)). ST131 was the most commonly isolated sequence type (64/473 [14%] of isolates) followed by ST410 (45/473 [10%] of isolates) and ST167 (38/473 [8%] of isolates).

The Roary pan-genome pipeline identified a core genome in the study isolates of 2966 genes, with a pan-genome of 26,840 genes. The resultant core gene pseudosequence of length 1,388,742 bases contained 99,693 variable sites, which were used to infer the maximum likelihood phylogenetic tree. The IQTREE ModelFinder module determined that a general time reversible (GTR) model with FreeRate site heterogeneity with 5 parameters provided the best fit to the data. The inferred tree is shown in Figure \@ref(fig:wgs-tree) along with isolate phylogroup and sequence types; in general, as expected, sequence types were largely monophyletic and phylogroups tended to cluster together.

```{r wgs-mlst, echo = F, warning = F, message = F, fig.cap="\\textit{E. coli} multilocus sequence type distribution", fig.height = 2, fig.width=6}

ggplot(mlst, aes(fct_infreq(ST))) + geom_bar()+
  xlab(expression(paste(italic("E. coli"), " sequence type"))) + ylab("Count") + theme_bw() + theme(axis.text.x=element_text(size=5, angle = 90) )

```

```{r wgs-tree, echo = F, warning = F, message = F, fig.scap="Phylogenetic tree of study \\textit{E. coli}", out.extra='', fig.cap="Midpoint rooted maximum likelihood phylogenetic tree of included study \\textit{E. coli} isolates showing phylogroups and sequence types. Bootstrap support of less than 90\\% is indicated by a black circle at a given node. Scale bar indicates 0.01 SNPs/site.", fig.height = 10, fig.width=8}

tree <- read.tree(paste0(wd, "core_genome_tree/core_alnD2ESCO_snp_sites.aln.treefile"))
midpoint.root(tree) -> tree

rownames(pgroups) <- pgroups$Lane
reshape2::dcast(mlst, lane ~ ST, fun.aggregate = base::length) -> mlst.onehot
#mlst.onehot <- subset(mlst.onehot, lane %in% tree$tip.label)

rownames(mlst.onehot) <- mlst.onehot$lane
mlst.onehot <- select(mlst.onehot, -lane)
n.mlsts <- sort(apply(mlst.onehot,2, sum), decreasing = T)
mlst.onehot[names(n.mlsts[n.mlsts > 1])] -> mlst.onehot

apply(mlst.onehot, 1, sum) == 0 -> mlst.onehot$other
mlst.onehot$other<- as.numeric(mlst.onehot$other)



mlst.onehot[mlst.onehot == 0] <- "Z_No"
mlst.onehot[mlst.onehot == 1] <- "Z_Yes"






#mlst.onehot[c(sort(as.numeric(names(mlst.onehot)[names(mlst.onehot) != "Novel"])), "Novel")] -> mlst.onehot


col <- c(hue_pal()(8), "lightgrey", "black")
names(col) <- c("A","B1","B2","C","Clade I or II", "D","F", "Unknown", "Z_No", "Z_Yes")

ggtree(tree) %>% gheatmap(select(pgroups, Phylogroup), font.size = 3, width = 0.1, 
                          colnames_position = "top", color = NA, colnames_offset_y = 5) +
  scale_fill_manual(values =col, labels = c("A","B1","B2","C","Clade I/II", "D","F", 
                                            "Unknown", "Absent", "Present"))-> p1x

leg1 <- get_legend(p1x)

ggtree(tree) %>% gheatmap(select(pgroups, Phylogroup), width = 0.1, 
                          color = NA, font.size = 4, colnames_angle = 90,
                          colnames_position = "top", colnames_offset_y = 30)  %>% gheatmap((mlst.onehot), font.size = 2.5, color = NA, 
                          colnames_angle = 90, offset=.03, colnames_offset_y = 10, colnames_position = "top",)  + geom_treescale(x =0, y = 0, offset = 5)  +   geom_point2(aes(subset = as.numeric(label) < 90 & !isTip), size = 1) + theme(legend.position = "none") +
  scale_fill_manual(values =col, labels = c("A","B1","B2","C","Clade I or II", "D","F", 
                                            "Unknown", "Absent", "Present")) + annotate("text", x= 0.28, y =510, label = "Sequence Type") -> p

plot_grid(p, leg1, ncol = 2, rel_widths = c(1,0.1))


```

### Study isolates in a global context

The global collection of _E. coli_ comprised 1273 samples, including the 473 from this study. 753/1253 (60%) were from Africa, 335/1253 (27%) from Asia and 167 (13%) from South America. The majority of samples, 1026/1253 (82%), were from stool, with 106/1253 (8%) truly invasive samples from blood or CSF and 63/1253 (5%) possibly invasive samples from urine, pus, or sputum. 65/1253 (5%) samples were environmental, all from Thailand. 670/1253 (53%) of samples contained at least one ESBL-encoding gene. The majority of isolates with an ESBL gene (622/670 [92%]) came from this study or the Thai ESBL study, a potential source of bias. Phylogroup A was the commonest phylogroup in the global collection (482/1273 [38%]), followed by B1 (333/1273 [26%]) and B2 (191/1273 [15%]); phylogroup C was uncommon in the global collection (74/1273 [6%]) but the majority of the phylogroup C samples came from this study (43/74 [58%]). All of these 43 phylogroup C isolates belonged to a single ST, ST410; this ST was not seen at all in the previous Malawian study of largely invasive isolates, despite being the second-commonest ST in this study, and was unusual in the global collection (11/800 [1%] ST410 in global collection vs 43/473 [9%] in this study). Similarly, the third-commonest ST in this study, ST167, was not seen at all in the global collection. However, ST131, the commonest ST in this study, was again the commonest ST in the global collection.

The Roary pan-genome pipeline identified 2872 core genes in a pan genome of 44,840 genes; this large pan-genome is consistent with the open _E. coli_ pan genome that will continue to increase in size as isolates are added. The core gene alignment contained 604,817 bases with 77,194 variable sites, which were used to infer the maximum likelihood phylogenetic tree, using the same nucleotide substitution model as previously.

The phylogeny is reconstructed in Figure \@ref(fig:wgs-global-tree). Isolates from this study are distributed throughout the tree, and there is widespread mixing of isolates from diverse geographic regions. Though invasive isolates are spread throughout the tree, there is a tendency for them to cluster together, particularly in phylogroup B2, a phylogroup with has a recognised association with ExPEC[@Dale2015]. The Malawian ST410 and ST167 isolates clustered tightly together, but by comparison, ST131 isolates from this study were distributed among ST131 isolates from other studies, both in Malawi and elsewhere (Figure \@ref(fig:wgs-131vs410-tree)).

```{r wgs-global-tree, echo = F, warning = F, message = F, fig.scap="Phylogenetic tree of study \\textit{E. coli} in global context", out.extra='', fig.cap="Midpoint rooted maximum likelihood phylogenetic tree of included study \\textit{E. coli} placed in the context of aglobal collection of genomes, showing phylogroups, source sample type and continent of isolation (coloured bars). Dark grey bars indicate isolates from this study or isolates with ESBL gene presence, as labelled (this study or ESBL, respectively). Three most frequently isolated STs in the current study (131, 410 and 167) labelled.  Bootstrap support of less than 90\\% is indicated by a black circle at a given node. Scale bar indicates 0.009 SNPs/site.", fig.height = 10, fig.width=8}


tree <- read.tree(paste0(wd, "global_tree/all_ESCO_core_snps.aln.treefile"))

midpoint.root(tree) -> tree

#metadata

metadata <- read_csv(paste0(wd, "global_tree/all_ESCO_metadata.csv"))
sub("#", "_", metadata$Lane.Name ) -> metadata$Lane.Name
metadata <- data.frame(metadata)
rownames(metadata) <- metadata$Lane.Name

metadata$Continent <- NA
metadata$Continent[metadata$Country == "Argentina" |
                     metadata$Country == "Bolivia" |
                     metadata$Country == "Guatemala" |
                     metadata$Country == "Mexico" |
                     metadata$Country == "Venezuela" ] <- "Z_S. America"

metadata$Continent[metadata$Country == "Bangladesh" | 
                     metadata$Country == "Burma" |
                     metadata$Country == "China" |
                     metadata$Country == "India" |
                     metadata$Country == "Indonesia" |
                     metadata$Country == "Japan" |
                     metadata$Country == "Nepal" |
                     metadata$Country == "Pakistan" |
                     metadata$Country == "thailand" |
                     metadata$Country == "Thailand"] <- "Z_Asia"

metadata$Continent[metadata$Country == "Egypt" | 
                     metadata$Country == "Gambia" |
                     metadata$Country == "Guinea Bissau" |
                     metadata$Country == "Kenya" |
                     metadata$Country == "Malawi" |
                     metadata$Country == "Mali" |
                     metadata$Country == "Morocco" |
                     metadata$Country == "Mozambique" |
                     metadata$Country == "Tunisia" |
                     metadata$Country == "Zaire"] <- "Z_Africa"

metadata$source.cat <- metadata$Source

metadata$source.cat[metadata$source.cat == "Canal" |
                      metadata$source.cat == "Farm" |
                      metadata$source.cat == "Untreated hospital sewage"] <- "_Environment"

metadata$source.cat[metadata$source.cat == "Blood" |
                      metadata$source.cat == "CSF"] <- "_Clinical - _Blood/CSF"

metadata$source.cat[metadata$source.cat == "Urine" |
                      metadata$source.cat == "Pus" |
                      metadata$source.cat == "Sputum" ] <- "_Clinical - Other"

metadata$source.cat[metadata$source.cat == "stool" |
                      metadata$source.cat == "Stool" |
                      metadata$source.cat == "RS"] <- "_Clinical - _Stool"

metadata$malawi <- as.factor(as.numeric(metadata$Country == "Malawi"))

metadata$this_study <- as.factor(as.numeric(grepl("DASSIM", metadata$study)))
# pgroups

pgroups <- read.csv(paste0(wd, "global_tree/phylogroups_and_mlst/all_ESCO_pgroups.csv"), stringsAsFactors = F)
pgroups.d <- read.csv(paste0(wd, "phylogroup_and_mlst/phylogroups.csv"), stringsAsFactors = F)
pgroups <- select(pgroups, V1, phylogroup)
names(pgroups) <- c("Lane", "Phylogroup") 
pgroups <- rbind(pgroups, pgroups.d)
rownames(pgroups) <- pgroups$Lane
pgroups$Phylogroup[pgroups$Phylogroup == "Clade I"] <- "Clade I or II"

# mlst


## load mlst and plot

mlst.d <- read.csv(paste0(wd, "phylogroup_and_mlst/mlst.csv"), stringsAsFactors = F)
mlst <- read.csv(paste0(wd, "global_tree/phylogroups_and_mlst/mlst_summary.csv"), stringsAsFactors = F)
mlst <- select(mlst, lane, ST)

mlst <- rbind(mlst.d, mlst)
dcast(mlst, lane ~ ST, fun.aggregate = base::length) -> mlst.onehot
#mlst.onehot <- subset(mlst.onehot, lane %in% tree$tip.label)

rownames(mlst.onehot) <- mlst.onehot$lane
mlst.onehot <- select(mlst.onehot, -lane)
n.mlsts <- sort(apply(mlst.onehot,2, sum), decreasing = T)
mlst.onehot[names(n.mlsts[n.mlsts > 1])] -> mlst.onehot

apply(mlst.onehot, 1, sum) == 0 -> mlst.onehot$other
mlst.onehot$other<- as.numeric(mlst.onehot$other)
mlst.onehot[mlst.onehot == 0] <- "Z_No"
mlst.onehot[mlst.onehot == 1] <- "Z_Yes"

#### AMR

srst2 <- read.csv(paste0(wd, "global_tree/srst2/srst2_summary_cleaned_pres_abs.csv"), stringsAsFactors = F)
srst2.d <- read.csv(paste0(wd, "amr/srst2_pres_abs.csv"), stringsAsFactors = F)
bind_rows(srst2, srst2.d) -> srst
srst[is.na(srst)] <- 0
rownames(srst) <- srst$name

narrow_spec <- c("TEM_1", "TEM_95", "TEM_135", "TEM_90",
                 "OXA_1", 
                 "SHV_1")

esbl <- c(names(srst)[grepl("CTX", names(srst))], 
          names(srst)[grepl("TEM", names(srst))],
          names(srst)[grepl("OXA", names(srst))],
          names(srst)[grepl("SHV", names(srst))],
          names(srst)[grepl("NDM", names(srst))])

esbl <- esbl[!(esbl %in% narrow_spec)]
data.frame(esbl = apply(srst[esbl], 1,sum)) -> esbl.cat
esbl.cat$esbl[esbl.cat$esbl > 0] <- 1
esbl.cat$Lane <- rownames(esbl.cat)
esbl.cat$esbl <- as.factor(esbl.cat$esbl)
##

col <- c(hue_pal()(8), "white", "grey30", brewer.pal(4,"Set1"),brewer.pal(4,"Dark2") )
names(col) <- c("A","B1","B2","C","Clade I or II", "D","F", "Unknown",
                "0", "1","_Clinical - _Blood/CSF", "_Clinical - _Stool", "_Environment",  "_Clinical - Other",
                "Z_Africa", "Z_Asia", "Z_S. America"
               )

names(metadata)[names(metadata) == "this_study"] <- "This study"
names(esbl.cat)[names(esbl.cat) == "esbl"] <- "ESBL"
names(metadata)[names(metadata) == "Source"] <- "Source1"
names(metadata)[names(metadata) == "source.cat"] <- "Source"

ggtree(tree, size = 0.3) %>% 
  gheatmap(select(metadata, `This study`), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80,offset = 0.006 ) %>%
  gheatmap(select(pgroups, Phylogroup), width = 0.05, 
           color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.001)  %>%
  gheatmap(select(esbl.cat, ESBL), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.011 ) %>%
  gheatmap(select(metadata, Source), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.016 ) %>%
  gheatmap(select(metadata, Continent), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.021 ) +
 # gheatmap(select(metadata, malawi), width = 0.04, color = NA, font.size = 4, colnames_angle = 90,
      #     colnames_position = "top", colnames_offset_y = 80, offset = 0.025 )+ 
  scale_fill_manual(values =col) + theme(legend.position = "none") +
  geom_treescale(x =0.02, y = 1100, offset = 5)  +   
  geom_point2(aes(subset = as.numeric(label) < 90 & !isTip), size = 0.3)+ 
  geom_cladelabel(node = 1351, label = "ST131", fontsize = 2.5) + 
  geom_cladelabel(node = 1864, label = "ST410", fontsize = 2.5) +
  geom_cladelabel(node = 2020, label = "ST167", fontsize = 2.5) -> p #, labels = c("0", "1","A","B1","B2","C","Clade I or II", "D","F", 
    
ggtree(tree) %>% 

  gheatmap(select(pgroups, Phylogroup), width = 0.05, 
           color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80) +
  scale_fill_manual(values =col) + labs(fill = "Phylogroup") + theme(legend.title = element_text()) -> p1x

leg.pgroup <- get_legend(p1x)

ggtree(tree) %>% 
  
  gheatmap(select(metadata, Source), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.015 ) +
  scale_fill_manual(values =col, labels = c("Clinical: Blood/CSF", "Clinical: Stool", "Clinicial: Other", "Environment", "")) + labs(fill = "Source") + theme(legend.title = element_text())-> p1x
  
leg.source <- get_legend(p1x)

ggtree(tree) %>% 
  
  gheatmap(select(metadata, Continent), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.020 ) +
  scale_fill_manual(values =col, labels = c("Africa", "Asia", "S. America","")) + labs(fill = "Continent") + theme(legend.title = element_text())-> p1x

leg.continent <- get_legend(p1x)

ggarrange(p, 
          ggarrange(NULL, leg.pgroup, leg.source, leg.continent, NULL, ncol = 1, nrow = 5, heights = c(1,1,0.7,0.4,1)),
          ncol = 2, nrow = 1, widths = c(3,1))

```

```{r wgs-131vs410-tree, echo = F, warning = F, message = F, fig.scap="Phylogeny of ST131, ST410 and ST167 \\textit{E. coli}.", out.extra='', fig.cap="Subtree of ST410 (A, left) and ST131 (B, right), and ST167 (C, bottom) showing multiple introductions of ST131 into Malawi, in comparison to a single introduction of clonal ST410 and ST167 clades. Colour of tree tip label indicates isolation from this study (red) or other studies (black), and coloured heatmap indicates country of isolation. Note that the scale bar in C is an order of magnitude different from A and B.", fig.height = 7, fig.width=6}

treeio::tree_subset(tree, 1864, levels_back = 0) -> st410tree

select(metadata, Lane.Name, `This study`) -> this.study
this.study$`This study` <- as.character(this.study$`This study`)
this.study$`This study`[this.study$`This study` == 1] <- "Yes"
this.study$`This study`[this.study$`This study` == 0] <- "No"

#colz <- c( "grey11", "grey89", brewer.pal(name= "Set3", n = 7)[1:5] )

metadata$Country[metadata$Country == "thailand"] <- "Thailand"

colz <- c( "grey11", "grey89", hue_pal()(5))
names(colz) <- c( "1","0","Malawi", "Guatemala",  "Thailand" , "Mexico", "Bolivia")
ggtree(st410tree) %<+% this.study   %>%
  #gheatmap(select(esbl.cat, ESBL), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
   #        colnames_position = "top", offset =  0.0003) %>%
  gheatmap(select(metadata, Country), width = 0.05, color = NA, colnames = F,font.size = 4, colnames_angle = 90,
           colnames_position = "top", offset = 0.0003 , colnames_offset_y = 2) + 
  geom_tiplab(size = 1, align = T, aes(color = `This study`)) + 
  geom_treescale(x = 0.0005, y = 30, width = 0.0003) + scale_fill_manual(values = colz) +
  scale_color_manual(values = c("black", hue_pal()(1))) + theme(legend.position = "none") -> a


data.frame(country = c("Malawi", "Guatemala",  "Thailand" , "Mexico", "Bolivia"), n = c(1,2,3,4,5)) -> fudge
fudge$country <- factor(fudge$country, levels = c("Malawi", "Thailand", "Guatemala", "Mexico", "Bolivia"))
ggplot(fudge, aes(country, n, fill = country)) + geom_col()+ scale_fill_manual(values = colz) + 
  theme_bw() + theme(legend.title = element_blank(), legend.position = "bottom")-> fudgeplot

leg <- get_legend(fudgeplot)

## ST131 is node 1351

treeio::tree_subset(tree, 1351, levels_back = 0) -> st131tree

ggtree(st131tree) %<+% select(metadata, Lane.Name, `This study`)   %>%
 # gheatmap(select(esbl.cat, ESBL), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
  #         colnames_position = "top", offset =  0.0002) %>%
  gheatmap(select(metadata, Country), width = 0.05, color = NA, colnames = F,font.size = 4, colnames_angle = 90,
           colnames_position = "top", offset = 0.0004, colnames_offset_y = 5) + 
  geom_tiplab(size = 1, align = T, aes(color = `This study`)) + geom_treescale(x = 0.0005, y = 57,offset = 0.5 )+ scale_fill_manual(values = colz) +
  scale_color_manual(values = c("black", hue_pal()(1)))  + theme(legend.position = "none") -> b

treeio::tree_subset(tree, 2020, levels_back = 0) -> st167tree


ggtree(st167tree) %<+% select(metadata, Lane.Name, `This study`)   %>%
  # gheatmap(select(esbl.cat, ESBL), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
  #         colnames_position = "top", offset =  0.0002) %>%
  gheatmap(select(metadata, Country), width = 0.05, color = NA, colnames = F,font.size = 4, colnames_angle = 90,
           colnames_position = "top", offset = 0.00005, colnames_offset_y = 5) + 
  geom_tiplab(size = 1, align = T, aes(color = `This study`)) + geom_treescale(x = 0.00005, y = 20, width = 0.00003 )+ scale_fill_manual(values = colz) +
  scale_color_manual(values = c( hue_pal()(1)))  + theme(legend.position = "none") -> c

ggarrange(ggarrange(a,b, ncol = 2, nrow =1, labels = c("A: ST410", "B: ST131") ),
          ggarrange(NULL,c,NULL, ncol =3, nrow = 1, widths = c(0.5,1,0.5), labels = c(NA,"C: ST167",NA)),
          ggarrange(NULL,leg,NULL, widths = c(1,0.5,1), ncol = 3),
          heights = c(1,1, 0.1), ncol = 1, nrow = 3)


```

### Antimicrobial resistance determinants

All identified AMR genes are shown in Figure \@ref(fig:wgs-amr-sum)A, alongside a summary of number of isolates with resistance mutations to given antimicrobial classes (Figure \@ref(fig:wgs-amr-sum)B) and the phenotypic resistance of the isolates for which phenotypic antimicrobial resistance testing was carried out (449/473 [95%]). The isolates contained a median (IQR) of 16 (12-17) resistance genes, and 100 different resistance alleles were identified in total. A description of resistance gene by class is given below. In this section I consider and present data only from the isolates from this study, rather than data from the context genomes presented above.

#### $\beta$-lactam resistance

All isolates contained at least one gene that conferred resistance to third-generation cephalosporins, either an ESBL gene (n= 472) or a carbapenemase (n=1). The majority of ESBL-gene containing isolates contained only one ESBL gene (459/472 [97%]); fewer contained 2 (13/472 [3%]) and none contained more than 2. The  $bla_{CTX-M}$ family was most commonly represented in this collection, and 319/473 (67%) of isolates contained $bla_{CTXM-15}$. (26/473 [5%] of isolates) contained $bla_{SHV}$  genes. The $\beta$-lactamases $bla_{TEM}$ and $bla_{OXA}$ were very common, most commonly the penicillinases $bla_{OXA-1}$ and $bla_{TEM-1}$ in 186/473 [39%] and 289/473 [61%] of isolates respectively. Plasmid-mediated $bla_{ampC}$ genes were identified in 45/473 (9%) of isolates, almost all (44/45) $bla_{CMY-94}$, which was lineage-restricted to the ST410 isolates. Presence of $bla_{ampC}$ is unexpected as all of these isolates were confirmed to be ESBL-producers by combination disc testing. This testing uses cephalosporin-containing discs both with and without clavulanic acid, and confirms EBSL production by a difference in zone size between these discs, as ESBL enzymes are inactivated by clavulanic acid. However, the cephalosporins used in this test are likely to be hydrolysed by $ampC$ enzymes, and if these isolates were producing such enzymes it could confer cephalosporin resistance regardless of the presence or absence of clavulanic acid. This was not the case for any of these isolates; none of them hydrolysed the cephalosporins used in the presence of clavulanic acid. It may be that the $bla_{CMY}$ genes were not expressed or have been inactivated in the ST410 clade.

The single carbapenemase gene identified was s $bla_{NDM-5}$; the isolate harbouring this gene was recovered from the  stool of a 67-year old man with no history of foreign travel nor hospitalisation. He had been admitted to the hospital with fever seven days previously and treated with seven days of intravenous ceftriaxone for sepsis, the source of which was not clear. He made an uneventful recovery, and no carbapenemase-containing isolate was recovered from his stool at any other time. The $bla_{NDM-5}$ gene was carried on a partially assembled IncX3 plasmid. BLAST of this assembly against the NCBI database showed that this contig had 99% sequence identity with a previously sequenced pNDM-MGR194 46.2 kbp blaNDM-5 containing Inc-X3 plasmid found in India between 2011-13[@Krishnaraju2015]. We fully assembled the plasmid by mapping reads back to pNDM-MGR194 with Burrows-Wheeler alignment and found it to be extremely similar, with only 13 SNPs compared to pNDM-MGR194.

```{r wgs-amr-sum, echo = F, warning = F, message = F, fig.scap="Frequency distribution of AMR genes", out.extra='', fig.cap="A: Frequency distribution of AMR genes identified in isolates. Class of antimicrobial to which gene confers resistance is shown. B: Number of isolates with any mutation to a given class. Any mutation that could possibly confer resistance to a given class is included, including any mutation in the QRDR for quinolones. C: Phenotypic resistance patterns for subset of samples in this analysis that also underwent phenotypic testing (n = 449)", fig.height = 10, fig.width=8}

m.c <- read.csv(paste0(wd, "amr/srst2_pres_abs.csv"), stringsAsFactors = F)
m.c <- subset(m.c, name %in% tree$tip.label)

q.res <- read.csv(paste0(wd, "amr/qrdr_mut_presence.csv"), stringsAsFactors = F)
amr <- merge(m.c, q.res, by.x = "name", by.y = "sample")
amr.melt <- melt(amr) %>% subset(value > 0)

amr.melt$class <- NA
amr.melt$class[grepl("Par", amr.melt$variable) | 
                 grepl("Gyr", amr.melt$variable) | 
                 grepl("Par", amr.melt$variable) |
                 grepl("Qnr", amr.melt$variable) |
                 grepl("Qep", amr.melt$variable) |
                 grepl("Oqx", amr.melt$variable) |
                 grepl("Nor", amr.melt$variable)] <- "Quinolone"

amr.melt$class[grepl("Tet", amr.melt$variable) ] <- "Tetracyclines"

narrow_spec <- c("TEM_1", "TEM_95", "TEM_135", "TEM_90",
                 "OXA_1", 
                 "SHV_1")

amr.melt$class[amr.melt$variable %in% esbl] <- "ESBL"
amr.melt$class[amr.melt$variable %in% narrow_spec] <- "Penicillins"

amr.melt$class[grepl("Sul", amr.melt$variable) ] <- "Sulphonamides"

amr.melt$class[grepl("Str", amr.melt$variable) | 
                 grepl("Aad", amr.melt$variable) | 
                 grepl("Aac", amr.melt$variable) |
                 grepl("Aph", amr.melt$variable) |
                 grepl("Rmt", amr.melt$variable) |
                 grepl("APH", amr.melt$variable)] <- "Aminoglycosides"

amr.melt$class[grepl("Sat", amr.melt$variable) ] <- "Streptothricin"

amr.melt$class[grepl("LEN", amr.melt$variable) ] <- "Penicillins"
amr.melt$class[grepl("SCO", amr.melt$variable)  |
                grepl("BlaZ", amr.melt$variable) |
                 grepl("CARB", amr.melt$variable) |
                 grepl("PBP", amr.melt$variable) ] <- "Penicillins"

amr.melt$class[grepl("NDM", amr.melt$variable) ] <- "Carbapenems"

amr.melt <- subset(amr.melt, variable != "MrdA")
amr.melt <- subset(amr.melt, variable != "LAP_2")

amr.melt$class[grepl("Mph", amr.melt$variable) ] <- "Macrolides"
amr.melt$class[grepl("Mef", amr.melt$variable) |
                grepl("Erm", amr.melt$variable)  ] <- "Macrolides"

amr.melt$class[grepl("Fos", amr.melt$variable) ] <- "Fosfomycin"

amr.melt$class[grepl("Cat", amr.melt$variable) | 
                 grepl("FloR", amr.melt$variable) |
                 grepl("Cml", amr.melt$variable) ] <- "Chloramphenicol"

amr.melt$class[grepl("Dha", amr.melt$variable) | 
                 grepl("CMY", amr.melt$variable) ] <- "AmpC"

amr.melt$class[grepl("Dfr", amr.melt$variable) ] <- "Trimethoprim"


amr.melt$class[grepl("Arr", amr.melt$variable) ] <- "Rifampicin"

amr.melt <- subset(amr.melt, !(grepl("Amp", variable) | grepl("AMP", variable)))

classlevels <- amr.melt %>% group_by(variable, class) %>% summarise(n = n())

temp <- c("Penicillins", "ESBL", "AmpC", "Carbapenems")


classlevels$class <- factor(classlevels$class, levels = rev(c(temp, unique(amr.melt$class)[!(unique(amr.melt$class) %in% temp)])))

amr.melt$variable<- factor(amr.melt$variable, levels = classlevels$variable[order(classlevels$class, classlevels$n)])



col <- c(brewer.pal(8,"Dark2"),brewer.pal(8,"Dark2") )

data.frame((table(classlevels$class))) -> linebreaks
linebreaks$cumsum <- cumsum(linebreaks$Freq)

start <- list()
end <- list()
label <- list()

for (i in 1:nrow(linebreaks)) {
  if (i == 1) {
    start[[i]] <- 1
    end[[i]] <- linebreaks$cumsum[i]
    label[[i]] <- as.character(linebreaks$Var1[i])
  } else {
    start[[i]] <- linebreaks$cumsum[i-1] + 1
    end[[i]] <- linebreaks$cumsum[i]
    label[[i]] <- as.character(linebreaks$Var1[i])
  }
}

do.call(rbind, start) -> start
do.call(rbind, end) -> end
do.call(rbind, label) -> label

data.frame(start = start, end = end, label = label) -> pos

pos$text.y <- pos$start + (pos$end - pos$start)/2

pos$start == pos$end -> chflag

pos$start[chflag] <- pos$start[chflag] - 0.2
pos$end[chflag] <- pos$end[chflag] + 0.2


ggplot(amr.melt, aes(variable, fill = class)) + geom_bar()+ theme_bw() + theme(axis.text.y = element_text(size = 6), legend.position = "none", plot.margin = unit(c(0.2,0.2,0.2,3.5), "cm"), axis.title = element_blank() )   + coord_flip(ylim = c(-5,473),clip = "off", expand = FALSE) +  annotate(geom = "segment", x = pos$start, xend = pos$end, y = -100, yend = -100 )  + annotate(geom = "text", y = -110, x = pos$text.y, label = pos$label, size = 3, hjust = 1) -> p1

amr.class <- unique(select(amr.melt, name, class))
amr.class$class <- factor(amr.class$class, levels = rev(levels(classlevels$class)))

ggplot(amr.class, aes(class, fill = as.character(class))) + geom_bar() +theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position = "none", axis.title = element_blank()) + geom_text(x = 4,y = 30, label = "n=1", size = 2) + geom_text(x = 6,y = 30, label = "n=2", size = 2) -> p2



# load phenotypic data

orgs <- read.csv("/Users/joelewis/Documents/PhD/Data/ESBL/ESBL_orgs.csv", stringsAsFactors = F)
orgs <- subset(orgs,profile_name == "DASSIM Culture")

orgs <- orgs[c(1,4,5:10)]

melt(orgs, id.vars = c('sample_number', 'organism')) -> orgs
subset(orgs, value != "") -> orgs
subset(orgs, organism == "Escherichia coli" ) -> orgs


#orgs$value <- factor(orgs$value, levels = c("Resistant", "Sensitive", "Intermediate" ))
orgs$variable <- map_chr(as.character(orgs$variable), function(x) strsplit(x , "\\.")[[1]][[1]])



sub("#","_", sample_ids$Lane) -> sample_ids$Lane
subset(sample_ids, Lane %in% tree$tip.label) -> sample_ids
orgs <- subset(orgs, sample_number %in% sample_ids$Supplier.Name)
orgsall <- orgs
subset(orgs, value != "Sensitive") -> orgs

rbind(orgs, 
      data.frame(sample_number = unique(orgs$sample_number),
                 organism = rep("Escherichia coli", base::length(unique(orgs$sample_number))),
                 variable = rep("ESBL", base::length(unique(orgs$sample_number))),
                 value = rep("Resistant", length(unique(orgs$sample_number)))
      )
      )-> orgs
                 
                                
cols <- hue_pal()(14)

orgs$variable <- factor(orgs$variable, 
                        levels = c("ESBL", "Meropenam", "Amikacin", "Gentamicin",
                                   "Chloramphenicol", "Cotrimoxazole", "Ciprofloxacin") )
ggplot(orgs, aes(variable, fill =variable)) + geom_bar() + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position  = "none", axis.title = element_blank()) + 
  scale_fill_manual(values = cols[c(5,3,1,1,4,14,8)]) -> p3

ggarrange(p1, ggarrange(NULL,p2, p3,NULL, ncol = 1, nrow = 4, labels =c(NA,"B","C", NA), heights = c(0.5,1,1,0.5)), ncol = 2, nrow = 1, widths = c(1.5,1), labels = c("A", NA))

```

#### Quinolone resistance

Quinolone resistance can be mediated by a number of different mechanisms and at least one known determinant of quinolone resistance was present in 88% (418/473) isolates. 108/473 (23%) of isolates contained one of the the plasmid-mediated quinolone resistance (PMQR) genes $qnrS$,$qnrB$ or $qepA$; 172/473 (36%) carried the acetyltransferase _aac(6')-Ib-cr_ which can hydrolyse fluoroquinolones with an amino nitrogen on the piperazinyl ring, such as ciprofloxacin and norfloxacin[@Robicsek2006]. Nonsynonymous mutations were identified in at least one of the DNA gyrase subunits $gyrA$ or $gyrB$ or topoisomerase IV subunits $parC$ or $parE$ in 349/449 (78%) of isolates. The majority of these mutations were well-described QRDR mutations (codon 83 and 87 in $gyrA$, codon 80 and 84 in $parC$ and codon 458 in $parE$[@Yan2017], Figure \@ref(fig:wgs-amr-quin)A), and they tended to cluster together (Figure \@ref(fig:wgs-amr-quin)B). Mutations in $gyrB$ were very unusual, with only 3 identified in the dataset.

The association between phenotypic and genotypic resistance is complex but some patterns seem clear (Figure \@ref(fig:wgs-amr-quin)B). The well recognised constellation of mutations in $gyrA$ at codon 83 and 87 and codon 80 of $parC$ were strongly associated with phenotypic ciprofloxacin resistance (OR 7.3 [95% CI 4.4-12.4], p < 0.001 by Fisher's exact test), but $gyrA$ codon 83 mutations alone seemed insufficient: of the 63 samples with this mutation alone and available AST data only 30% (19/63) showed phenotypic resistance. Similarly, presence of the $qnrS$ plasmid-mediated gene seemed insufficient alone to confer phenotypic resistance: of 57 samples in which this gene was present and AST data were available, only 25/57 (43%) had phenotypic ciprofloxacin resistance by disc diffusion testing.

```{r wgs-quin-setup, include = F}

read_tsv(paste0(wd, "amr/QRDR_ariba_output.tsv")) -> q.df




names(q.df)[length(names(q.df))] <- "sample"

subset(q.df, ref_name != "ref_name") -> q.df

sub("#", "_", q.df$sample) -> q.df$sample
subset(q.df, sample %in% tree$tip.label) -> q.df

select(q.df, sample, ref_name, ref_ctg_change) -> q.df

q.df$codon_posn <- str_sub(q.df$ref_ctg_change,2,-2)

q.df$wt_codon <-  str_sub(q.df$ref_ctg_change,1,1)

q.df$mut_codon <- str_sub(q.df$ref_ctg_change,-1,-1)

q.df$codon_posn <- as.numeric(q.df$codon_posn)

q.df <- subset(q.df, (ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108) |
                 (ref_name == "GyrB" & codon_posn >= 331 & codon_posn <= 480) |
                (ref_name == "ParC" & codon_posn >= 38 & codon_posn <= 169) |
                (ref_name == "ParE" & codon_posn >= 365 & codon_posn <= 525) )

df.gyra <- subset(q.df, ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108)

q.df %>% dplyr::group_by(ref_name, codon_posn) %>% dplyr::summarise(n=n()) -> df.sum

df.gyra <- subset(df.sum, ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108)
df.gyrb <- subset(df.sum, ref_name == "GyrB" & codon_posn >= 331 & codon_posn <= 480)
df.parc <- subset(df.sum, ref_name == "ParC" & codon_posn >= 38 & codon_posn <= 169)
df.pare <- subset(df.sum, ref_name == "ParE" & codon_posn >= 365 & codon_posn <= 525)


#ggplot(df.gyra, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(67,108)) +
#  theme_bw() + ylim(c(0,400)) + ggtitle("GyrA") + xlab("Codon position")-> p1
  
#ggplot(df.gyrb, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(331,480)) +
  #theme_bw() + ylim(c(0,400)) + ggtitle("GyrB")  + xlab("Codon position") -> p2
     
#ggplot(df.parc, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(38,169)) +
  #theme_bw() + ylim(c(0,400)) + ggtitle("ParC")  + xlab("Codon position")-> p3

#ggplot(df.pare, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(365,525)) +
 # theme_bw()  + ylim(c(0,400)) + ggtitle("ParE") + xlab("Codon position") -> p4

ggplot(q.df, aes(fct_rev(fct_infreq(ref_ctg_change)))) + geom_bar() + coord_flip() + facet_wrap(~ref_name, scales = "free") + xlab("Mutation") + ylab("Count") + theme_bw() -> p1

orgs.cip <- subset(orgsall, variable == "Ciprofloxacin")

orgs.cip <- merge( select(sample_ids, Supplier.Name, Lane),select(orgs.cip,sample_number, value),   by.x = "Supplier.Name", by.y = "sample_number")

orgs.cip <- merge(orgs.cip, q.res, by.x = "Lane", by.y = "sample", al)

q.df$key <- paste0(q.df$ref_name, q.df$codon_posn)

q.df.new <- subset(q.df, key == "GyrA83" |  key == "GyrA87" | key == "ParC80" | key == "ParE458")

dcast(q.df.new,sample ~ key, fun.aggregate = base::length) -> q.df.new

orgs.cip <- merge(orgs.cip, q.df.new, by.x = "Lane", by.y = "sample", all.x = T)
orgs.cip[is.na(orgs.cip)] <- 0
orgs.cip$value[orgs.cip$value == "Resistant" |orgs.cip$value == "Intermediate" ] <- 1
orgs.cip$value[orgs.cip$value == "Sensitive" ] <- 0
orgs.cip$value <- as.numeric(orgs.cip$value)

pmqr <- m.c[ c("name", grep("Qnr", names(m.c), value = T) , grep("Qep", names(m.c), value = T) , grep("Aac6Ib_cr", names(m.c), value = T))]

pmqr$QnrS <- apply(pmqr[c(grep("QnrS", names(pmqr), value = TRUE ), grep("Qnr_S", names(pmqr), value = TRUE ))], 1, sum)
pmqr$QnrB <- apply(pmqr[grep("QnrB", names(pmqr), value = TRUE )], 1, sum)

pmqr$QnrS[pmqr$QnrS > 1] <- 1
pmqr$QnrB[pmqr$QnrB > 1] <- 1

orgs.cip <- merge(orgs.cip, select(pmqr, name, QnrS,QnrB, QepA, Aac6Ib_cr), by.x = "Lane", by.y = "name")
names(orgs.cip)[names(orgs.cip) == "value"] <- "CIP R on AST"


#names(orgs.cip)[names(orgs.cip) == "Quinolone resistance"] <- "CIP resistance"
#dev.off()
grid.grabExpr(pheatmap(orgs.cip[-c(1,2,4:7)], show_rownames = FALSE,
                       legend = FALSE)) -> hm
hm <- as_ggplot(hm)
#dev.off()

#ggarrange(p1, hm)

ggarrange(p1, ggarrange(NULL,hm,NULL, widths = c(0.2,1,0.2), ncol = 3, nrow = 1), ncol =1, nrow = 2, labels = c("A","B")) -> parrange

ggsave(paste0(wd,"quinolone_plot.pdf"), plot = parrange, width = 5, height = 8, units = "in")

```

```{r wgs-amr-quin, echo = F, warning = F, message = F, fig.scap="Quinolone resistance mutations and phenotypic resistance", out.extra='', fig.cap="A: Mutations in quinolone resistance-determining regions, showing that most mutations are well-recognised (see text for details) B: Row and column clustered heatmap of commonest QRDR mutations (\\textit{gyrA83}, \\textit{gyrA87}, \\textit{parC80} or \\textit{parE458}), plamid-mediated quinolone resistance mutations (\\textit{qnr}, \\textit{qep} or \\textit{aac(6\\`)-Ib-cr)} and phenotypic resistance. The constellation of \\textit{gyrA83}, \\textit{gyrA87} and \\textit{parC80} is strongly associated with phenotypic resistance, but \\textit{gyrA83} or \\textit{qnrS} alone are insufficient to confer resistance. Each row is one sample, red = presence, blue = absence. CIP R on AST indicates those samples which are show phenotypic ciprofloxacin resistance on antimicrobial sensitivity testing (AST). Those isolates without AST data are excluded from this heatmap. ", out.height = '80%', fig.align='center'}

knitr::include_graphics(paste0(wd,"quinolone_plot.pdf"))
```

#### Aminoglycoside resistance

The identified aminoglycoside resistance determinants are shown in Figure \@ref(fig:wgs-amr-aminogly). Most aminoglycoside resistance genes are classified into three families based on their ability to acetylate, phosphorylate, or adenylate amino or hydroxyl groups found at various positions around the aminoglycoside molecule, and are called acetyltransferases (AACs), nucleotidyltranferases (ANTs), or phosphotransferases (APHs)[@Ramirez2010]. They usually further identified by the 
site of action in terms of the aminoglycoside carbon atom upon which they act, subclass, and individual identifier. For example, the enzyme AAC(3)-Ib (gene _aac(3)-Ib_) refers to an acetyltransferases acting at position 3 of subclass I and individual enzyme identifier b. All enzymes of a given class and subclass would be expected to confer similar resistance. There is also a second nomenclature, where genes are referred to _aac_, _aad_ (instead of ANT) or _aph_, along with a letter to indicate side of modification; so _aadA_, for example, is equivalent to ANT(3")-Ia[@Ramirez2010]. There are some genes that confer aminoglycoside resistance in different ways and are not included in this classification: the streptomycin resistance genes _strA_ and _strB_ and the 16S rRNA methylase _rmtB_ that confers resistance to amikacin, gentamicin and gentamicin[@Galimand2003].

Aminoglycoside resistance genes were very common in the sequenced isolates, with 469/473 (99%) of isolates containing at least one aminoglycoside resistance gene, and most containing multiple different genes: median number of aminoglycoside resistance genes per isolate was 4 (IQR 3-5). Despite streptomycin being absent from all Malawian treatment guidelines save for retreatment of tuberculosis, the streptomycin resistance genes $strA$ and $strB$ were near ubiquitous (Figure \@ref(fig:wgs-amr-aminogly)). Genes encoding the AAC and ANT enzyme families were commonly seen: genes encoding ANT family enzymes were all _aadA_ alleles (alternately known as _aad(3")-Ia_) which would be expected to also confer streptomycin resistance. Identified genes encoding AAC family enzymes were _aac(3)-Ib_ and _aac(3)-IIa_, both of which have been associated with gentamicin resistance, as has _aac(6')-Ib-cr_ which also can confer resistance to quinolones. Genes encoding APH family enzymes were unusual. Four were identified, and were all _aph(3')-Ia_, which has been associated with amikacin and kanamycin resistance. One $rmtB$ gene was identified, which again has been associated with amikacin resistance.

Genes of the _aac_ family tended to co-occur (Figure \@ref(fig:wgs-amr-aminogly)B) and presence of any _aac_ family gene was strongly associated with gentamicin resistance (OR 9.3 [95% CI 6.0-14.8], p < 0.001). Of the five isolates that were resistant to amikacin on AST, two contained _aac_ family genes, five contained _aad_ family genes and none contained _rmtB_ or _aph_ genes.

```{r wgs-amr-aminogly-setup, include = F}

amr.melt %>% filter(class == "Aminoglycosides") %>% group_by(variable) %>% summarise(n= n()) -> ag

ag$agent <- NA 
ag$agent[grepl("Str", ag$variable) | grepl("AadA", ag$variable)] <- "Streptomycin"

ag$agent[grepl("Aac6", ag$variable) | grepl("Aac3_II", ag$variable)] <- "Gentamicin"

ag$agent[grepl("Aph", ag$variable)] <- "Kanamycin"
ag$agent[grepl("Rmt", ag$variable)] <- "Amikacin"

ag$agent <- factor(ag$agent, levels = c("Streptomycin", "Gentamicin", "Amikacin","Kanamycin"))

ag$variable <- factor(ag$variable, levels = (ag$variable[order(ag$agent, ag$n)]))
ggplot(ag, aes(x= variable, y =n , fill = agent)) + geom_col() + theme_bw() + theme(axis.text.x = element_text(angle = -90), legend.title = element_blank(), legend.position = "bottom") + labs(x= "Aminoglycoside resistance gene") -> p1

orgsall %>% filter(variable == "Amikacin" | variable == "Gentamicin") %>% dcast(sample_number ~ variable, value.var = "value") -> ag.sens

ag.sens <- merge(select(sample_ids, Supplier.Name, Lane), ag.sens, by.x = "Supplier.Name", by.y = "sample_number")

m.c[c("name", grep("Aac", names(m.c), value = TRUE),grep("Aad", names(m.c), value = TRUE),grep("Aph", names(m.c), value = TRUE),
  grep("Rmt", names(m.c), value = TRUE),
  grep("Str", names(m.c), value = TRUE))  ] -> amg

ag.sens$Gentamicin[ag.sens$Gentamicin != "Sensitive"] <- 1
ag.sens$Gentamicin[ag.sens$Gentamicin == "Sensitive"] <- 0
ag.sens$Gentamicin <- as.numeric(ag.sens$Gentamicin)

ag.sens$Amikacin[ag.sens$Amikacin != "Sensitive"] <- 1
ag.sens$Amikacin[ag.sens$Amikacin == "Sensitive"] <- 0
ag.sens$Amikacin <- as.numeric(ag.sens$Amikacin)

amg <- merge(amg, select(ag.sens, Lane, Amikacin, Gentamicin), all.x = T, by.x = "name", by.y = "Lane")

amg <- amg[complete.cases(amg),]

names(amg)[names(amg) == "Amikacin"] <- "AMK R on AST"
names(amg)[names(amg) == "Gentamicin"] <- "GEN R on AST"

grid.grabExpr( pheatmap(amg[-1], legend = F, show_rownames = F)) -> hm
hm <- as_ggplot(hm)

ggarrange(p1, ggarrange(NULL,hm,NULL, widths = c(0.2,1,0.2), ncol = 3, nrow = 1), ncol= 1, nrow = 2,labels  = c("A", "B")) -> parrange

 #sens
#binom.test(summ$TP, (summ$TP + summ$FN))
# spec
#binom.test(summ$TN, (summ$TN + summ$FP))

ggsave(paste0(wd,"aminoglycoside_plot.pdf"), plot = parrange, width = 5, height = 8, units = "in")

```

```{r wgs-amr-aminogly, echo = F, warning = F, message = F, fig.scap="Aminoglycoside resistance mutations and phenotypic resistance", out.extra='', fig.cap="A: Aminoglycoside mutations classified by expected resistance to gentamicin, amikacin and kanamycin B: Row and column clustered heatmap showning phenotypic amikacin and gentamicin resistance and identified aminoglycoside resistance genes. AadA indicates \\textit{aadA} genes, also referred to as ANT(3\")-Ia. Aac3\\_IIa and \\_IId refers to \\textit{aac(3)-IIa} and \\textit{-IId} genes respectively, and Aac6\\_Ib\\_cr to \\textit{aac(6\')-Ib-cr}. Aph3 referrs to \\textit{aph(3\')-Ia}. GEN R on AST and AMK R on AST refer to samples that were resistant to gentamicin or amikacin on antimicrobial sensitivity testing (AST). Each row is one sample, red = presence, blue = absence. Samples lacking AST data were excluded from this heatmap.", out.height='80%', fig.align='center'}

knitr::include_graphics(paste0(wd,"aminoglycoside_plot.pdf"))
```

#### Chloramphenicol resistance

Presence of chloramphenicol resistance determinants was common; 248/473 (52%) of isolates contained at least one chloramphenicol resistance gene (Figure \@ref(fig:wgs-amr-sum)), usually 1 (210/248 [85%]), less commonly 2 (37/248 [15%]) or 3 (1/248 [<1%]). The most commonly identified gene was _catB4_,  but presence of _catB4_ was not associated with phenotypic chloramphenicol resistance (OR 0.9 [95% CI 0.6-1.4], p = 0.65). In comparison, presence of any other chloramphenicol resistance gene was associated with phenotypic resistance (OR 2.5 [95% CI 1.6-3.9], p < 0.001 for a composite variable of all other genes). The reason for this is not clear, but in addition partially assembled _catB4_ genes were very common; of the 328 isolates in which there was no fully assembled _catB4_, 93% (306/323) were reported by ARIBA to contain a partially assembled _catB4_ gene. In many cases, these partial genes seemed to be truncated by an IS26 insertion sequence (see Chapter 8).

#### Co-trimoxazole, tetracycline, macrolide and other resistance determinants

Almost all isolates contained either a trimethoprim resistance (459/473 [97%]) or a sulphonamide resistance gene (468/473 [99%]); only 3/473 isolates did not contain either. Trimethoprim resistance genes were all of the _dfrA_ family; _sulII_ was the commonest sulphonamide resistance determinant (Figures \@ref(fig:wgs-amr-sum) and \@ref(fig:wgs-amr-chlor-cotrim)B). Phenotypic cotrimoxazole resistance was also near-ubiquitous, in 96% (433/448) of those isolates in which antimicrobial sensitivity testing was done.

Tetracycline and macrolide resistance genes were also very common: tetracycline resistance determinants were identified in 411/473 (86%) of isolates, most commonly _tetB_ (262/473 [55%] of isolates) and _tetA_ (193/473 [41%] of isolates). Macrolide resistance determinants were detected in 325/473 (69%); all isolates carried _mphA_ most commonly alone but rarely with _ermB_ (9 isolates) or _mefB_ (1 isolate). No antimicrobial sensitivity testing was carried out for any agent of the tetracycline or macrolide class. Resistance determinants for rifampicin (_arr2_ and _arr3_) were rarely identified, in 2 isolates and the _sat2_ gene, conferring resistance to streptothricin (a nucleoside antibiotic with no clinical compounds in use) was seen in 69/473 [15%] of isolates; the significance of this is unknown. Finally, the fosfomycin resistance determinant _fosA_ was seen in 28/473 [6%] of isolates, despite this antimicrobial being unavailable in Malawi. It was not restricted to any one clade (Figure \@ref(fig:amr-map-to-tree)).

```{r wgs-amr-chlor-cotrim-setup, include= FALSE}


amr.melt %>% filter(class == "Chloramphenicol") %>% group_by(variable) %>% summarise(n= n()) -> cl
orgsall %>% filter(variable == "Chloramphenicol") %>% dcast(sample_number ~ variable, value.var = "value") -> cl.sens
cl.sens <- merge(select(sample_ids, Supplier.Name, Lane), cl.sens, by.x = "Supplier.Name", by.y = "sample_number")
cl.genes <- m.c[c("name", names(m.c)[names(m.c) %in% cl$variable])]
cl.sens <- merge(cl.sens, cl.genes, by.x = "Lane", by.y= "name", all.x = T)
cl.sens$Chloramphenicol[cl.sens$Chloramphenicol == "Sensitive"] <- 0
cl.sens$Chloramphenicol[cl.sens$Chloramphenicol == "Resistant" | cl.sens$Chloramphenicol =="Intermediate"] <- 1
cl.sens$Chloramphenicol <- as.numeric(cl.sens$Chloramphenicol)
names(cl.sens)[names(cl.sens) == "Chloramphenicol"] <- "CHL R on AST"

grid.grabExpr( pheatmap(cl.sens[c(3,8,5,9,7,6,4,10)],  legend = F, show_rownames = F)) -> hm1
hm1<- as_ggplot(hm1)

amr.melt %>% filter(class == "Trimethoprim" | class == "Sulphonamides") %>% group_by(variable) %>% summarise(n= n()) -> cot

orgsall %>% filter(variable == "Cotrimoxazole") %>% dcast(sample_number ~ variable, value.var = "value") -> cot.sens
cot.sens <- merge(select(sample_ids, Supplier.Name, Lane), cot.sens, by.x = "Supplier.Name", by.y = "sample_number")
cot.genes <- m.c[c("name", names(m.c)[names(m.c) %in% cot$variable])]
cot.sens <- merge(cot.sens, cot.genes, by.x = "Lane", by.y= "name", all.x = T)
cot.sens$Cotrimoxazole[cot.sens$Cotrimoxazole == "Sensitive"] <- 0
cot.sens$Cotrimoxazole[cot.sens$Cotrimoxazole == "Resistant" | cot.sens$Cotrimoxazole =="Intermediate"] <- 1
cot.sens$Cotrimoxazole <- as.numeric(cot.sens$Cotrimoxazole)

names(cot.sens)[names(cot.sens) == "Cotrimoxazole"] <- "Cotrimoxazole resistance"

names(cot.sens)[names(cot.sens) == "Cotrimoxazole resistance"] <- "SXT R on AST"

grid.grabExpr( pheatmap((cot.sens[3:16]), legend = F, show_rownames = F)) -> hm2
hm2<- as_ggplot(hm2)




 #sens
#binom.test(summ$TP, (summ$TP + summ$FN))
# spec
#binom.test(summ$TN, (summ$TN + summ$FP))

ggarrange(hm1,NULL,hm2, ncol = 3, nrow = 1, labels = c("A",NA,"B"), widths = c(1,0.2,1)) -> parrange

ggsave(paste0(wd,"chor_cotrim_plot.pdf"), plot = parrange, width = 7, height = 4, units = "in")
```

```{r wgs-amr-chlor-cotrim, echo = F, warning = F, message = F, fig.scap="Chloramphenicol and cotrimoxazole resistance mutations and phenotypic resistance", out.extra='', fig.cap="Heatmap showning phenotypic chloramphenicol (A) and cotrimoxazole (B) resistance and identified resistance genes that could be expected to confer resistance to these agents. Each row is one sample, red = presence, blue = absence. Both heatmaps are row and column clustered."}

knitr::include_graphics(paste0(wd,"chor_cotrim_plot.pdf"))

```

#### Clustering and lineage association of AMR determinants

Next, I explored associations of AMR determinants, both with each other in an attempt to identify putative clusters that could represent mobile genetic elements (MGE) that could be tracked within and between patients, and with lineages of the phylogeny. There was clear clustering of AMR genes beyond what would be expected by chance (Figures \@ref(fig:wgs-amr-cooccur-plot) and \@ref(fig:amr-net-plot)), including clustering of the ESBL gene $bla_{CTXM-15}$ with penicillinases $bla_{OXA-1}$ and $bla_{TEM-1}$. Though some identified clusters correspond to known MGE (e.g. the _sulII-strA-strB_ cluster[@Anantham2012]), there was a clear lineage association of certain gene combinations on mapping the presence or absence of AMR determinants back to the phylogeny (Figure \@ref(fig:amr-net-plot)), meaning that these AMR-gene associations likely represent a combination of co-location on MGE and confounding by association with lineage, and suggesting that using clusters of AMR genes to track MGE within participants is likely to be confounded by lineage. 

```{r wgs-amr-maptotree,include = F}
tree <- read.tree(paste0(wd, "core_genome_tree/core_alnD2ESCO_snp_sites.aln.treefile"))
midpoint.root(tree) -> tree
plasm <- read.csv(paste0(wd, "amr/plasm_Inc_pres_abs.csv"), stringsAsFactors = F)
m.c <- read.csv(paste0(wd, "amr/srst2_pres_abs.csv"), stringsAsFactors = F)
m.c$TEM_1[is.na(m.c$TEM_1)] <- 0

m.c[c("name", names(m.c)[-1][order(names(m.c)[-1])]) ] -> m.c
m.c <- subset(m.c, name %in% tree$tip.label)
qr <- read.csv(paste0(wd, "amr/qrdr_mut_presence.csv"), stringsAsFactors = F)
rownames(qr) <- qr$sample
rownames(m.c) <- m.c$name
rownames(plasm) <- plasm$name

sapply(m.c[,-1], as.character) -> m.c[2:ncol(m.c)]
sapply(qr[,-1], as.character) -> qr[2:ncol(qr)]
sapply(plasm[,-1], as.character) -> plasm[2:ncol(plasm)]
qr[qr == "1"] <- "3"
plasm[plasm == "1"] <- "2"
hm <- merge(qr, m.c, by.x = "sample", by.y = "name")
hm <- merge(hm, plasm, by.x = "sample", by.y = "name")
hm <- subset(hm, sample %in% tree$tip.label)
# get rid of 0 cols
apply(sapply(hm[-1], as.numeric),2, sum) -> sums
hm <- hm[c("sample", names(sums[sums != 0]))]
hm <- select(hm, -AmpC1, -AmpC2, -AMPH, -MrdA)
rownames(hm) <- hm$sample
ggtree(tree) %>% gheatmap(hm[,-1], font.size = 2,
colnames_position = "top", colnames_angle = 90, colnames_offset_y = 20,
color = NA, width = 3, offset = 0.02)  + scale_fill_manual(values = c("white", cols[1],cols[12], cols[8])) +
  theme(legend.position = "none") -> p3
```


```{r wgs-amr-jaccmat, include = F}

jcoefs <- read.csv(paste0(wd, "amr/amr_jaccard_matrix.csv"), stringsAsFactors = F, row.names = 1)

#jpvals <- read.csv("/Users/joelewis/Documents/Sanger/final_analysis/all_DASSIM_esco/ariba #amr/cleaned/amr_jaccard_pvals.csv", stringsAsFactors = F, row.names = 1)


 grid.grabExpr( pheatmap(jcoefs, fontsize = 5, color = viridis(n = 100, option = "plasma"), border_color = NA) ) ->p2
p2 <- as_ggplot(p2)

```

```{r wgs-amr-cooccurnet, include = F}

jcoefs <- select(jcoefs, -AmpC1, -AmpC2, -AMPH, -MrdA)
jcoefs[!(rownames(jcoefs) %in% c("AmpC1", "AmpC2", "AMPH", "Mrd")),] -> jcoefs

jpvals <- read.csv(paste0(wd, "amr/amr_jaccard_pvals.csv"), stringsAsFactors = F, row.names = 1)
jpvals <- select(jpvals, -AmpC1, -AmpC2, -AMPH, -MrdA)
jpvals[!(rownames(jpvals) %in% c("AmpC1", "AmpC2", "AMPH", "Mrd")),] -> jpvals
jpvals$gene <- rownames(jpvals)
melt(jpvals, id.vars = "gene") -> jpvals_long
names(jpvals_long) <- c("gene1", "gene2", "pval")
jcoefs$gene <- rownames(jcoefs)
melt(jcoefs, id.vars = "gene") -> jcoefs_long
names(jcoefs_long) <- c("gene1", "gene2", "jacc")
jtots <- merge(jcoefs_long, jpvals_long, by = c("gene1", "gene2"))
jpvals_sig <- subset(jpvals_long, pval < 0.05/nrow(jpvals_long))

# ok lets make a graph of these

jtots2 <- subset(jtots, jacc > 0.5 & pval < 0.05/nrow(jpvals_long))
apply(select(jpvals_sig, gene1, gene2), 1, 
      function(x) paste(sort(x), collapse = "")) -> jpvals_sig$key
subset(jpvals_sig, !duplicated(key)) -> jpvals_sig_unique
g1 <- graph_from_data_frame(jtots[1:3], directed = F)
g2 <- graph_from_data_frame(jtots2[1:3], directed = F)
all <- merge(qr,m.c, by.x = "sample", by.y = "name")
n_genes <- apply(sapply(all[-1], as.numeric), 2, sum)


g2 %>%
  ggraph(layout="kk") +
  geom_edge_link(colour = "grey", alpha = 0.7) + geom_node_point(aes(size = (n_genes[V(g2)$name])), colour = viridis::plasma(100)[50], alpha = 0.8) +
  geom_node_text(aes(label = name), size = 4, repel = T) +
  theme_graph(foreground=NA) + theme(legend.position = "none") -> p1
 

```

```{r wgs-amr-cooccur-plot, echo = F, warning = F, message = F, fig.scap="Co-occurance and lineage asociation of AMR genes", out.extra='', fig.cap="Row and column clustered heatmap of pairwise Jaccard index matrix, showing clustering of AMR genes.", fig.width = 6, fig.height = 6}

#ggarrange(ggarrange(p2,
#                    ggarrange(NULL,p1,NULL, heights = c(0.1,1,0.1), ncol = 1, nrow =3, labels = c("B", #NA,NA)),
#                    nrow =1, ncol= 2, widths = c(1.2,0.8), labels = c("A", NA)), 
#          p3, ncol = 1, nrow = 2, labels = c(NA, "C"), heights = c(0.8,1))

p2

```

```{r amr-net-plot, echo = F, warning = F, message = F, fig.width = 6, fig.height= 6, fig.scap="Networks of co-occuring AMR genes", out.extra='', fig.cap="Networks of associated (jaccard index > 0.5) AMR genes that occur more often than expected by chance (Bonferroni corrected p-value < 0.05).", out.height = '40%', fig.align= "center"}

p1

```

```{r amr-map-to-tree, echo = F, warning = F, message = F, fig.width = 10, fig.height= 6, fig.scap="AMR genes and plasmid replicons mapped back to tree", out.extra='', fig.cap="AMR genes (blue, chromasomal quinolone resistance and red, other AMR genes) and plasmid replicons (purple) mapped back to tree showing that some AMR gene associations are also associated with lineage."}

p3

```



### Plasmid replicons

The frequency of isolation of different plasmid replicons is shown in Table \@ref(tab:plasm-tab) and presence or absence of the identified plasmid replicons is shown mapped to the phylogeny in Figure \@ref(fig:amr-map-to-tree). IncFIb was most commonly identified (399/473 [84%] of isolates), followed by IncFII (383/473 [81%] of isolates) and IncF1a (324/373 [68%] of isolates). Col plasmids were also frequently identified, in 308/473 [65%] of isolates. Once again, there seems to be some lineage associations of presence or absence of replicons (Figure \@ref(fig:amr-map-to-tree)).

```{r plasm-tab, echo = F, warning = F, message = F }

plasm[plasm == 2] <- 1

plasm %>% pivot_longer(-name) %>%
  filter(value != 0) %>%
  group_by(name..2) %>% 
  tally() %>%
  arrange(desc(n)) -> plasm.sum

kable(plasm.sum, "latex", longtable = T, booktabs = T, caption = "Frequency of identification of plasmid replicons", col.names = c("Plasmid replicon", "n")) %>%
kable_styling(full_width = F, latex_options = c("repeat_header"), font_size = 9) 


```
## Discussion

In this chapter, I have presented the results of whole genome sequencing of 473 ESBL _E. coli_ recovered from serial sampling of 230 Malawian adults from a combination of healthcare-associated and community settings. I have placed these Malawian isolates in the context of the global diversity of _E. coli_ using phylogroup, sequence type and phylogenetic modelling. I have described the AMR determinants and plasmids present in the isolate of and explored clustering of AMR genes. 

### Genomic landscape of ESBL _E. coli_ in Malawi: global diversity and high-risk clones

The _E. coli_ recovered from stool of the study participants in this study are diverse, encompassing the spectrum of diversity of the species with all major phylogroups and 56 STs represented. Phylogroup A was the commonest phylogroup seen, consistent with the traditional view of this phylogroup as associated with commensal strains[@Dale2015]. When placed into the context of genomes from throughout the world, the Malawian isolates are largely distributed throughout the phylogeny: in a global context, Malawi is sampling the worldwide diversity of _E. coli_. 

There were, however, several areas of the global phylogeny where the Malawian isolates clustered tightly together, perhaps initially suggestive of Malawi-restricted clones; however in considering the significance of this tree topology it is important to be cognisant of the biases inherent in the global _E. coli_ collection. ESBL-producing _E. coli_ are unusual in the ETEC[@VonMentzer2014] and GEMS[@Ingle2016] collections and all samples in these two studies were collected before 2011. Both of these collections are exclusively recovered from stool. In contrast, the clinical isolates from the Thai study[@Runcharoen2017] are all invasive, from a single centre, are selected on the basis of being ESBL-producers, and were isolated in 2014 or 2015. The isolates from the previous Malawian study were largely invasive[@Musicha2017], were selected for diversity in AMR profile, and were all isolated before 2014. There was no study that selectively cultured for ESBL producing _E. coli_ in stool, as this study has done; in that, all of these studies are slightly sub-optimal for comparison. It may be that these biases contribute to apparent polyphyletic clustering of isolates from the current study in phylogroup A in the global tree. It would be expected that ESBL producing phylogroup A _E. coli_ would be under-represented in the global collection compared to this study, as this phylogroup is associated with commensal (and hence stool) associated strains, and the two studies performing stool culture did not enrich for ESBL producers; the only study to do this collected invasive isolates, which may be expected to lie in phylogroup B2 over A. 

Nevertheless, two of the three commonest STs identified in this study, ST410 and ST167, are unusual or absent in the global collection and could be considered to have a single introduction into Malawi in the context of the topology of the inferred phylogenetic tree. These could represent Malawi-restricted clades or, more likely given the diversity otherwise seen in the tree, clades that are not represented in the global collection because of biases in sample selection. ST410 is recognised as an emerging high-risk clone which has been with isolated worldwide with some regularity since 2011 (including in Tanzania) and is associated with $bla_{CTXM-15}$ and $bla_{NDM-5}$; coalescence analysis suggested a most recent common ancestor of ST410 of the early 1800s (similar to ST131[@Stoesser2016]), and acquisition of $bla_{CTXM-15}$ on a multreplicon IncFII-IncFIA-IncFIB plasmid in the late 1980s[@Roer2018]. Similarly, ST167 has been recognised as commonly carrying ESBL genes and carbapenemases in Chinese invasive isolates[@Zong2018] - indeed, it was the commonest _E. coli_ ST in one longitudinal surveillance study of carbapenemases in 2012-16 in 25 Chinese provinces[@Wang2018] -  as well as being very prevalent among meat-associated _E. coli_ in Germany between 2011-2013 in one study[@Irrgang2017]. As such it, too, is also likely a successful global AMR-associated lineage. It is therefore likely that these STs are not represented in the global tree because they have recently emerged worldwide, and have also recently arrived in Malawi. If this is the case, then ST410 and 167 have become rapidly established in Blantyre over the course of only 2-3 years (as they were not seen at all in the previous Malawian strain collection); in fact, this is exactly what was seen in longitudinal nationwide genomic surveillance of _E. coli_ in the UK in 2003-04 when ST131 first arrived[@Kallonen2017]. 

ST131 is a globally established high risk clone and the commonest ST in this study and the global collection. Indeed this ST is thought to account for 40-80% of invasive ESBL _E. coli_ infection worldwide[@Nicolas-Chanoine2014; @Pitout2017]. The topography of the global tree suggests no obvious barriers to mixing between Malawian and global ST131. It may be that unbiased global sampling would reveal the same pattern for ST167 and ST410. Though some progress has been made in understanding the genomics of the emergence of ST131[@Stoesser2016], the factors that contribute to its apparent fitness are unknown: it is impossible to predict, at present, from the genome of ST167 and ST410 whether they will repeat the course of ST131 to become truly globally dominant as a cause of human disease. Such an understanding of the determinants of fitness would be of great benefit in predicting and preventing global AMR spread.

### Antimicrobial resistance determinants: domination of $bla_{CTXM-15}$ and emergence of carbapenemases

The 473 isolates contained a diverse selection of antimicrobial resistance determinants, most with genotypic  multiclass resistance. Genotypic and phenotypic co-trimoxazole resistance was near universal, as might be expected from a setting where lifelong co-trimoxazole preventative therapy (CPT) is mandated by the Malawian HIV treatment guidelines for HIV all infected adults[@MinistryofHealth.GovernmentofMalawi2016], and mediated by _dfrA_ and _Sul_ alleles. 

Determinants of aminoglycoside resistance were also present in almost every sample, most commonly streptomycin resistance determinants, despite the lack of drug pressure. Well recognised gentamicin-resistance determinants - _aac(3)_ and _aac(6")_ - were strongly associated with gentamicin resistance, but the aminoglycoside resistance genes present in isolates displaying apparent amikacin resistance (_aad5_ and _aac(3)_) would not usually be expected to confer amikacin resistance. Given the extreme rarity of phenotypic amikacin resistance, re-testing these isolates with a more accurate AST method should be the first step (see limitations, below). 

Quinolone resistance determinants were also common. Quinolones are widely used in Blantyre, and are the current treatment of choice for invasive _Salmonella_ infections, one of the commonest cause of bloodstream infection here[@Musicha2017b]. The genotypic determinants of quinolone resistance are complex, mediated by point mutations in the drug target regions, acquisition of modifying enzymes or up regulation of multidrug resistance pumps, and usually multiple genes or mutations are needed to bring about a resistant phenotype[@Redgrave2014]. This is the pattern observed in this dataset where the proportion of isolates with identifiable determinants of quinolone resistance was greater than those with phenotypic resistance. Nevertheless, the genotype-phenotype associations seen in this data are largely those that have been described in the literature: the _gyrA83–parC80–gyrA87_ mutation combination has been shown to strongly predict combination quinolone resistance in a study of 10099 _E. coli_ genomes[@Ostrer2019], and it is recognised that the presence of the _qnrS_ gene alone is usually not sufficient to bring about a resistant phenotype[@Garoff2018]. 

Presence of chloramphenicol resistance genes, particularly _catB4_, was more common than phenotypic chloramphenicol resistance such that _catB4_ was not associated with phenotypic resistance. This is unexpected, as presence of chloramphenicol resistance genes has been shown to correlate well with phenotypic resistance[@Neuert2018], including in a study of 94 Malawian invasive isolates[@Williams2019], though in this Malawian collection (the same collection of isolates as were included in the global collection in the current study), _catB_ genes were rarely seen.  Interestingly, truncated _catB4_ elements (often in conjunction with an IS26 transposon) were almost universal in the isolates in this study: of 323 isolates in which ARIBA did not assemble a full _catB4_ sequence, 93% (306/323)) contained a truncated _catB4_ element. This configuration (_catB4_ truncated by an IS26 element) has been described in Enterobacteriaceae[@Totsika2011; @Sekizuka2018]. It could certainly be unrelated but its ubiquity in this study raises at least the possibility of misassembly and false-positive identification of _catB4_ in some cases. Laboratory errors resulting in erroneous AST should also be considered. It is also possible that the _catB4_ gene in this collection is not expressed or downregulated in some way; exploring these hypotheses is a possible direction for future work.

ESBL resistance in this collection is dominated by $bla_{CTXM}$ and $bla_{CTXM-15}$ in particular; this latter gene was carried by 319/473 [67%] of isolates. The only non-$bla_{CTXM}$ ESBL gene identified in any significant prevalence was $bla_{SHV-12}$; ESBL $bla_{OXA}$ and $bla_{TEM}$ were very rare, though narrow-spectrum penicillinase alleles of this family were common. The dominance of $bla_{CTXM-15}$ is in keeping with the situation seen worldwide[@Bevan2017]. In this collection, $bla_{CTXM-15}$ was spread throughout the phylogeny rather than associated with any particular clade, as would be expected from its global dominance. I identified one carbapenemase, $bla_{NDM-5}$, carried on a globally successful IncX3 plasmid. To my knowledge, this is the first carbapenemase to be described in Malawi. Carbapenem antimicrobials were introduced to the Malawian essential medicines list in 2015 but are at best sporadically available, and only in tertiary centres, often for truncated courses. The emergence of carbapenemases with such minimal carbapenem use and so soon after introduction is troubling, and should prompt discussions regarding the best use of this precious antimicrobial class; certainly, given the high prevalence of ESBL production among invasive _K. pneumoniae_ and _E. coli_[@Musicha2017b], there is a case for expanded access but optimal antimicrobial stewardship strategies in this context are unknown.

### Study limitations

There are several limitations to the analysis carried out in this chapter. Only one _E. coli_ pick from each time point was sequenced which may have missed significant within-participant diversity. This is seriously problematic for the analyses of the next chapter; here, diversity may have been missed but the data presented here should at least reflect a random sample from the _E. coli_ carried by the participants in the study. Community isolates are under represented, largely because of a lower ESBL-E prevalence in community members. The sequenced _E. coli_ isolates presented here do not represent all the samples in the study from which ESBL producing _E. coli_ were isolated, but only those up to the point at which the samples were shipped for sequencing. I have focused on _E. coli_; in fact, these were often isolated in conjunction with other bacteria, most notably _Klebsiella pneumoniae_, which were not induced in this analysis. Given the propensity of _K pneumoniae_ to carry AMR genes, the circulating AMR gene diversity in Enterobacteriaeciae carried by participants in this study is likely greater than I describe here. 

For context genomes, I selected a global collection of _E. coli_ based on what was available but, as described above and in common with many analyses of this type, this is a biased collection. This must be borne in mind when interpreting the global phylogeny. There are inherent limitations in the short-read Illumina sequencing that was carried out: assembly of areas with multiple nucleotide repeats (as found in plasmids and transposable elements in particular) is difficult or impossible, making it impossible to fully characterise the MGE in this dataset upon which the AMR genes are carried. 

There are some discrepancies between phenotypic resistance and what would be expected from identified genes. Partly phenotype may have been misclassified. The AST method used was disc diffusion; certainly this is less accurate than an MIC determination method such as E-tests or dilution methods, and a true comparison of genotypic to phenotypic resistance - not the aim of this study - should use one of these methods. It is also possible that there were technical problems with the AST procedure (e.g. an overly heavy inoculum) though every attempt was made to avoid this, including with internal QC, and the work was carried out in a laboratory which subscribes to the UK NEQAS QC procedure.

### Conclusions and further work

In conclusion, I have shown that the _E. coli_ population in this study is diverse, representing global _E. coli_ diversity, and suggesting significant mixing of _E. coli_ between Blantyre and the rest of the world.  The AMR genes identified seem to reflect the local antibiotic pressures: near universal cotrimoxazole resistance, moderate quinolone resistance and very little carbapenem resistance. The presence of carbapenem resistance mediated by a $bla_{NDM-5}$ carbapenemase is to my knowledge the first carbapenemase described in Malawi. Its presence along with apparently unrestricted mixing of _E. coli_ from Malawi with the rest of the world is a reminder that international spread of bacteria and AMR determinants, even to low-resource settings like Malawi, can be rapid. Treatment options for invasive Gram-negative infections in Malawi are often limited by ESBL resistance[@Musicha2017b]. There is a significant unmet need for carbapenem antibiotics, but antibiotic stewardship interventions that can balance access to against restriction of these valuable last-line antimicrobials are needed to prevent rapid emergence of resistance following wider-scale carbapenem roll-out in Malawi.

Further work is planned:

* The discordance between chloramphenicol phenotype and genotype should be explored, especially as this antibiotic - largely fallen from favour in Malawi since ceftriaxone has been available - may have a role to play in treating some invasive ESBL-E infections. Long-read sequencing of representative isolates will exclude the possibility of missassembly and allow assessment of the genetic environment of the _catB4_ gene; it is possible that it has been downregulated or is not expressed, for example. More accurate chloramphenicol AST using e.g. E-tests may also be helpful to confirm phenotype, which should also be done with the apparent amikacin resistance isolates presented here.

* The apparent recent arrival of the putative high-risk clones ST410 and ST167 deserves further attention. The isolates here can be placed in a global context with other publicly available ST410 and ST167 _E. coli_, but a question remains as to whether these clones are causing invasive disease. This question can only be answered by sequencing invasive isolates; this work is ongoing and will allow such a comparison to be made.

* The remainder of the samples (post October 2018) will have one _E. coli_ sent for sequencing.

* All samples where _K. pneumoniae_ were isolated will have one _K. pneumoniae_ colony pick sequenced allowing a comparative analysis of AMR determinants by _K. pneumoniae_ and _E. coli_ carried within a single participant.

* More broadly, I have presented a description of the isolates sequenced in this study, but largely without reference to the available metadata. The motivation in sequencing these isolates was to use WGS as a high-resolution typing scheme to track bacteria within and between participants. This analysis is the focus of the next chapter.


## Appendix


```{r wgs-assembly-stats-dassim, echo = F, warning = F, message = F}

 md <- metadata %>% select(Lane.Name, Sample, study, Source1, Country)

md$study[grepl("DASSIM", md$study)] <- "This study"

md$study[grepl("ingle", md$study)] <- "Ingle 2018"
md$study[grepl("mentzer", md$study)] <- "Mentzer 2014"
md$study[grepl("musicha", md$study)] <- "Musicha 2017"
md$study[grepl("runchaeron", md$study)] <- "Runchaeron 2017"




acc <- read.csv(paste0(wd, "global_tree/all_acc.csv"), stringsAsFactors = F)
sub("#", "_", acc$Lane.name ) -> acc$Lane.name
merge(md, select(acc, Lane.name, Lane.accession), by.x = "Lane.Name", by.y = "Lane.name") -> md



md$Sample[md$study == "This study"] <- md$Lane.Name[md$study == "This study"]

md <- select(md, -Lane.Name)
md <- select(md, study, Sample, Lane.accession, Source1, Country)

md$study <- factor(md$study, levels = c("This study","Ingle 2018" , "Mentzer 2014","Musicha 2017", "Runchaeron 2017" ))
md <- md[order(md$study, md$Sample),]
rownames(md) <- NULL
md$Lane.accession[md$Lane.accession == "not found"] <- ""

md$Country[md$Country == "thailand"] <- "Thailand"
md$Source1[md$Source1 == "stool"] <- "Stool"

kable(md, "latex", longtable = T, booktabs = T, caption = "Details of samples included in global phylogeny", col.names = c("Study",
                           "Sample ID",
                           "Accession No.",
                           "Source", "Country")) %>%
kable_styling(font_size = 7,full_width = F, latex_options = c("repeat_header")) %>%
   footnote(general = "RS = Rectal swab")



```

