# Whole genome sequencing of ESBL _E. coli_ carriage isolates

## Chapter overview

This chapter describes the use of whole-genome sequencing (WGS) of ESBL producing _E. coli_ to understand the drivers of gut mucosal ESBL-E carriage. I will begin with a description of the genomic landscape of the isolates form this study: using simple descriptions of _E. coli_ phylogroup and multilocus sequence type (MLST) I will place the isolates from this study in the context of the _E. coli_ population, followed by higher-resolution contextulaistaion using phylogenetics to place isolates from this study in the context of a global _E. coli_ collection. I will describe the genetic basis of antimicrobial resistance in these isolates and explore the extent to which AMR genes tend to cluster together beyond what would be expected by chance. Finally, I will attempt to use the resolution offered by WGS to attempt to answer two specific questions: firstly, what is the mechanism of rapid increase in ESBL-E carriage prevalence following hospital admission and antimicrobial exposure we see in this study? Secondly, what is the likely unit of ESBL-E transmission in this study? Are bacteria, or mobile genetic elements (MGE) implicated? And if, MGE, which: plasmids, transposons, integrons - or a combination?

These questions, a phrased in this way seem difficult or impossible to answer given the available WGS data, but by slightly reframing them they become tractable: first, what is the diversity of apparent hospital-acquired ESBL _E. coli_ in comparison to apparent community-acquired isolates? Apparent hospital acquisitions could represent true acquisitions of, for example, a hospital-associated clone - but equally they could be an "unmasking" of minority variant _E. coli_ in the microbiota, acquired in the community but not detected by culture because of low abundance, until enriched for by antimicrobial exposure. If the diversity of apparently hospital aquired isolates is contained within the diversity of community isolates, this would lend support to this latetr hyothesis. The second question - what is the unit of transmission in this system - can be reframed by asking: what is the unit that is most conserved within patients, as compared to between patients?  The questions then reduce to a dimensionality redcution problem: in order to address them both, it is necessary to classify either bacteria or MGE into mutually exclusive categories, in order to compare hospital to community isolates, and between-patient to within-patient. I describe the approach I have taken to this below. 

## Methods

### Bioniformatic pipeline

The basic bioinformatic pipeline used is described in detail in Chapter 2, methods. Briefly, one _E. coli_ colony from each patient sample was taken forward for DNA extraction and paired-end short-read whole genome sequencing using Illumina HiSeq X at the Wellcome Sanger Institute. Read quality control was undertaken with Kraken[@Wood2014] v0.10.6 to assign reads to species and WSI QC pipeline which maps a random 100 Mbases fron each sample to a reference and calculates depth of coverage, number of heterogeneous SNPs, GC content and insert size. Samples that contained > 20% non _E coli._ reads were discarded and _de novo_ assembly was undertaken with SPAdes[@Bankevich2012a] v3.11.0. Assembly statistics were calculated with QUAST[@Gurevich2013] v4.6.0 and completeness and contamination of the assemblies assessed by checkM[@Parks2015] v1.0.7. Contaminated assemblies (with checkM-defined contamination of > 25%) or poor assemblies (with N50 < xxx) were discarded. Annotation was carried out with Prokka[@Seemann2014a] v1.5 with a genus specific database from RefSeq and the Roary v1.007 pan-genome pipeline[@Page2015] was used to construct a core genome. A core gene multiple sequence alignment was generated using maaft[@Katoh2013] v7.205, SNP-sites identified using SNP-sites[@Page2016] v2.4.1 and the resultant SNP alignment used to build a maximum likelihood phylogenetic tree using IQ-TREE[@Nguyen2015] v1.6.3, with the ModelFinder module used to find the best fitting nucleotide substition model. Trees were visualised in the ggtree v1.14.4 package[@Yu2017] in R. 

Ariba[@Hunt2017a] v2.12.1 was used to identify AMR-associated genes using the SRST2[@Inouye2014] database, to identify plasmid replicons using the PlasmidFinder database[@Carattoli2014] and to perform _in silico_ muli-locus sequence typing (MLST) using the database from http://mlst.warwick.ac.uk/mlst/dbs/Ecoli accessed via www.pubmlst.org. Because quinolone resistance often results from SNPs in the chromasome in the quinolone resistance determining regions (QRDRs) of the _gyrA_, _gyrB_, _parE_ and _parC_ genes - rather than acquisition of whole AMR-determining genes, as is the case with the other genes sought by Ariba - these genes were downloaded from the comprehensive antimicrobial resistance database (CARD, https://card.mcmaster.ca/) and Ariba again used to call SNPs in them, with default settings. _E. coli_ phylogrouping was performed with a quadruplex _in silico_ PCR using the Clermont scheme[@Clermont2013] and isPcr v33x2 (https://github.com/bowhan/kent/tree/master/src/isPcr) 

Rhierbaps package v1.1.0 in R[@Cheng2013] was used to cluster the core genome pseudosequence into sequence clusters (SCs) with 2 levels and these level 2 clusters used to test associations (see statistical analysis, below). To track putative mobile genetic elements ESBL-gene containing contigs were identified using BLASTn[@Altschul1990] v2.7.0 with and then contigs containing any given ESBL gene (for example, $bla_{ctxm15}$) were clustered using cd-hit[@Li2006] v4.6 to produce mutually exclusive ESBL-gene contig-clusters for each identified ESBL gene. In order to attempt to determine what the biological significance of the identified contig-clusters is (i.e. what kind of MGE element they are likely to represent), basic statistics were plotted (number of samples contained withing each cluster, length of longest contig in cluster in kbp, length distribution of all contigs is cluster relative to longest contig and distribution of sequence identity compared to the longest contig in the cluster). Presence of compound trasposons, AMR determinents and plasmid replicons were elucidated by using BLAST with default settings of each contig-cluster representative sequence (i.e one for each contig-cluster) against the insertion sequence finder (ISfinder) database and the SRST2 database, taking the top hit (as determined by bitscore) for any given location, and visualising the results in gggenes v0.3.2. To assess lineage association, the contig clusters were mapped back to the core genome SNP tree. 

### Global _E. coli_ collection



### Statistical analysis

In order to explore clustering of AMR genes, Jaccard index was calculated for a given AMR-gene pair using he Jaccard v0.1.0 package in R. The Jaccard index, a measure of the similarity of two sets of data, is defined as _intersection over union_; in this context, for a given pair of AMR genes $x$ and $y$, the Jaccard index $J(x,y)$ is the number of isolates that contain both gene $x$ and $y$ divided by the total number that contain either $x$ or $y$:

$$J(x,y) = \displaystyle \frac{| x \cap y |}{| x \cup y| }$$

By definition it lies between 0 ($x$ and $y$ never co-occur) and 1 ($x$ and $y$ always co-occur). Co-occurance matrices using the Jaccard index were plotted using the pheatmap v 1.0.12 package in R. p values generated using a simple Fisher's test with Bonferroni correction were used to assess the statistical significance of co-occurance of genes; a p value of less that 0.05 was considered statistically significant. Co-occurance networks of genes occuring commonly together (defined as Jaccard index > 0.5) at a rate greater than expected by chance (p < 0.05 following Bonferroni correction) and uncommonly occuring together (defined as Jaccard index < 0.1 and p < 0.05 following Bonferroni correction) were plotted using igraph v1.2.2 and ggraph v1.0.2 in R.

To explore hospital or community associations of any given _E. coli_ clade, the location of isolation was first plotted against the maximum likelihood tree; location of isolation was classified as hospital, community, or recent hospital discharge (defined as a date of isolation within 2 weeks of hospital discharge). This latter category was used because it is possible that a patient could acquire an ESBL-E clone in hospital but only be sampled once leaving hospital; using only hospital isolated and community isolted categories could therefore introduce bias. Hospital or community association of each sequence cluster was assessed using a Fisher's test of proportion of hospital associated samples (defined as proportion of total of hospital isolated plus recent hospital discharge) for the given sequence cluster as compared to proportion of hospital associated samples in the remainder of the samples, with a Bonferroni correction for multiple comparisons. p < 0.05 was again considered statistically significant.

To compare within-patient to between-patient clustering of bacteria (as represented by core genome alignment and sequence cluster) and MGE (as represented by the contig-clusters), several approaches were taken. Firstly, it was hypothesised that any within-patient correlation is likely to be a function of time: samples closer together in time may be more likely to be more similar. To assess if this was the case, pairwise core genome pseudosequence SNP distance was was calculated using snp-dists v0.4 (https://github.com/tseemann/snp-dists) for all samples nd plotted against the time difference (in days) between samples, within and between patients with a smoothed curve fitted using a general additive model with cubic splines. Because of significant overplotting, this was also plotted as a 2D density plot. Based on these plots, the within and between patient SNP distances were compared in two post-hoc defined groups binned by time distance between the samples, and distributions compared with Kruskal-Wallace tests.

I then compared the within patient temporal clustering using the sequence-clusters and contig-clusters, by estimating the probability at any given time that two samples from the same patient would contain the same sequence cluster or contig-cluster. To estimate this, for any given time between samples, $t$ I defined a window of +/-5 days and estimated the probabilities as the sum of all within-patient samples in the window $(t - 5, t+5)$ that contained the same sequence cluster or contig-cluster divided by the sum of all samples within that time window. Exact binomial confidence intervals for these probabilities were generated and probabilities plotted as a function of time. In order to estimate the probability of two samples containing the same sequence cluster of contig-cluster purely by chance, 1000 samples were randomly drawn from different patients with replacement and used to generate a probability of two samples containing the same sequence or contig-cluster

Finally

## Results

### Samples and quality control

### Phylogroup, MLST and ML phylogeny of study isoltes

### Study isolates in a global context

### Antimicrobial resistance determinents