# Whole genome sequencing of ESBL _E. coli_ carriage isolates
\chaptermark{WGS of ESBL-E}

``` {r ch7-setup, include = F}

wd <- "chapter_7/"

# and packages
library(plyr)
library(tidyverse)
library(reshape2)
library(knitr)
library(kableExtra)
library(grid)
library(ggpubr)
library(ggtree)
library(ape)
library(phytools)
library(RColorBrewer)
library(scales)
library(cowplot)
library(pheatmap)


 parse_checkm_lines <- function(s, n) {
  strsplit(s, " ") -> stemp
  stemp <- stemp[[1]][stemp[[1]] != ""] 
  stemp <- c(stemp[1], paste0(stemp[2],"_", stemp[3]), stemp[4:15])
  names(stemp) <- n
  return(stemp)
 }


output = "latex"
source("other_scripts/load_metadata.R")


```
## Chapter overview

This chapter describes the use of whole-genome sequencing (WGS) of ESBL producing _E. coli_ to understand the drivers of gut mucosal ESBL-E carriage. I will begin with a description of the genomic landscape of the isolates form this study: starting with simple descriptions of _E. coli_ phylogroup and multilocus sequence type (MLST) I will place the isolates from this study in the context of the _E. coli_ population, followed by higher-resolution contextulaistaion using phylogenetics to place isolates from this study in the context of a global _E. coli_ collection. I will describe the genetic basis of antimicrobial resistance in these isolates and explore the extent to which AMR genes tend to cluster together beyond what would be expected by chance. Finally, I will attempt to use the resolution offered by WGS to attempt to answer two specific questions: firstly, what is the mechanism of rapid increase in ESBL-E carriage prevalence following hospital admission and antimicrobial exposure we see in this study? Secondly, what is the likely unit of ESBL-E transmission in this study? Are bacteria, or mobile genetic elements (MGE) implicated? And if, MGE, which: plasmids, transposons, integrons - or a combination?

These questions, phrased in this way, seem difficult or impossible to answer given the available WGS data, but by slightly reframing them they become tractable: first, what is the diversity of apparent hospital-acquired ESBL _E. coli_ in comparison to apparent community-acquired isolates? Apparent hospital acquisitions could represent true acquisitions of, for example, a hospital-associated clone - but equally they could be an "unmasking" of minority variant _E. coli_ in the microbiota, acquired in the community but not detected by culture because of low abundance, until enriched for by antimicrobial exposure. If the diversity of apparently hospital aquired isolates is contained within the diversity of community isolates, this would lend support to this latter hyothesis. The second question - what is the unit of transmission in this system - can be reframed by asking: what is the unit that is most conserved within patients, as compared to between patients?  The questions then reduce to a dimensionality redcution problem: in order to address them both, it is necessary to classify either bacteria or MGE into mutually exclusive categories, in order to compare hospital to community isolates, and between-patient to within-patient. I describe the approach I have taken to this below. 

## Methods

### Bioinformatic pipeline

The basic bioinformatic pipeline used is described in detail in Chapter 2, methods. Briefly, one _E. coli_ colony from each patient sample was taken forward for DNA extraction and paired-end short-read whole genome sequencing using Illumina HiSeq X at the Wellcome Sanger Institute. Read quality control was undertaken with Kraken[@Wood2014] v0.10.6 to assign reads to species and WSI QC pipeline which maps a random 100 Mbases fron each sample to a reference and calculates depth of coverage, number of heterogeneous SNPs, GC content and insert size. Samples that contained > 80% non _E coli._ reads were discarded and _de novo_ assembly was undertaken with SPAdes[@Bankevich2012a] v3.11.0. Assembly statistics were calculated with QUAST[@Gurevich2013] v4.6.0 and completeness and contamination of the assemblies assessed by checkM[@Parks2015] v1.0.7. Contaminated assemblies (with checkM-defined contamination of > 25%) or poor assemblies (with less than 1Mb assembled length) were discarded. Annotation was carried out with Prokka[@Seemann2014a] v1.5 with a genus specific database from RefSeq and the Roary v1.007 pan-genome pipeline[@Page2015] was used to identify a core genome. A core gene multiple sequence alignment was generated using maaft[@Katoh2013] v7.205, SNP-sites identified using SNP-sites[@Page2016] v2.4.1 and the resultant SNP alignment used to build a maximum likelihood phylogenetic tree using IQ-TREE[@Nguyen2015] v1.6.3, using ascertainment bias correction to correct for the fact that the input pseudosequence contained only variable sites, and using the ModelFinder module used to find the best fitting nucleotide substition model. This calculates the likeliood of a number of different models and chooses the model with the lowest (best fitting) Bayesian Information Criterion, a statistic which penalises model parameters. Reliability of inferred branch partitions was assessed with 1000 bootstrap replicates. Trees were visualised in the ggtree v1.14.4 package[@Yu2017] in R. 

Ariba[@Hunt2017a] v2.12.1 was used to identify AMR-associated genes using the SRST2[@Inouye2014] database, to identify plasmid replicons using the PlasmidFinder database[@Carattoli2014] and to perform _in silico_ muli-locus sequence typing (MLST) using the database from http://mlst.warwick.ac.uk/mlst/dbs/Ecoli accessed via www.pubmlst.org. The $\beta$-lactamase genes $ampC1$, $ampC2$ and $ampH$ were excluded from the analysis of AMR determinents as they do not usually cause a resistant phenotype in _E. coli_. Because quinolone resistance often results from SNPs in the chromasome in the quinolone resistance determining regions (QRDRs) of the _gyrA_, _gyrB_, _parE_ and _parC_ genes - rather than acquisition of whole AMR-determining genes, as is the case with the other genes sought by Ariba - these genes were downloaded from the comprehensive antimicrobial resistance database (CARD, https://card.mcmaster.ca/) and Ariba used to call SNPs in them, with default settings. _E. coli_ phylogrouping was performed with a quadruplex _in silico_ PCR using the Clermont scheme[@Clermont2013] and isPcr v33x2 (https://github.com/bowhan/kent/tree/master/src/isPcr) 

Rhierbaps package v1.1.0 in R[@Cheng2013] was used to cluster the core genome pseudosequence into sequence clusters (SCs). Two levels were used and these level 2 clusters used to test associations (see statistical analysis, below). To track putative mobile genetic elements ESBL-gene containing contigs were identified using BLASTn[@Altschul1990] v2.7.0 of all contigs against the SRST2 database and then contigs containing any given ESBL gene  were grouped by the ESBL gene they contained (for example, all $bla_{ctxm15}$ gene-containing clusters were grouped together), and each group clustered using cd-hit[@Li2006] v4.6 to produce mutually exclusive ESBL-gene-containing contig clusters for each identified ESBL gene. Henceforth, these clusters will be referred to as ESBL-clusters, for brevity. In order to attempt to determine the biological significance of the identified ESBL-clusters (i.e. what kind of MGE element they are likely to represent), basic statistics were plotted (number of samples contained withing each cluster, length of longest contig in cluster in kbases, length distribution of all contigs is cluster relative to longest contig and distribution of sequence identity compared to the longest contig in the cluster). Presence of compound trasposons, AMR determinents and plasmid replicons were identified by using BLAST with default settings of each ESBL-cluster representative sequence (as determined by cd-hit i.e one, the longest, for each ESBL-cluster) against the insertion sequence finder (ISfinder) database and the SRST2 database, taking the top hit (as determined by bitscore) for any given location, and visualising the results in gggenes v0.3.2. To assess lineage association, the ESBL-clusters were mapped back to the core genome SNP tree. 

### Global _E. coli_ collection

In order to place the isolates from this study in a global context, published _E. coli_ assemblies were downloaded from the WSI servers. These included 149 ESBL-producing _E. coli_ from a single centre study in Chachoengsao province, eastern Thailand[@Runcharoen2017]. In this study, human clinical isolates from standard care in Bhuddhasothorn hospital were selected on the basis of the ESBL phenotype, and environmental samples were collected as part of a cross sectional study and selectively cultured for ESBL-E in 2014-2015. I also downloaded assemblies of 362 enterotoxogenic _E. coli_ (ETEC), selected for an ETEC genomic study from the Gothenburg University ETEC collection to represent a broad collection of ETEC isolated worldwide from 1980-2011[@VonMentzer2014]; 185 atypical enteropathogenic _E. coli_ (aEPEC) sequenced for a study of aEPEC and selected from samples frrm the Global Enteric Multicentre Study (GEMS) in seven centres in Africa and Asia between 2007-2011[@Ingle2016]; and 94 _E. coli_ from QECH in Blantyre, Malawi, a combination of invasive (bloodstream and CSF) and carriage isolates, selected for diversity in AMR phenotype from 1996-2014[@Musicha2017]. Details of the year, sample andcountry of isolation for these samples are given in the appendix to this chapter.

Phylogroup and MLST were determined for these context genomes as described above. AMR genes were identified with Ariba and the SRST2 database, as above, and context genomes were classified as ESBL if they contained any Bush-Jacoby group 2be ESBL gene .

### Statistical analysis

In order to explore clustering of AMR genes, the Jaccard index was calculated for a given AMR-gene pair using he Jaccard v0.1.0 package in R. The Jaccard index, a measure of the similarity of two sets of data, is defined as _intersection over union_; in this context, for a given pair of AMR genes $x$ and $y$, the Jaccard index $J(x,y)$ is the number of isolates that contain both gene $x$ and $y$ divided by the total number that contain either $x$ or $y$:

$$J(x,y) = \displaystyle \frac{| x \cap y |}{| x \cup y| }$$

By definition it lies between 0 ($x$ and $y$ never co-occur) and 1 ($x$ and $y$ always co-occur). Co-occurance matrices using the Jaccard index were plotted using the pheatmap v 1.0.12 package in R. The statistical significance of co-occurance of genes was assessed by generating 2x2 contingency tables for a given gene pair and p values generated using a Fisher's test with Bonferroni correction; a p value of less that 0.05 was considered statistically significant. Co-occurance networks of genes occuring commonly together (defined as Jaccard index > 0.5) at a rate greater than expected by chance (p < 0.05 following Bonferroni correction) and uncommonly occuring together (defined as Jaccard index < 0.1 and p < 0.05 following Bonferroni correction) were plotted using igraph v1.2.2 and ggraph v1.0.2 in R.

To explore hospital or community associations of any given _E. coli_ clade, the location of isolation was first plotted against the phylogenetic tree; location of isolation was classified as hospital, community, or recent hospital discharge (defined as a date of isolation within 2 weeks of hospital discharge). This latter category was used because it is possible that a patient could acquire an ESBL-E clone in hospital but only be sampled once leaving hospital; using only hospital isolated and community isolted categories could therefore introduce bias. Hospital or community association of each sequence cluster was assessed using a Fisher's test of proportion of hospital associated samples (defined as sum of hospital isolated and recent hospital discharge) for the given sequence cluster as compared to proportion of hospital associated samples in the remainder of the samples, with a Bonferroni correction for multiple comparisons. p < 0.05 was again considered statistically significant.

To compare within-patient to between-patient conservation of bacteria (as represented by core genome alignment and sequence cluster) and ESBL-containing MGE (as represented by the ESBL-clusters) several approaches were taken. Firstly, I assessed whether either sequence cluster or ESBL-cluster were conserved within an individual at all. I hypothesised that any within-patient correlation is likely to be a function of time: samples closer together in time may be more likely to be similar. To assess if this was the case for bacteria, pairwise core genome pseudosequence SNP distance was was calculated using snp-dists v0.4 (https://github.com/tseemann/snp-dists) for all samples and plotted against the time difference (in days) between samples, within and between patients, and with a smoothed curve fitted using a general additive model with cubic splines. Because of significant overplotting, this was also plotted as a 2D density plot. Based on these plots, the within and between patient SNP distances were compared in two post-hoc defined groups binned by time distance between the samples (50 days or less vs. more than 50 days), and distributions compared with Kruskal-Wallace tests.

I then compared the within patient temporal clustering of ESBL-clusters and sequence clusters, by estimating the proportion of within-patient samples that contain the same ESBL-cluster or sequence cluster, as a function of time; essentially a temporal autocorrelation function. To estimate this, I considered pairwise comparsion of all within-patient samples. For any given time between samples, $t$ I defined a window of +/-5 days and estimated the probabilities as the number of all within-patient sample pairs in the window $[t - 5, t+5]$ that contained the same sequence cluster or ESBL-cluster divided by the total number of all within-patient sample pairs within that time window. Exact binomial confidence intervals for these proportions were generated and probabilities plotted as a function of time. In order to estimate the probability of two samples containing the same sequence cluster of contig-cluster purely by chance, 1000 sample pairs were randomly drawn from all samples with replacement and the proportion of these samples that contained the same sequence cluster or ESBL-cluster calculated.

Finally, to inform the question as to what the likely unit of transmission in this system is, I assessed what was most conserved within patients, in pairwise sample comparison: bacteria (as represented by core gene sequence cluster), ESBL-containing MGE (as represented by ESBL-cluster), or both. Simple proportions in all-against-all pairwise comparison - stratified by whether between-patient or within-patient - were calculated: the proportion of samples that contain the same core gene sequence cluster only, the proportion of samples contain the same ESBL-cluster only, and the proportion that contain both sequence cluster and ESBL-cluster. Proportions were compared between within and between-patient strat in these three groups using Fisher's exact test, with p < 0.05 considered statistically significant.

## Results

### Samples and quality control

In total, 520 _E. coli_ underwent DNA extraction and were shipped from Malawi to WSI; these represented all sequential isolates at the time of final DNA extraction, which occured in two batches in February 2018 and October 2018. Kracken/Bracken read assignment of these samples is shown in Figure \@ref(fig:wgs-qc-braken). The majority of samples have > 90% or reads assigned to _E. coli_; a minority have < 90% of reads assigned to _E. coli_ but a very closely related species such as _Shigella_, amd as such are likely to be pure _E. coli_ culture with read misclassification. However, 12 samples have > 80% reads assigned to a non- _E. coli_ species such as _Klebsiella pneumoniae_. These samples were assumed to represent upstream species misidentification or, perhaps more likely, selection of the wrong sample from the freezer archive for culture and DNA extraction, given that for any sample ID there are often several bacterial species identified and cryopreserved. These samples were excluded from further analysis.

```{r wgs-qc-braken,  echo = F, warning = F, message = F, fig.scap="Kraken/bracken read assignment", out.extra='', fig.cap= "Read assignment by Kracken and Bracken of all samples", fig.align= "center",fig.height = 10, fig.width = 6 }

df1 <- read_tsv(paste0(wd, "D1_Braken_readscreen_species_composition.tsv"))
df2 <- read_tsv(paste0(wd, "D2_Bracken_species_composition.tsv"))
df3 <- read_tsv(paste0(wd, "D2_extra_krakenout_species_composition_16.tsv"))

df.brac <- bind_rows(df1, df2, df3)
df.brac[is.na(df.brac)] <- 0

melt(df.brac, id.vars = "name") -> dftemp

dftemp$variable <- factor(dftemp$variable, levels = unique(dftemp$variable[order(as.character(dftemp$variable))]))

dftemp$name <- factor(dftemp$name, levels = df.brac$name[order(df.brac$`Escherichia coli`, decreasing = T)])

ggplot(dftemp, aes(name, value, fill = variable)) + geom_col() + coord_flip() + scale_fill_brewer(palette = "Paired") +  theme_bw() + theme(axis.text.y = element_text(size = 2), legend.title = element_blank(), legend.text = element_text(size = 8)) + labs(x= "Sample ID", y = "% reads" ) 




```

```{r wgs-load-qc,  echo = F, warning = F, message = F}

# load sanger QC

df1 <- read.csv(paste0(wd, "5341.pathfind_stats.csv"), stringsAsFactors = F)
df2 <- read.csv(paste0(wd, "5510.pathfind_stats.csv"), stringsAsFactors = F)

#df2$`No. Het SNPs`<- as.numeric(df2$`No. Het SNPs`)

df.qc <- rbind(df1, df2)


# laod checkM

con <- file(paste0(wd, "checkm_quast/D1/checkm.report"))
lines <- readLines(con)
close(con)

con <- file(paste0(wd, "checkm_quast/D220190318/checkm.report"))
lines2 <- readLines(con)
close(con)

con <- file(paste0(wd, "checkm_quast/D220190503/checkm.report"))
lines3 <- readLines(con)
close(con)

# line 2 has headings - get em
strsplit(lines[[2]]," ") -> heads
heads[[1]][heads[[1]] != "" & heads[[1]] != "#"] -> heads

# sigh
heads <- c(paste0(heads[1], "_", heads[2]), paste0(heads[3], "_", heads[4]), heads[5:6],
           paste0(heads[7],"_",heads[8]), heads[9:16], paste0(heads[17], "_", heads[18]))

lines <- lines[-c(1,2,3, length(lines))]
lines2 <- lines2[-c(1,2,3, length(lines2))]
lines3 <- lines3[-c(1,2,3, length(lines3))]
 
 

lines <- lapply(lines, parse_checkm_lines, heads)
lines2 <- lapply(lines2, parse_checkm_lines, heads)
lines3 <- lapply(lines3, parse_checkm_lines, heads)

lines <- data.frame(do.call(rbind, lines))
lines2 <- data.frame(do.call(rbind, lines2))
lines3 <- data.frame(do.call(rbind, lines3))

lines <- rbind(lines, lines2, lines3)
rm(lines2)
rm(lines3)
lines[3:ncol(lines)] <- sapply(lines[3:ncol(lines)], function(x) as.numeric(as.character(x)))

names(lines)[names(lines) == "Bin_Id"] <- "Assembly"

#sub( "_1_", "_1#", q$Assembly) -> q$Assembly
sub(".contigs_spades", "", lines$Assembly) -> lines$Assembly

checkm <- lines
rm(lines)

## and quast
rbind(
  read.delim(paste0(wd, "checkm_quast/D1/transposed_report.tsv"), stringsAsFactors = F ),
  read.delim(paste0(wd, "checkm_quast/D220190318/transposed_report.tsv"), stringsAsFactors = F ),
  read.delim(paste0(wd, "checkm_quast/D220190503/transposed_report.tsv"), stringsAsFactors = F )
  ) -> quast

for (i in 1:10) {
sub("\\.", "", names(quast)) -> names(quast)
}

sub(".contigs_spades","", quast$Assembly) -> quast$Assembly
sub("_1_","_1#", quast$Assembly) -> quast$Assembly
sub("_2_","_2#", quast$Assembly) -> quast$Assembly

df.qc <- merge(df.qc, df.brac, by.x = "Lane.Name", by.y = "name", all.x = T)
df.qc <- merge(df.qc, checkm, by.x = "Lane.Name", by.y = "Assembly", all.x = T)
df.qc <- merge(df.qc, quast, by.x = "Lane.Name", by.y = "Assembly", all.x = T)

# remove non ESCO

df.qc <- subset(df.qc, `Escherichia coli` > 20)


```

Of the remaining 508 samples, there were a median (IQR) of `r paste0(median(df.qc$Reads), " (", quantile(df.qc$Reads, 0.25)[[1]], "-", quantile(df.qc$Reads, 0.75)[[1]], ")")` reads, with a median (IQR) depth of coverage (obtained by mapping a random 100Mbases to a refernce _E. coli_ genome, Escherichia coli strain K-12 substrain MG1655, NCBI reference NC_000913.3) of `r paste0(round(median(df.qc$Depth.of.Coverage)), " (", round(quantile(df.qc$Depth.of.Coverage, 0.25)[[1]]), "-", round(quantile(df.qc$Depth.of.Coverage, 0.75)[[1]]), ")")`. One sample had an order of magnitude lower number of reads (291556) with depth of coverage 0; this was assumed to represent sequencing failure and it was excluded from further analysis.

The output from quast and checkM are shown in Figure \@ref(fig:wgs-qc), where N50 (the minimum contig length upon which at least half assembled bases are contained) is plotted as a function of total assembled length. The expected _E. coli_ genome length is around 4.6Mb and most samples cluster close to this at a total assembled length of ~ 5Mb. Howvever it is clear that some assemblies  have failed, with low N50 and low assembled length. It is also apparent that some samples seem to be contaminated, as indicated by low N50 and much longer than expected total assembled length. Defining assembly failure as < 1Mb assembled length (triangles in the plot, n = 9) and contamination as checkM-defined contamination of > 25% (blue points in the plot, n = 24) and excluding both groups results in 33 further samples being excluded from further analysis.

In total, therefore, 46/520 (11%) of samples which were submitted for sequencing were excluded from downstream analysis. The remaining 474 samples represent 69% (474/686) of the cutured _E. coli_ in this study; 354 are from patients with sepsis, 86 are from hospitalised inpatients and 33 are from community members, with a median of 2 (range 1-5) samples per participant. N50, total assembled length and number of assembled contigs are shown in the appendix to this chapter.

```{r wgs-qc, echo = F, warning = F, message = F, fig.scap="N50 as a function of total assembly length for included assemblies", out.extra='', fig.cap= "N50 as a function of total assembled length. Failed assemblies with less than 1Mb assembled shown as triangles. Contaminated assemblies with checkM-defined contamination above 25\\% shown in blue.", fig.height = 3, fig.width = 5, fig.align="center"  }

df.qc <- subset(df.qc, Reads > 300000)

ggplot(df.qc, aes(Totallength, N50, colour = Contamination > 25, shape = Xcontigs < 10)) + geom_point() + theme_bw() + labs(shape = "< 1Mb assembled", colour = "Contamination > 25%", x = "Total assembled length (bp)") 

```

```{r wgs-mlst-pgroup, echo = F, warning = F, message = F }

mlst <- read_csv(paste0(wd, "phylogroup_and_mlst/mlst.csv"))
pgroups <- read_csv(paste0(wd, "phylogroup_and_mlst/phylogroups.csv"))
```

### Phylogroup, MLST and core genome phylogeny of study isolates

The commonset _E. coli_ phylogroup was phylogroup A: 204/473 (43%) samples belonged to phylogroup A, followed by phylogroup B2 (96/473 [20%]), F (53/473 [11%]), B1 (43/473 [9%]) and C (43/473 [9%]) and D (26/473 [5%]). Two samples were Clade I or II (so called cryptic clades) and 6/473 (1%) were unknown phylogroup using the Clermont PCR scheme. In the MLST analysis, 56 recognised sequence types (STs) were identified, and 12 samples were novel STs; however over half (249/473 [53%]) of samples were represented by the top seven most frequent STs (Figure \@ref(fig:wgs-mlst)). ST131 was the most commonly isolated sequence type (64/473 [14%] of isolates) followed by ST410 (45/473 [10%] of isolates) and ST167 (38/473 [8%] of isolates).

The Roary pan-genome pipeline identified a core genome in the study isolates of 2966 genes, with a pan-genome of 26840 genes. The resultant core gene pseudosequence of length 1388742 bases contained 99693 variable sites, which were used to infer the maximum likelihood phylogenetic tree. The IQTREE ModelFinder module determined that a general time reversible (GTR) model with FreeRate site heterogeneity with 5 parameters provided the best fit to the data. The inferred tree is shown in Figure \@ref(fig:wgs-mlst) along with isolate phylogroup and sequence types; in general, as expected, sequence types were largely monophyletic and phylogroups tended to cluster together.

```{r wgs-mlst, echo = F, warning = F, message = F, fig.cap="\\textit{E. coli} sequence type distribution", fig.height = 2, fig.width=6}

ggplot(mlst, aes(fct_infreq(ST))) + geom_bar()+
  xlab("E. coli sequence type") + ylab("Count") + theme(axis.text.x=element_text(size=5, angle = 90) )

```



```{r wgs-tree, echo = F, warning = F, message = F, fig.scap="ML phylogenetic tree of study \\textit{E. coli}", out.extra='', fig.cap="Maximum likelihood phylogenetic tree of included study \\textit{E. coli} isolates showing phylogroups and sequence types. Bootstrap support of less than 90\\% is indicated by a black circle at a given node. Scale bar indicates 0.01 SNPs/site.", fig.height = 10, fig.width=8}

tree <- read.tree(paste0(wd, "core_genome_tree/core_alnD2ESCO_snp_sites.aln.treefile"))
midpoint.root(tree) -> tree

rownames(pgroups) <- pgroups$Lane
dcast(mlst, lane ~ ST, fun.aggregate = length) -> mlst.onehot
#mlst.onehot <- subset(mlst.onehot, lane %in% tree$tip.label)

rownames(mlst.onehot) <- mlst.onehot$lane
mlst.onehot <- select(mlst.onehot, -lane)
n.mlsts <- sort(apply(mlst.onehot,2, sum), decreasing = T)
mlst.onehot[names(n.mlsts[n.mlsts > 1])] -> mlst.onehot

apply(mlst.onehot, 1, sum) == 0 -> mlst.onehot$other
mlst.onehot$other<- as.numeric(mlst.onehot$other)



mlst.onehot[mlst.onehot == 0] <- "Z_No"
mlst.onehot[mlst.onehot == 1] <- "Z_Yes"






#mlst.onehot[c(sort(as.numeric(names(mlst.onehot)[names(mlst.onehot) != "Novel"])), "Novel")] -> mlst.onehot


col <- c(hue_pal()(8), "lightgrey", "black")
names(col) <- c("A","B1","B2","C","Clade I or II", "D","F", "Unknown", "Z_No", "Z_Yes")

ggtree(tree) %>% gheatmap(select(pgroups, Phylogroup), font.size = 3, width = 0.1, 
                          colnames_position = "top", color = NA, colnames_offset_y = 5) +
  scale_fill_manual(values =col, labels = c("A","B1","B2","C","Clade I/II", "D","F", 
                                            "Unknown", "Absent", "Present"))-> p1x

leg1 <- get_legend(p1x)

ggtree(tree) %>% gheatmap(select(pgroups, Phylogroup), width = 0.1, 
                          color = NA, font.size = 4, colnames_angle = 90,
                          colnames_position = "top", colnames_offset_y = 30)  %>% gheatmap((mlst.onehot), font.size = 2.5, color = NA, 
                          colnames_angle = 90, offset=.03, colnames_offset_y = 10, colnames_position = "top",)  + geom_treescale(x =0, y = 0, offset = 5)  +   geom_point2(aes(subset = as.numeric(label) < 90 & !isTip), size = 1) + theme(legend.position = "none") +
  scale_fill_manual(values =col, labels = c("A","B1","B2","C","Clade I or II", "D","F", 
                                            "Unknown", "Absent", "Present")) + annotate("text", x= 0.28, y =510, label = "Sequence Type") -> p

plot_grid(p, leg1, ncol = 2, rel_widths = c(1,0.1))


```


### Study isolates in a global context

The global collection of _E. coli_ comprised 1273 samples, including the 473 from this study. 753/1253 (60%) were from Africa, 335/1253 (27%) from Asia and 167 (13%) from South America. The majority of samples, 1026/1253 (82%), were from stool, with 106/1253 (8%) truly invasive samples from blood or CSF and 63/1253 (5%) possibly invasive samples from urine, pus, or sputum. 65/1253 (5%) of samples were environmental, all from Thailand. 670/1253 (53%) of samples contained at least one ESBL-encoding gene. The majority of isolates with ESBL gene (622/670 [92%]) came from this study or the Thai ESBL study. Phylogroup A was the commonest phylogroup in the global collection (482/1273 [38%]), followed by B1 (333/1273 [26%]) and B2 (191/1273 [15%]); phylogroup C was uncommon in the global collection (74/1273 [6%]) but the majority of the phylogroup C samples came from this study (43/74 [58%]). All of these 43 phyogroup C isolates belonged to a single ST, ST410; this ST was not seen at all in the previous Malawian study of largely invasive isolates, despite beingthe second-commonest ST in this study. ST131 was again the commonest ST in the global collection.

The Roary pan-genome pipeline identified 2872 core genes in a pan genome of 44840 genes; this large pan-genome is consistent with the open _E. coli_ pan genome that will continue to increase in size as isolates are added. The core gene alignment contained 604817 bases with 77194 variable sites, which were used to infer the maximum likelihood phylogenetic tree, using same nucleotide substitution model as previously.

The inferred tree is shown in Figure \@ref(fig:wgs-global-tree)). Isolates from this study are distributed throughout the tree, and there is widespread mixing of isolates from diverse geographic regions. Though invasive isolates are spread throughout the tree, there is a tendance for them to cluster together, particularly in phylogroup B2, a phylogroup with has a recognised association with ExPEC (needs ref). The Malawian ST410 isolates clusterd tightly together, though are most closely related to clinical ESBL-procucing ST410 isolates from Thailand. By comparison, ST131 isolates from this study were distributed amongst ST131 isolates from other studies, both in Malawi and elsewhere.

```{r wgs-global-tree, echo = F, warning = F, message = F, fig.scap="ML phylogenetic tree of study \\textit{E. coli} in global context", out.extra='', fig.cap="Midpoint rooted maximum likelihood phylogenetic tree of included study \\textit{E. coli} isolates along with global context isolates, showing phylogroups, source sample type and continent of isolation (coloured bars). Dark grey bars indicate isolates from this study or isolates with ESBL gene presence, as labelled (this study or ESBL, respectively). Two most frequently isolated STs (131 and 410) labelled.  Bootstrap support of less than 90\\% is indicated by a black circle at a given node. Scale bar indicates 0.009 SNPs/site.", fig.height = 10, fig.width=8}


tree <- read.tree(paste0(wd, "global_tree/all_ESCO_core_snps.aln.treefile"))

midpoint.root(tree) -> tree

#metadata

metadata <- read_csv(paste0(wd, "global_tree/all_ESCO_metadata.csv"))
sub("#", "_", metadata$Lane.Name ) -> metadata$Lane.Name
metadata <- data.frame(metadata)
rownames(metadata) <- metadata$Lane.Name

metadata$Continent <- NA
metadata$Continent[metadata$Country == "Argentina" |
                     metadata$Country == "Bolivia" |
                     metadata$Country == "Guatemala" |
                     metadata$Country == "Mexico" |
                     metadata$Country == "Venezuela" ] <- "Z_S. America"

metadata$Continent[metadata$Country == "Bangladesh" | 
                     metadata$Country == "Burma" |
                     metadata$Country == "China" |
                     metadata$Country == "India" |
                     metadata$Country == "Indonesia" |
                     metadata$Country == "Japan" |
                     metadata$Country == "Nepal" |
                     metadata$Country == "Pakistan" |
                     metadata$Country == "thailand" |
                     metadata$Country == "Thailand"] <- "Z_Asia"

metadata$Continent[metadata$Country == "Egypt" | 
                     metadata$Country == "Gambia" |
                     metadata$Country == "Guinea Bissau" |
                     metadata$Country == "Kenya" |
                     metadata$Country == "Malawi" |
                     metadata$Country == "Mali" |
                     metadata$Country == "Morocco" |
                     metadata$Country == "Mozambique" |
                     metadata$Country == "Tunisia" |
                     metadata$Country == "Zaire"] <- "Z_Africa"

metadata$source.cat <- metadata$Source

metadata$source.cat[metadata$source.cat == "Canal" |
                      metadata$source.cat == "Farm" |
                      metadata$source.cat == "Untreated hospital sewage"] <- "_Environment"

metadata$source.cat[metadata$source.cat == "Blood" |
                      metadata$source.cat == "CSF"] <- "_Clinical - _Blood/CSF"

metadata$source.cat[metadata$source.cat == "Urine" |
                      metadata$source.cat == "Pus" |
                      metadata$source.cat == "Sputum" ] <- "_Clinical - Other"

metadata$source.cat[metadata$source.cat == "stool" |
                      metadata$source.cat == "Stool" |
                      metadata$source.cat == "RS"] <- "_Clinical - _Stool"

metadata$malawi <- as.factor(as.numeric(metadata$Country == "Malawi"))

metadata$this_study <- as.factor(as.numeric(grepl("DASSIM", metadata$study)))
# pgroups

pgroups <- read.csv(paste0(wd, "global_tree/phylogroups_and_mlst/all_ESCO_pgroups.csv"), stringsAsFactors = F)
pgroups.d <- read.csv(paste0(wd, "phylogroup_and_mlst/phylogroups.csv"), stringsAsFactors = F)
pgroups <- select(pgroups, V1, phylogroup)
names(pgroups) <- c("Lane", "Phylogroup") 
pgroups <- rbind(pgroups, pgroups.d)
rownames(pgroups) <- pgroups$Lane
pgroups$Phylogroup[pgroups$Phylogroup == "Clade I"] <- "Clade I or II"

# mlst


## load mlst and plot

mlst.d <- read.csv(paste0(wd, "phylogroup_and_mlst/mlst.csv"), stringsAsFactors = F)
mlst <- read.csv(paste0(wd, "global_tree/phylogroups_and_mlst/mlst_summary.csv"), stringsAsFactors = F)
mlst <- select(mlst, lane, ST)

mlst <- rbind(mlst.d, mlst)
dcast(mlst, lane ~ ST, fun.aggregate = length) -> mlst.onehot
#mlst.onehot <- subset(mlst.onehot, lane %in% tree$tip.label)

rownames(mlst.onehot) <- mlst.onehot$lane
mlst.onehot <- select(mlst.onehot, -lane)
n.mlsts <- sort(apply(mlst.onehot,2, sum), decreasing = T)
mlst.onehot[names(n.mlsts[n.mlsts > 1])] -> mlst.onehot

apply(mlst.onehot, 1, sum) == 0 -> mlst.onehot$other
mlst.onehot$other<- as.numeric(mlst.onehot$other)
mlst.onehot[mlst.onehot == 0] <- "Z_No"
mlst.onehot[mlst.onehot == 1] <- "Z_Yes"

#### AMR

srst2 <- read.csv(paste0(wd, "global_tree/srst2/srst2_summary_cleaned_pres_abs.csv"), stringsAsFactors = F)
srst2.d <- read.csv(paste0(wd, "amr/srst2_pres_abs.csv"), stringsAsFactors = F)
bind_rows(srst2, srst2.d) -> srst
srst[is.na(srst)] <- 0
rownames(srst) <- srst$name

narrow_spec <- c("TEM_1", "TEM_95", "TEM_135", "TEM_90",
                 "OXA_1", 
                 "SHV_1")

esbl <- c(names(srst)[grepl("CTX", names(srst))], 
          names(srst)[grepl("TEM", names(srst))],
          names(srst)[grepl("OXA", names(srst))],
          names(srst)[grepl("SHV", names(srst))],
          names(srst)[grepl("NDM", names(srst))])

esbl <- esbl[!(esbl %in% narrow_spec)]
data.frame(esbl = apply(srst[esbl], 1,sum)) -> esbl.cat
esbl.cat$esbl[esbl.cat$esbl > 0] <- 1
esbl.cat$Lane <- rownames(esbl.cat)
esbl.cat$esbl <- as.factor(esbl.cat$esbl)
##

col <- c(hue_pal()(8), "white", "grey30", brewer.pal(4,"Set1"),brewer.pal(4,"Dark2") )
names(col) <- c("A","B1","B2","C","Clade I or II", "D","F", "Unknown",
                "0", "1","_Clinical - _Blood/CSF", "_Clinical - _Stool", "_Environment",  "_Clinical - Other",
                "Z_Africa", "Z_Asia", "Z_S. America"
               )

names(metadata)[names(metadata) == "this_study"] <- "This study"
names(esbl.cat)[names(esbl.cat) == "esbl"] <- "ESBL"
names(metadata)[names(metadata) == "Source"] <- "Source1"
names(metadata)[names(metadata) == "source.cat"] <- "Source"

ggtree(tree, size = 0.3) %>% 
  gheatmap(select(metadata, `This study`), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80,offset = 0.006 ) %>%
  gheatmap(select(pgroups, Phylogroup), width = 0.05, 
           color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.001)  %>%
  gheatmap(select(esbl.cat, ESBL), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.011 ) %>%
  gheatmap(select(metadata, Source), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.016 ) %>%
  gheatmap(select(metadata, Continent), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.021 ) +
 # gheatmap(select(metadata, malawi), width = 0.04, color = NA, font.size = 4, colnames_angle = 90,
      #     colnames_position = "top", colnames_offset_y = 80, offset = 0.025 )+ 
  scale_fill_manual(values =col) + theme(legend.position = "none") +
  geom_treescale(x =0.02, y = 1100, offset = 5)  +   
  geom_point2(aes(subset = as.numeric(label) < 90 & !isTip), size = 0.3)+ 
  geom_cladelabel(node = 1351, label = "ST131", fontsize = 2.5) + 
  geom_cladelabel(node = 1866, label = "ST410", fontsize = 2.5) -> p #, labels = c("0", "1","A","B1","B2","C","Clade I or II", "D","F", 
    
ggtree(tree) %>% 

  gheatmap(select(pgroups, Phylogroup), width = 0.05, 
           color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80) +
  scale_fill_manual(values =col) + labs(fill = "Phylogroup") + theme(legend.title = element_text()) -> p1x

leg.pgroup <- get_legend(p1x)

ggtree(tree) %>% 
  
  gheatmap(select(metadata, Source), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.015 ) +
  scale_fill_manual(values =col, labels = c("Clinical: Blood/CSF", "Clinical: Stool", "Clinicial: Other", "Environment", "")) + labs(fill = "Source") + theme(legend.title = element_text())-> p1x
  
leg.source <- get_legend(p1x)

ggtree(tree) %>% 
  
  gheatmap(select(metadata, Continent), width = 0.05, color = NA, font.size = 4, colnames_angle = 90,
           colnames_position = "top", colnames_offset_y = 80, offset = 0.020 ) +
  scale_fill_manual(values =col, labels = c("Africa", "Asia", "S. America","")) + labs(fill = "Continent") + theme(legend.title = element_text())-> p1x

leg.continent <- get_legend(p1x)

ggarrange(p, 
          ggarrange(NULL, leg.pgroup, leg.source, leg.continent, NULL, ncol = 1, nrow = 5, heights = c(1,1,0.7,0.4,1)),
          ncol = 2, nrow = 1, widths = c(3,1))

```

### Antimicrobial resistance determinents

#### $\beta$-lactamse genes 

Identified AMR genes are shown in Figure \@ref(fig:wgs-amr-sum). All isolates contained at least one gene that conferred resistance to third-generation cephalosporins, either an ESBL gene (n= 472) or a carbapenamase (n=1). The majority of ESBL-gene containing isolates contained only one ESBL gene (432/472 [92%]); fewer contained 2 (36/472 [8%]) or 3 (4/472 [1%]); none contained more than 3. $bla_{CTX-M}$ was the commonest ESBL gene, and over two thirds (320/473 [68%]) of isolates contained $bla_{CTX-M-15}$. ESBL $bla_{SHV}$ (27/473 [6%] of isolates) and $bla_{TEM}$ (16/473 [3%] of isolates) were also seen, though were rare; however, narrow spectrum $bla_{TEM}$ $\beta$-lactamases were common (297/473 [62%] of isolates). This pattern was also seen with $bla_{OXA}$ $\beta$-lactamases: narrow spectrum $bla_{OXA-1}$ was commonly seen (187/473 [40%] of isolates) but ESBL $bla_{OXA}$ genes were very unusual, occuring in only 2/473 (0.4%) of isolates. Plasmid-mediated $ampC$ genes were identified in 45/473 (9%) of isolates, almost all (44/45) $bla_{CMY-42}$; this was unexpected as all of these isolates were comfirmed to be ESBL-producers by combination disc testing. This testing uses cephalosporin-containing discs both with and without clavulanic acid, and confirms EBSL production by a difference in zone size between these discs, as ESBL enzymes are inactivated by clavulanic acid. However, the cephalosporins used in this test are hydrolysed by $ampC$ enzymes, and if these isolates were producing such enzymes it would be expected to confer cephalosporin resistance regardless of the presence or absence of clavulanic acid, and the test would therefore not confirm ESBL production. This was not the case for any of these isolates; none of them hydorolised the cephalosporins used in the combination disc testing. It may be that the $bla_{CMY}$ genes were not expressed.

The carbapenamase gene identified was s $bla_{NDM-5}$; the isolate harbouring this gene was recovered from the  stool of a 67-year old man with no history of foreign travel nor hospitalistion. He had been admitted to the hospital with fever seven days previously and treated with seven days of intravenous ceftriaxone for sepsis, the source of which was not clear. He made an uneventful recovery, and no carpapenamase-containing isolate was recovered from his stool at any other time. The $bla_{NDM-5}$ gene was carried on a partially assembled IncX3 plasmid. BLAST of this assembly against the NCBI database showed that this contig had 99% sequence identity with a previously sequenced pNDM-MGR194 46.2 kbp blaNDM-5 containing Inc-X3 plasmid found in India between 2011-13[@Krishnaraju2015]. We fully assembled the plasmid by mapping reads back to pNDM-MGR194 with Burrows-Wheeler alignment and found it to be extremely similar, with only 13 SNPs compared to pNDM-MGR194.

```{r wgs-amr-sum, echo = F, warning = F, message = F, fig.scap="Frequency distribution of AMR genes", out.extra='', fig.cap="A: Frequency distribution of AMR genes identified in isolates. Class of antimicrobial to which gene confers resistance is shown. B: Number of isolates with any mutation to a given class. Any mutation that could possibly confer resistance to a given class is included, including any mutation in the QRDR for quinolones. C: Phenotypic resistance patterns for subset of samples in this analysis that also underwent phenotypic testing (n = 449)", fig.height = 10, fig.width=8}

m.c <- read.csv(paste0(wd, "amr/srst2_pres_abs.csv"), stringsAsFactors = F)
q.res <- read.csv(paste0(wd, "amr/qrdr_mut_presence.csv"), stringsAsFactors = F)
amr <- merge(m.c, q.res, by.x = "name", by.y = "sample")
amr.melt <- melt(amr) %>% subset(value > 0)

amr.melt$class <- NA
amr.melt$class[grepl("Par", amr.melt$variable) | 
                 grepl("Gyr", amr.melt$variable) | 
                 grepl("Par", amr.melt$variable) |
                 grepl("Qnr", amr.melt$variable) |
                 grepl("Qep", amr.melt$variable) |
                 grepl("Oqx", amr.melt$variable) |
                 grepl("Nor", amr.melt$variable)] <- "Quinolone"

amr.melt$class[grepl("Tet", amr.melt$variable) ] <- "Tetracycline"

narrow_spec <- c("TEM_1", "TEM_95", "TEM_135", "TEM_90",
                 "OXA_1", 
                 "SHV_1")

amr.melt$class[amr.melt$variable %in% esbl] <- "ESBL"
amr.melt$class[amr.melt$variable %in% narrow_spec] <- "Penicillins"

amr.melt$class[grepl("Sul", amr.melt$variable) ] <- "Sulphonamides"

amr.melt$class[grepl("Str", amr.melt$variable) | 
                 grepl("Aad", amr.melt$variable) | 
                 grepl("Aac", amr.melt$variable) |
                 grepl("Aph", amr.melt$variable) |
                 grepl("Rmt", amr.melt$variable) |
                 grepl("APH", amr.melt$variable)] <- "Aminoglycosides"

amr.melt$class[grepl("Sat", amr.melt$variable) ] <- "Streptothricin"

amr.melt$class[grepl("LEN", amr.melt$variable) ] <- "Penicillins"
amr.melt$class[grepl("SCO", amr.melt$variable)  |
                grepl("BlaZ", amr.melt$variable) |
                 grepl("CARB", amr.melt$variable) |
                 grepl("PBP", amr.melt$variable) ] <- "Penicillins"

amr.melt$class[grepl("NDM", amr.melt$variable) ] <- "Carbapenems"

amr.melt <- subset(amr.melt, variable != "MrdA")
amr.melt <- subset(amr.melt, variable != "LAP_2")

amr.melt$class[grepl("Mph", amr.melt$variable) ] <- "Tetracyclines"
amr.melt$class[grepl("Mef", amr.melt$variable) |
                grepl("Erm", amr.melt$variable)  ] <- "Tetracyclines"

amr.melt$class[grepl("Fos", amr.melt$variable) ] <- "Fosfomycin"

amr.melt$class[grepl("Cat", amr.melt$variable) | 
                 grepl("FloR", amr.melt$variable) |
                 grepl("Cml", amr.melt$variable) ] <- "Choramphenicol"

amr.melt$class[grepl("Dha", amr.melt$variable) | 
                 grepl("CMY", amr.melt$variable) ] <- "AmpC"

amr.melt$class[grepl("Dfr", amr.melt$variable) ] <- "Trimethoprim"


amr.melt$class[grepl("Arr", amr.melt$variable) ] <- "Rifampicin"

amr.melt <- subset(amr.melt, !(grepl("Amp", variable) | grepl("AMP", variable)))

classlevels <- amr.melt %>% group_by(variable, class) %>% summarise(n = n())

temp <- c("Penicillins", "ESBL", "AmpC", "Carbapenems")


classlevels$class <- factor(classlevels$class, levels = rev(c(temp, unique(amr.melt$class)[!(unique(amr.melt$class) %in% temp)])))

amr.melt$variable<- factor(amr.melt$variable, levels = classlevels$variable[order(classlevels$class, classlevels$n)])



col <- c(brewer.pal(8,"Dark2"),brewer.pal(8,"Dark2") )

data.frame((table(classlevels$class))) -> linebreaks
linebreaks$cumsum <- cumsum(linebreaks$Freq)

start <- list()
end <- list()
label <- list()

for (i in 1:nrow(linebreaks)) {
  if (i == 1) {
    start[[i]] <- 1
    end[[i]] <- linebreaks$cumsum[i]
    label[[i]] <- as.character(linebreaks$Var1[i])
  } else {
    start[[i]] <- linebreaks$cumsum[i-1] + 1
    end[[i]] <- linebreaks$cumsum[i]
    label[[i]] <- as.character(linebreaks$Var1[i])
  }
}

do.call(rbind, start) -> start
do.call(rbind, end) -> end
do.call(rbind, label) -> label

data.frame(start = start, end = end, label = label) -> pos

pos$text.y <- pos$start + (pos$end - pos$start)/2

pos$start == pos$end -> chflag

pos$start[chflag] <- pos$start[chflag] - 0.2
pos$end[chflag] <- pos$end[chflag] + 0.2


ggplot(amr.melt, aes(variable, fill = class)) + geom_bar()+ theme_bw() + theme(axis.text.y = element_text(size = 5), legend.position = "none", plot.margin = unit(c(0.2,0.2,0.2,3.5), "cm"), axis.title = element_blank() )   + coord_flip(ylim = c(-5,473),clip = "off", expand = FALSE) +  annotate(geom = "segment", x = pos$start, xend = pos$end, y = -100, yend = -100 )  + annotate(geom = "text", y = -110, x = pos$text.y, label = pos$label, size = 3, hjust = 1) -> p1

amr.class <- unique(select(amr.melt, name, class))
amr.class$class <- factor(amr.class$class, levels = rev(levels(classlevels$class)))

ggplot(amr.class, aes(class, fill = as.character(class))) + geom_bar() +theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position = "none", axis.title = element_blank()) -> p2



# load phenotypic data

orgs <- read.csv("/Users/joelewis/Documents/PhD/Data/ESBL/ESBL_orgs.csv", stringsAsFactors = F)
orgs <- subset(orgs,profile_name == "DASSIM Culture")

orgs <- orgs[c(1,4,5:10)]

melt(orgs, id.vars = c('sample_number', 'organism')) -> orgs
subset(orgs, value != "") -> orgs
subset(orgs, organism == "Escherichia coli" ) -> orgs


#orgs$value <- factor(orgs$value, levels = c("Resistant", "Sensitive", "Intermediate" ))
orgs$variable <- map_chr(as.character(orgs$variable), function(x) strsplit(x , "\\.")[[1]][[1]])



sub("#","_", sample_ids$Lane) -> sample_ids$Lane
subset(sample_ids, Lane %in% tree$tip.label) -> sample_ids
orgs <- subset(orgs, sample_number %in% sample_ids$Supplier.Name)
orgsall <- orgs
subset(orgs, value != "Sensitive") -> orgs

rbind(orgs, 
      data.frame(sample_number = unique(orgs$sample_number),
                 organism = rep("Escherichia coli", length(unique(orgs$sample_number))),
                 variable = rep("ESBL", length(unique(orgs$sample_number))),
                 value = rep("Resistant", length(unique(orgs$sample_number)))
      )
      )-> orgs
                 
                                
cols <- hue_pal()(14)

orgs$variable <- factor(orgs$variable, 
                        levels = c("ESBL", "Meropenam", "Amikacin", "Gentamicin",
                                   "Chloramphenicol", "Cotrimoxazole", "Ciprofloxacin") )
ggplot(orgs, aes(variable, fill =variable)) + geom_bar() + theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position  = "none", axis.title = element_blank()) + 
  scale_fill_manual(values = cols[c(5,3,1,1,4,14,8)]) -> p3

ggarrange(p1, ggarrange(NULL,p2, p3,NULL, ncol = 1, nrow = 4, labels =c(NA,"B","C", NA), heights = c(0.5,1,1,0.5)), ncol = 2, nrow = 1, widths = c(1.5,1), labels = c("A", NA))

```

#### Quinolone resistance genes

116/473 (25%) of isolates contained plasmid-mediated quinolone resistance PMQR) genes, either $qnr$ or $qep$. Mutations were identified in at least one quinolone resistance-determining regions (QRDR) in 349/449 (78%) of isolates. The majority of mutations were well-described QRDR mutations (codon 83 and 87 in $gyrA$, codon 80 and 84 in $parC$ and codon 458 in $parE$, Figure \@ref(fig:wgs-amr-quin)A). QRDR mutations tended to cluster togather (Figure \@ref(fig:wgs-amr-quin)B) but alone they correlated poorly with phenotypic resistance. Of the 449 samples with available phenotypic sensitivity data, 294/449 (65%) were intermediate or resistant to ciprofloxacin, but 349/449 (78%) had a mutation in any codon in one of the four QRDR; presence of any QRDR mutation together with presence of PMQR had sensitivity of 95% (95% CI 93-98%) but specificity of 23% (95% CI 17-31%) for phenotypic quinolone resistance. Presence of mutations at all of codon 83 and 87 of $gyrA$ and at codon 80 of $parC$ had improved, but still poor, discrimination for phenotypic resistance with sensitivity 89% (95% 85-93%) and specificity 51% (95% CI 43-59%).

```{r wgs-quin-setup, include = F}

read_tsv(paste0(wd, "amr/QRDR_ariba_output.tsv")) -> q.df




names(q.df)[length(names(q.df))] <- "sample"

subset(q.df, ref_name != "ref_name") -> q.df

sub("#", "_", q.df$sample) -> q.df$sample
subset(q.df, sample %in% tree$tip.label) -> q.df

select(q.df, sample, ref_name, ref_ctg_change) -> q.df

q.df$codon_posn <- str_sub(q.df$ref_ctg_change,2,-2)

q.df$wt_codon <-  str_sub(q.df$ref_ctg_change,1,1)

q.df$mut_codon <- str_sub(q.df$ref_ctg_change,-1,-1)

q.df$codon_posn <- as.numeric(q.df$codon_posn)

q.df <- subset(q.df, (ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108) |
                 (ref_name == "GyrB" & codon_posn >= 331 & codon_posn <= 480) |
                (ref_name == "ParC" & codon_posn >= 38 & codon_posn <= 169) |
                (ref_name == "ParE" & codon_posn >= 365 & codon_posn <= 525) )

df.gyra <- subset(q.df, ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108)

q.df %>% dplyr::group_by(ref_name, codon_posn) %>% dplyr::summarise(n=n()) -> df.sum

df.gyra <- subset(df.sum, ref_name == "GyrA" & codon_posn >= 67 & codon_posn <= 108)
df.gyrb <- subset(df.sum, ref_name == "GyrB" & codon_posn >= 331 & codon_posn <= 480)
df.parc <- subset(df.sum, ref_name == "ParC" & codon_posn >= 38 & codon_posn <= 169)
df.pare <- subset(df.sum, ref_name == "ParE" & codon_posn >= 365 & codon_posn <= 525)


ggplot(df.gyra, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(67,108)) +
  theme_bw() + ylim(c(0,400)) + ggtitle("GyrA") + xlab("Codon position")-> p1
  
ggplot(df.gyrb, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(331,480)) +
  theme_bw() + ylim(c(0,400)) + ggtitle("GyrB")  + xlab("Codon position") -> p2
     
ggplot(df.parc, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(38,169)) +
  theme_bw() + ylim(c(0,400)) + ggtitle("ParC")  + xlab("Codon position")-> p3

ggplot(df.pare, aes(x= codon_posn, y = n)) + geom_point()  + xlim(c(365,525)) +
  theme_bw()  + ylim(c(0,400)) + ggtitle("ParE") + xlab("Codon position") -> p4



orgs.cip <- subset(orgsall, variable == "Ciprofloxacin")

orgs.cip <- merge( select(sample_ids, Supplier.Name, Lane),select(orgs.cip,sample_number, value),   by.x = "Supplier.Name", by.y = "sample_number")

orgs.cip <- merge(orgs.cip, q.res, by.x = "Lane", by.y = "sample")

q.df$key <- paste0(q.df$ref_name, q.df$codon_posn)

q.df.new <- subset(q.df, key == "GyrA83" |  key == "GyrA87" | key == "ParC80" | key == "ParE458")

dcast(q.df.new,sample ~ key, fun.aggregate = length) -> q.df.new

orgs.cip <- merge(orgs.cip, q.df.new, by.x = "Lane", by.y = "sample", all.x = T)
orgs.cip[is.na(orgs.cip)] <- 0
orgs.cip$value[orgs.cip$value == "Resistant" |orgs.cip$value == "Intermediate" ] <- 1
orgs.cip$value[orgs.cip$value == "Sensitive" ] <- 0
orgs.cip$value <- as.numeric(orgs.cip$value)

pmqr <- m.c[ c("name", grep("Qnr", names(m.c), value = T) , grep("Qep", names(m.c), value = T) )]

pmqr$Qnr <- apply(pmqr[grep("Qnr", names(pmqr), value = TRUE )], 1, sum)
pmqr$Qep <- apply(pmqr[grep("Qep", names(pmqr), value = TRUE )], 1, sum)

pmqr$Qnr[pmqr$Qnr > 1] <- 1
pmqr$Qep[pmqr$Qep > 1] <- 1

orgs.cip <- merge(orgs.cip, select(pmqr, name, Qnr, Qep), by.x = "Lane", by.y = "name")
names(orgs.cip)[names(orgs.cip) == "value"] <- "Phenotypic resistance"

orgs.cip$gyra.parc.and.plasm <-(orgs.cip$GyrA83 & orgs.cip$GyrA87 & orgs.cip$ParC80) | orgs.cip$Qnr | orgs.cip$Qep

orgs.cip$gyra.parc.or.plasm <-(orgs.cip$GyrA83 | orgs.cip$GyrA87 | orgs.cip$ParC80) | orgs.cip$Qnr | orgs.cip$Qep

orgs.cip %>% summarise(TP = sum(`Phenotypic resistance` == 1 & gyra.parc.and.plasm == TRUE),
                       FP = sum(`Phenotypic resistance` == 0 & gyra.parc.and.plasm == TRUE),
                       TN = sum(`Phenotypic resistance` == 0 & gyra.parc.and.plasm == FALSE),
                       FN = sum(`Phenotypic resistance` == 1 & gyra.parc.and.plasm == FALSE)) -> summ

#binom.test(summ$TP, (summ$TP + summ$FN))
#binom.test(summ$TN, (summ$TN + summ$FP))

orgs.cip %>% summarise(TP = sum(`Phenotypic resistance` == 1 & gyra.parc.or.plasm == TRUE),
                       FP = sum(`Phenotypic resistance` == 0 & gyra.parc.or.plasm == TRUE),
                       TN = sum(`Phenotypic resistance` == 0 & gyra.parc.or.plasm == FALSE),
                       FN = sum(`Phenotypic resistance` == 1 & gyra.parc.or.plasm == FALSE)) -> summ2

#binom.test(summ2$TP, (summ2$TP + summ2$FN))
#binom.test(summ2$TN, (summ2$TN + summ2$FP))
#dev.off()
grid.grabExpr(pheatmap(orgs.cip[c(3,8,9,10,11,12,13)], cluster_cols = FALSE, show_rownames = FALSE,
                       legend = FALSE)) -> hm
hm <- as_ggplot(hm)
#dev.off()

ggarrange(ggarrange(p1,p2,p3,p4, ncol =2, nrow =2), hm, ncol = 2, nrow = 1, labels = c("A", "B")) -> parrange

ggsave(paste0(wd,"quinolone_plot.pdf"), plot = parrange, width = 18, height = 8, units = "cm")

```




```{r wgs-amr-quin, echo = F, warning = F, message = F, fig.scap="Quinolone resistance mutations and phenotypic resistance", out.extra='', fig.cap="A: Mutation positions in Quinolone resistance-determining regions, showing that most mutations are well-recognised (see text for details) B: Co-occurance heatmap of QRDR mutations, plamid-mediated quinolone resistance and phenotypic resistance. Each row is one sample, red = presence, blue = absence."}

knitr::include_graphics(paste0(wd,"quinolone_plot.pdf"))
```

#### Aminoglycoside resistance

Blah blah blah blah aminogycoside resistance.

Yes indeedy aminoglycosie resistance

Oh yes

### Testing associations: Sequence clusters and ESBL-clusters


## Appendix